---
title: "ArchR_IndividualPatient_ReclusteringAnalysis"
output: html_document
date: "2023-12-05"
---

Load libraries etc.
```{r}
library(Seurat)
library(ggplot2)
library(ArchR)
library(BSgenome.Hsapiens.UCSC.hg38)
addArchRThreads(threads = 1)
addArchRGenome("hg38")
options(future.globals.maxSize=10000000000000000) 
```
Load in
```{r}
archR <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_NoControls_NoProgens_chromVAR_DownstreamObject")
```
# Patient subset function
```{r}
patient_subset <- function(archR_obj, id_str) {
  idx <- BiocGenerics::which(archR$samplesClean == id_str)
  cells <- archR_obj$cellNames[idx]
  sub_archR <- archR_obj[cells,]
  return(sub_archR)
}
```
# Subset patients
```{r}
p01_076_archr <- patient_subset(archR, "01-076")
p01_115_archr <- patient_subset(archR, "01-115")
p01_131_archr <- patient_subset(archR, "01-131")
p01_163_archr <- patient_subset(archR, "01-163")
p01_190_archr <- patient_subset(archR, "01-190")
p04_003_archr <- patient_subset(archR, "04-003")
p04_006_archr <- patient_subset(archR, "04-006")
wm25_archr <- patient_subset(archR, "WM25")
wm43_archr <- patient_subset(archR, "WM43")
wm46_archr <- patient_subset(archR, "WM46")
wm47_archr <- patient_subset(archR, "WM47")
wm54_archr <- patient_subset(archR, "WM54")
wm65_archr <- patient_subset(archR, "WM65")
```
Reclustering individual patient function
```{r}
patient_recluster <- function(archR_obj, sample_str) {
  print(paste0("Reclustering patient ", sample_str, "..."))
  print("Running Iterative LSI...")
  
  # Add iterative LSI
  archR_obj <- addIterativeLSI(
  ArchRProj = archR_obj,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)
  print("Adding UMAP...")
  
  archR_obj <- addClusters(
    input = archR_obj,
    reducedDims = "Patient_Recluster",
    method = "Seurat",
    name = "Individual_Reclusters",
    resolution = 0.8,
    force = T
)
  
# Add UMAP
  archR_obj <- addUMAP(
      ArchRProj = archR_obj, 
      reducedDims = "Patient_Recluster", 
      name = "Patient_Recluster_UMAP", 
      nNeighbors = 30, 
      minDist = 0.5, 
      metric = "cosine",
      force = T
  )
  return(archR_obj)
}
```
# Recluster patients
```{r}
p01_076_archr <- patient_recluster(p01_076_archr, "01-076")
p01_115_archr <- patient_recluster(p01_115_archr, "01-115")
p01_131_archr <- patient_recluster(p01_131_archr, "01-131")
p01_163_archr <- patient_recluster(p01_163_archr, "01-163")
p01_190_archr <- patient_recluster(p01_190_archr, "01-190")
p04_003_archr <- patient_recluster(p04_003_archr, "04-003")
p04_006_archr <- patient_recluster(p04_006_archr, "04-006")

wm25_archr <- patient_recluster(wm25_archr, "WM25")
wm43_archr <- patient_recluster(wm43_archr, "WM43")
wm46_archr <- patient_recluster(wm46_archr, "WM46")
wm47_archr <- patient_recluster(wm47_archr, "WM47")
wm54_archr <- patient_recluster(wm54_archr, "WM54")
wm65_archr <- patient_recluster(wm65_archr, "WM65")
```
Plot function
```{r}
patientRecluster_plots <- function(archR_obj, sample_str) {
  print(paste0("Making plots for sample: ", sample_str))
  p1 <- plotEmbedding(ArchRProj = archR_obj, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")
  p2 <- plotEmbedding(ArchRProj = archR_obj, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
  ggsave(p1, filename = paste0("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/", sample_str, "_PatientReclustered_CellTypes_UMAP.png"), height = 6, width = 10)
  ggsave(p2, filename = paste0("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/", sample_str, "_PatientReclustered_PredictedGroupCo_UMAP.png"), height = 6, width = 10)
}
```
# Plot it
```{r}
patientRecluster_plots(p01_076_archr, "p01_076")
patientRecluster_plots(p01_115_archr, "p01_115")
patientRecluster_plots(p01_131_archr, "p01_131")
patientRecluster_plots(p01_163_archr, "p01_163")
patientRecluster_plots(p01_190_archr, "p01_190")
patientRecluster_plots(p04_003_archr, "p04_003")
patientRecluster_plots(p04_006_archr, "p04_006")
patientRecluster_plots(wm25_archr, "WM25")
patientRecluster_plots(wm43_archr, "WM43")
patientRecluster_plots(wm46_archr, "WM46")
patientRecluster_plots(wm47_archr, "WM47")
patientRecluster_plots(wm54_archr, "WM54")
patientRecluster_plots(wm65_archr, "WM65")

```
# patient 01-163 explore
```{r}
geneList <- c("BCL6", "PRDM1", "XBP1", "JCHAIN", "POU2F1", "ID3", "ID4", "IRF8", "SNAI2", "MESP1", "TCF4", "POU2F2",
              "REL", "RELA", "NFKB1", "NFKB2",
              "FOS", "FOSB", "JUN", "JUNB") # PC associated according to Hussein email + xbp1 and jchain added from me

# Extract specific motifs
specificMotifs <- getFeatures(p01_163_archr, select = paste(geneList, collapse="|"), useMatrix = "MotifMatrix")
specificMotifs <- grep("z:", specificMotifs, value = TRUE)

# Overlay onto UMAPs
p <- plotEmbedding(
    ArchRProj = p01_163_archr, 
    colorBy = "MotifMatrix", 
    name = sort(specificMotifs), 
    embedding = "Patient_Recluster_UMAP",
    imputeWeights = getImputeWeights(p01_163_archr)
)

p$`z:BCL6_187`
p$`z:PRDM1_163`
p$`z:XBP1_109`
p$`z:ID3_38`
p$`z:ID4_75`
p$`z:POU2F2_609`
p$`z:IRF8_633`
p$`z:SNAI2_161`
p$`z:MESP1_69`
p$`z:TCF4_97`
p$`z:POU2F1_614`
p$`z:REL_721`
p$`z:RELA_722`
p$`z:NFKB1_719`
p$`z:NFKB2_714`
p$`z:FOS_137`
p$`z:FOSB_121`
p$`z:JUN_143`
p$`z:JUNB_139`

###


p2 <- plotEmbedding (
    ArchRProj = p01_131_archr, 
    colorBy = "MotifMatrix", 
    name = sort(specificMotifs), 
    embedding = "Patient_Recluster_UMAP",
    imputeWeights = getImputeWeights(p01_163_archr)
)

p2$`z:BCL6_187`
p2$`z:PRDM1_163`
p2$`z:XBP1_109`
p2$`z:ID3_38`
p2$`z:ID4_75`
p2$`z:POU2F2_609`
p2$`z:IRF8_633`
p2$`z:SNAI2_161`
p2$`z:MESP1_69`
p2$`z:TCF4_97`
p2$`z:POU2F1_614`
p2$`z:REL_721`
p2$`z:RELA_722`
p2$`z:NFKB1_719`
p2$`z:NFKB2_714`
p2$`z:FOS_137`
p2$`z:FOSB_121`
p2$`z:JUN_143`
p2$`z:JUNB_139`

plotEmbedding(ArchRProj = p01_076_archr, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = p01_131_archr, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = p01_163_archr, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = p04_006_archr, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = p01_190_archr, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")

geneList2 <- c("PAX5")

wm54_archr

plotEmbedding (
    ArchRProj = wm54_archr, 
    colorBy = "GeneIntegrationMatrix", 
    name = geneList2, 
    embedding = "Patient_Recluster_UMAP",
    imputeWeights = getImputeWeights(wm54_archr)
)

### 
geneList <- c("BCL6", "PRDM1", "XBP1", "JCHAIN", "POU2F1", "ID3", "ID4", "IRF8", "SNAI2", "MESP1", "TCF4", "POU2F2",
              "REL", "RELA", "NFKB1", "NFKB2",
              "FOS", "FOSB", "JUN", "JUNB",
              "BCL11A", "BCL11B", "SPI1", "SPIB",
              "IRF4", "IRF8") 
specificMotifs <- getFeatures(p04_006_archr, select = paste(geneList, collapse="|"), useMatrix = "MotifMatrix")
specificMotifs <- grep("z:", specificMotifs, value = TRUE)

p3 <- plotEmbedding (
    ArchRProj = p04_006_archr, 
    colorBy = "MotifMatrix", 
    name = sort(specificMotifs), 
    embedding = "Patient_Recluster_UMAP",
    imputeWeights = getImputeWeights(p04_006_archr)
)

p3$`z:BCL6_187`
p3$`z:PRDM1_163`
p3$`z:XBP1_109`
p3$`z:ID3_38`
p3$`z:ID4_75`
p3$`z:POU2F2_609`
p3$`z:IRF8_633`
p3$`z:SNAI2_161`
p3$`z:MESP1_69`
p3$`z:TCF4_97`
p3$`z:POU2F1_614`
p3$`z:REL_721`
p3$`z:RELA_722`
p3$`z:NFKB1_719`
p3$`z:NFKB2_714`
p3$`z:FOS_137`
p3$`z:FOSB_121`
p3$`z:JUN_143`
p3$`z:JUNB_139`
p3$`z:BCL11A_194`
p3$`z:BCL11B_825`
p3$`z:SPI1_322`
p3$`z:SPIB_336`

#### 01-190
p4 <- plotEmbedding (
    ArchRProj = p01_190_archr, 
    colorBy = "MotifMatrix", 
    name = sort(specificMotifs), 
    embedding = "Patient_Recluster_UMAP",
    imputeWeights = getImputeWeights(p01_190_archr)
)

p4$`z:BCL6_187`
p4$`z:PRDM1_163`
p4$`z:XBP1_109`
p4$`z:ID3_38`
p4$`z:ID4_75`
p4$`z:POU2F2_609`
p4$`z:IRF8_633`
p4$`z:SNAI2_161`
p4$`z:MESP1_69`
p4$`z:TCF4_97`
p4$`z:POU2F1_614`
p4$`z:REL_721`
p4$`z:RELA_722`
p4$`z:NFKB1_719`
p4$`z:NFKB2_714`
p4$`z:FOS_137`
p4$`z:FOSB_121`
p4$`z:JUN_143`
p4$`z:JUNB_139`
p4$`z:BCL11A_194`
p4$`z:BCL11B_825`
p4$`z:SPI1_322`
p4$`z:SPIB_336`
p4$`z:IRF4_632`
p4$`z:IRF8_633`


geneList2 <- c("CD38", "SDC1", "CD27", "IRF4", "PRDM1", "XBP1", "JCHAIN")


plotEmbedding (
    ArchRProj = p04_003_archr, 
    colorBy = "GeneIntegrationMatrix", 
    name = geneList2, 
    embedding = "Patient_Recluster_UMAP",
    imputeWeights = getImputeWeights(p04_003_archr)
)

plotEmbedding(ArchRProj = p01_190_archr, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = p01_163_archr, colorBy = "cellColData", name = "celltypes", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = wm25_archr, colorBy = "cellColData", name = "Clusters", embedding = "Patient_Recluster_UMAP")

# Fixing WM54 annotatons
wm54_archr$fixed_celltypes <- NA
wm54_archr$fixed_celltypes[wm54_archr$Clusters == "C1"] <- "Malignant MBC"
wm54_archr$fixed_celltypes[wm54_archr$Clusters == "C2"] <- "Malignant MBC"
wm54_archr$fixed_celltypes[wm54_archr$Clusters == "C3"] <- "Naive B"
wm54_archr$fixed_celltypes[wm54_archr$Clusters == "C4"] <- "Normal MBC"
wm54_archr$fixed_celltypes[wm54_archr$Clusters == "C5"] <- "Normal MBC"
wm54_archr$fixed_celltypes[wm54_archr$Clusters == "C6"] <- "Malignant MBC"
table(wm54_archr$fixed_celltypes)

# Fixing 04-003 annotatons
p04_003_archr$fixed_celltypes <- NA
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C1"] <- "Normal MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C2"] <- "Plasma cells"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C3"] <- "Normal MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C4"] <- "Malignant MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C5"] <- "Normal MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C6"] <- "Malignant MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C7"] <- "Malignant MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C8"] <- "Malignant MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C9"] <- "Malignant MBC"
p04_003_archr$fixed_celltypes[p04_003_archr$Clusters == "C10"] <- "Malignant MBC"

# Fixing WM25 annotatons
wm25_archr$fixed_celltypes <- NA
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C1"] <- "Normal MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C2"] <- "Normal MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C3"] <- "Normal MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C4"] <- "Normal MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C5"] <- "Normal MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C6"] <- "Plasma cells"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C7"] <- "Plasma cells"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C8"] <- "Malignant MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C9"] <- "Malignant MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C10"] <- "Malignant MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C11"] <- "Malignant MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C12"] <- "Malignant MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C13"] <- "Malignant MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C14"] <- "Malignant MBC"
wm25_archr$fixed_celltypes[wm25_archr$Clusters == "C15"] <- "Malignant MBC"

plotEmbedding(ArchRProj = wm25_archr, colorBy = "cellColData", name = "fixed_celltypes", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = wm25_archr, colorBy = "cellColData", name = "hasTopCDR3", embedding = "Patient_Recluster_UMAP")
plotEmbedding(ArchRProj = p04_003_archr, colorBy = "cellColData", name = "Clusters", embedding = "Patient_Recluster_UMAP")

```
### "NORMAL" vs MALIGNANT MBC CASE STUDY - 01-076 and 01-115 and WM54 and 04-003 ###
```{r}
# Get differential genes
markersPeaks <- getMarkerFeatures(
    ArchRProj = wm54_archr, 
    useMatrix = "GeneIntegrationMatrix", 
    groupBy = "fixed_celltypes",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

fmarkersGEX_list <- getMarkers(markersPeaks, cutOff = "FDR < 0.1 & Log2FC > 0.5")

# Up Normal
hasPC.df.gex <- markersGEX_list$`Normal MBC`
hasPC.df.gex <- as.data.frame(hasPC.df.gex)
head(hasPC.df.gex[order(-hasPC.df.gex$MeanDiff),], n = 20)

# Up Naive
noPC.df.gex <- markersGEX_list$`Naive B`
noPC.df.gex <- as.data.frame(noPC.df.gex)
head(noPC.df.gex[order(-noPC.df.gex$MeanDiff),], n = 20)

# Plot some genes
geneList1 <- c("BACH2", "AFF3", "ARHGAP24", "CD74", "PLCG2", "ZCCHC7", "CDK14", "TCF4", "HLA-DRA", "BCL2", "BCL6", "PAX5", "PRDM1", "XBP1", "JCHAIN", "SDC1", "MZB1")
geneList2 <- c()
plotEmbedding(wm54_archr, colorBy = "GeneIntegrationMatrix", name = geneList1, embedding = "Patient_Recluster_UMAP", imputeWeights = getImputeWeights(wm54_archr))
```

```{r}
##### PEAKS
markersPeaksReally <- getMarkerFeatures(
    ArchRProj = wm54_archr, 
    useMatrix = "PeakMatrix", 
    groupBy = "fixed_celltypes",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon",
  useGroups = "Naive B",
  bgdGroups = "Normal MBC"
)

###
motifsUp <- peakAnnoEnrichment(
    seMarker = markersPeaksReally,
    ArchRProj = wm54_archr,
    peakAnnotation = "Motif",
    cutOff = "FDR < 0.1 & Log2FC > -1"
  )

df.up <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df.up <- df.up[order(df.up$mlog10Padj, decreasing = TRUE),]
df.up$rank <- seq_len(nrow(df.up))

ggUp <- ggplot(df.up, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df.up[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

plotPDF(ggUp, name = "WM54-Nonclonal-vs-Clonal-MBC_Motifs_RankSorted_UpNormal", width = 5, height = 5, ArchRProj = wm54_archr, addDOC = FALSE)

df.up
```
Motif Devs
```{r}
vardev <- getVarDeviations(wm54_archr, name = "MotifMatrix", plot = FALSE)
vardev.df <- as.data.frame(vardev)
head(vardev.df, n = 70)

vardev.list <- c("RELA", "EGR2", "KLF16", "SP3", "NFKB2", "NFKB1", "EGR1",
                 "IRF4", "PRDM1", "XBP1")
#vardev.list2 <- vardev.df$name[1:50] # just get top 50 of em

# Extract specific motifs
specificMotifs <- getFeatures(wm54_archr, select = paste(vardev.list, collapse="|"), useMatrix = "MotifMatrix")
specificMotifs <- grep("z:", specificMotifs, value = TRUE)

# Overlay onto UMAPs
p <- plotEmbedding(
    ArchRProj = wm54_archr, 
    colorBy = "MotifMatrix", 
    name = sort(specificMotifs), 
    embedding = "Patient_Recluster_UMAP",
    imputeWeights = getImputeWeights(wm54_archr)
)

p$`z:RELA_722`
p$`z:EGR1_195`
p$`z:EGR2_196`
p$`z:KLF16_205`
p$`z:SP3_247`
p$`z:NFKB1_719`
p$`z:NFKB2_714`
p$`z:IRF4_632`
p$`z:PRDM1_163`
p$`z:XBP1_109`
```
# MAKE GOOD LOOKING PLOT FOR BROAD COMPARISONS ??
```{r}
motifs <- c("GATA1", "CEBPA", "EBF1", "IRF4", "TBX21", "PAX5")
markerMotifs <- getFeatures(projHeme5, select = paste(motifs, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs <- grep("z:", markerMotifs, value = TRUE)
markerMotifs <- markerMotifs[markerMotifs %ni% "z:SREBF1_22"]
markerMotifs

p <- plotGroups(ArchRProj = projHeme5, 
  groupBy = "Clusters2", 
  colorBy = "MotifMatrix", 
  name = markerMotifs,
  imputeWeights = getImputeWeights(projHeme5)
)

p2 <- lapply(seq_along(p), function(x){
  if(x != 1){
    p[[x]] + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6) +
    theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
    theme(
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        axis.title.y=element_blank()
    ) + ylab("")
  }else{
    p[[x]] + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6) +
    theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
    theme(
        axis.ticks.y=element_blank(),
        axis.title.y=element_blank()
    ) + ylab("")
  }
})
do.call(cowplot::plot_grid, c(list(nrow = 1, rel_widths = c(2, rep(1, length(p2) - 1))),p2))
```

