---
title: "ArchR_PeakAnalysis"
output: html_document
date: "2023-11-10"
---

Load library
```{r}
library(Seurat)
library(ggplot2)
library(ArchR)
library(BSgenome.Hsapiens.UCSC.hg38)
addArchRThreads(threads = 1)
addArchRGenome("hg38")
options(future.globals.maxSize=10000000000000000) 
```
Load in
```{r}
archR <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_Annotated_TRUST4_NoControls_PeaksAdded")
```
Try subsetting out only PCs and MBCs
```{r}
idsRemove <- as.character(archR$cellNames[archR$celltypes %in% c("Large pre-B cells", "Small pre-B cells", "Pro-B cells", "Naive B cells")])
archR_PC_MBC_only <- archR[!archR$cellNames %in% idsRemove,]

# Create column for normals vs malignants
archR_PC_MBC_only$N_vs_M <- archR_PC_MBC_only$cluster_names
archR_PC_MBC_only$N_vs_M[archR_PC_MBC_only$cluster_names == "Normal MBC"] <- "Normal"
archR_PC_MBC_only$N_vs_M[archR_PC_MBC_only$cluster_names == "Malignant MBC"] <- "Malignant"
archR_PC_MBC_only$N_vs_M[archR_PC_MBC_only$cluster_names == "Plasma cells"] <- "Malignant"
table(archR_PC_MBC_only$N_vs_M)

archR$clusters <- archR$Harmony_Batch_Clusters_NoControls

# Fixing annotations
archR$celltypes <- archR$predictedGroup_Co
archR$celltypes[archR$predictedGroup_Co == "Memory B cells"] <- "Malignant MBC"
archR$celltypes[archR$Harmony_Batch_Clusters_NoControls == "C11"] <- "Malignant MBC"
archR$celltypes[archR$Harmony_Batch_Clusters_NoControls == "C4"] <- "Normal MBC"
archR$celltypes[archR$Harmony_Batch_Clusters_NoControls == "C5"] <- "Normal MBC"
archR$celltypes[archR$Harmony_Batch_Clusters_NoControls == "C6"] <- "Normal MBC"
archR$celltypes[archR$Harmony_Batch_Clusters_NoControls == "C7"] <- "Normal MBC"

#archR_PC_MBC_only$celltypes[archR_PC_MBC_only$NoProgen_Recluster == "C7"] <- "Normal MBC"

table(archR$celltypes)
plotEmbedding(ArchRProj = archR_PC_MBC_only, colorBy = "cellColData", name = "celltypes", embedding = "UMAP_Harmony_Batch_NoControls")


p1 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "celltypes", embedding = "UMAP_Harmony_Batch_NoControls")
p2 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesClean", embedding = "UMAP_Harmony_Batch_NoControls")
p3 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "clusters", embedding = "UMAP_Harmony_Batch_NoControls")
plotPDF(p1,p2,p3, name = "WM_FullSet_NewAnnotation_MainUMAPs.pdf", ArchRProj = archR, addDOC = FALSE, width = 5, height = 5)
```
Recluster
```{r}
archR_PC_MBC_only <- addIterativeLSI(
  ArchRProj = archR_PC_MBC_only,
  useMatrix = "TileMatrix",
  name = "IterativeLSI_NoProgen",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

archR_PC_MBC_only <- addClusters(
  input = archR_PC_MBC_only,
  reducedDims = "IterativeLSI_NoProgen",
  method = "Seurat",
  name = "NoProgen_Recluster",
  resolution = 0.8,
  force = TRUE
)

archR_PC_MBC_only <- addUMAP(
    ArchRProj = archR_PC_MBC_only, 
    reducedDims = "IterativeLSI_NoProgen", 
    name = "NoProgen_Recluster_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = archR_PC_MBC_only, colorBy = "cellColData", name = "samplesClean", embedding = "NoProgen_Recluster_UMAP")
p2 <- plotEmbedding(ArchRProj = archR_PC_MBC_only, colorBy = "cellColData", name = "celltypes", embedding = "NoProgen_Recluster_UMAP")
p3 <- plotEmbedding(ArchRProj = archR_PC_MBC_only, colorBy = "cellColData", name = "NoProgen_Recluster", embedding = "NoProgen_Recluster_UMAP")

plotPDF(p1,p2,p3, name = "WM_NoControl_NoProgen_Reclustered_UMAPs.pdf", ArchRProj = archR, addDOC = FALSE, width = 5, height = 5)
```
# Save just MBC/PCs
```{r}
saveArchRProject(ArchRProj = archR_PC_MBC_only, outputDirectory = "WM_NoControls_NoProgens_DownstreamObject", load = F)
```

Just check in those weird clusters...
```{r}
c7_test_gex <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "clusters",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)")
)

cluster_list_gex <- getMarkers(c7_test_gex, cutOff = "FDR <= 0.1 & Log2FC >= 0.5")
c7_list_gex <- as.data.frame(cluster_list_gex$C7)
head(c7_list_gex[order(-c7_list_gex$MeanDiff),], n = 20)

c15_list_gex <- as.data.frame(cluster_list_gex$C15)
head(c15_list_gex[order(-c15_list_gex$MeanDiff),], n = 20)

c16_list_gex <- as.data.frame(cluster_list_gex$C16)
head(c16_list_gex[order(-c16_list_gex$MeanDiff),], n = 20)

# Gene score
c7_test_gs <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneScoreMatrix",
  groupBy = "clusters",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)")
)

c7_list_gs <- getMarkers(c7_test_gs, cutOff = "FDR <= 0.1 & Log2FC >= 0.5")
c7_list_gs <- as.data.frame(c7_list_gs$C7)
head(c7_list_gs[order(-c7_list_gs$MeanDiff),], n = 20)
```

Refind markers
```{r}
markersPeaks_PC_MBC_only <- getMarkerFeatures(
    ArchRProj = archR_PC_MBC_only, 
    useMatrix = "PeakMatrix", 
    groupBy = "N_vs_M",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

# Visualize via heatmap
heatmapPeaks_PC_MBC_only <- markerHeatmap(
  seMarker = markersPeaks_PC_MBC_only, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.5",
  transpose = TRUE,
  plotLog2FC = TRUE
)

draw(heatmapPeaks_PC_MBC_only, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```
Browser tracks on MYD88 and a few others?
```{r}
p <- plotBrowserTrack(
    ArchRProj = archR_PC_MBC_only, 
    groupBy = "cluster_names", 
    geneSymbol = c("MYD88"),
    features =  getMarkers(markersPeaks_PC_MBC_only, cutOff = "FDR <= 0.1 & Log2FC >= 0.5", returnGR = TRUE),
    upstream = 50000,
    downstream = 50000
)

grid::grid.newpage()
grid::grid.draw(p$MYD88)
```
MBC comparison
```{r}
idsRemove <- as.character(archR$cellNames[archR$cluster_names %in% c("Naive B", "Pre-B", "Pro-B", "Plasma cells")])
archR_MBC <- archR[!archR$cellNames %in% idsRemove,]

# Test marker peaks for heatmap
marker_GEX <- getMarkerFeatures(
    ArchRProj = archR_MBC, 
    useMatrix = "GeneIntegrationMatrix", 
    groupBy = "cluster_names",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

# Get top markers in both conditions
markerList_GEX <- getMarkers(marker_GEX, cutOff = "FDR <= 0.01 & Log2FC >= 1")

malig.df.gex <- markerList_GEX$`Malignant MBC`
malig.df.gex <- as.data.frame(malig.df.gex)
head(malig.df.gex[order(-malig.df.gex$Log2FC),], n = 20)

normal.df.gex <- markerList_GEX$`Normal MBC`
normal.df.gex <- as.data.frame(normal.df.gex)
head(normal.df.gex[order(-normal.df.gex$Log2FC),], n = 20)

pv <- markerPlot(seMarker = marker_GEX, name = "Malignant MBC", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 1", plotAs = "Volcano")
pv

# Top malignant marker genes by GEX (sorted by Log2FC)
markerGenes <- c("NPTX1", "STK32B", "DAPK1", "MSR1", "CXXC4-AS1", "SGCD", "LINC01239", "MYEOV", "PRDM5", "EFNA5", "MERTK", "FBP1", "PIK3R3", "CLLU1", "PIEZO2", "CDH9", "GABARAPL1")

p <- plotEmbedding(
    ArchRProj = archR_MBC,
    colorBy = "GeneIntegrationMatrix",
    name = markerGenes,
    embedding = "UMAP_Harmony_Batch_NoControls",
    imputeWeights = getImputeWeights(archR_MBC)
)

p$NPTX1
p$STK32B
p$DAPK1
p$MSR1
p$`CXXC4-AS1`
p$SGCD
p$MYEOV
p$PRDM5
p$EFNA5
p$MERTK
p$PIK3R3
p$CLLU1
p$PIEZO2

# Now sorting by Mean Diff
head(malig.df.gex[order(-malig.df.gex$MeanDiff),], n = 20)

markerGenes2 <- c("ZNF804A", "PELI1", "MSI2", "PTPRG", "ETV6", "ZEB1", "EIF2AK3", "SYNE2", "CARMIL1", "JCHAIN", "PMAIP1", "EML6", "MAN1A1", "TLE4", "MBNL2", "IFNG-AS1", "SNX9", "CPEB4", "RASSF6", "LRCH1")

p2 <- plotEmbedding(
    ArchRProj = archR_MBC,
    colorBy = "GeneIntegrationMatrix",
    name = markerGenes2,
    embedding = "UMAP_Harmony_Batch_NoControls",
    imputeWeights = getImputeWeights(archR_MBC)
)

p2$ZNF804A
p2$PELI1
p2$MSI2
p2$PTPRG
p2$`IFNG-AS1`
p2$ETV6
p2$ZEB1
p2$EIF2AK3
p2$SYNE2
p2$CARMIL1
p2$JCHAIN
p2$PMAIP1
p2$EML6
p2$MAN1A1
p2$TLE4
p2$MBNL2
p2$SNX9
p2$CPEB4
p2$RASSF6
p2$LRCH1

# Now what's up in normals???
head(normal.df.gex[order(-normal.df.gex$MeanDiff),], n = 20)

markerGenes3 <- c("PDE4D", "ANK3", "HIF1A", "ITPR1", "UST", "PAG1", "AUTS2", "SSPN", "TOX", "CRIP1", "KCNQ5", "PIK3R5", "SSBP2", "SH3BGRL3", "APPL1", "PTK2", "CR1")

p3 <- plotEmbedding(
    ArchRProj = archR_MBC,
    colorBy = "GeneIntegrationMatrix",
    name = markerGenes3,
    embedding = "UMAP_Harmony_Batch_NoControls",
    imputeWeights = getImputeWeights(archR_MBC)
)

p3$PDE4D
p3$ANK3
p3$HIF1A
p3$ITPR1
p3$UST
p3$PAG1
p3$AUTS2
p3$SSPN
p3$TOX
p3$CRIP1
p3$KCNQ5
p3$PIK3R5
p3$SSBP2
p3$SH3BGRL3
p3$APPL1
p3$PTK2
p3$CR1

```
Now gene scores for normal vs malignant MBC
```{r}
# Test marker peaks for heatmap
marker_GS <- getMarkerFeatures(
    ArchRProj = archR_MBC, 
    useMatrix = "GeneScoreMatrix", 
    groupBy = "cluster_names",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

# Get top markers in both conditions
markerList_GS <- getMarkers(marker_GS, cutOff = "FDR <= 0.01 & Log2FC >= 1")

malig.df.GS <- markerList_GS$`Malignant MBC`
malig.df.GS <- as.data.frame(malig.df.GS)
head(malig.df.GS[order(-malig.df.GS$MeanDiff),], n = 20)

normal.df.GS <- markerList_GS$`Normal MBC`
normal.df.GS <- as.data.frame(normal.df.GS)
head(normal.df.GS[order(-normal.df.GS$MeanDiff),], n = 20)

pv <- markerPlot(seMarker = marker_GS, name = "Malignant MBC", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 1", plotAs = "Volcano")
pv

# Top malignant marker genes by GEX (sorted by MeanDiff)
markerGenes <- c("MIR4539", "MIR4269", "ZFAT-AS1", "PTPRG", "MIR5571", "MIR3139", "LINC01484", "ZNF804A", "GAB1", "CARMIL1", "GPRIN3", "TAS1R3", "SMIM6", "ELFN2", "MIR596", "MIR5700", "LOC100506271", "MIR548AE1", "KIF26B", "SNORA49")

p <- plotEmbedding(
    ArchRProj = archR_MBC,
    colorBy = "GeneScoreMatrix",
    name = markerGenes,
    embedding = "UMAP_Harmony_Batch_NoControls",
    imputeWeights = getImputeWeights(archR_MBC)
)

p$MIR4539
p$MIR4269
p$`ZFAT-AS1`
p$PTPRG
p$MIR5571
p$MIR3139
p$LINC01484
p$ZNF804A
p$GAB1
p$CARMIL1
p$GPRIN3
p$TAS1R3
p$SMIM6
p$ELFN2
p$MIR596
p$MIR5700
p$LOC100506271
p$MIR548AE1
p$KIF26B
p$SNORA49

# Top GENE SCORE UPREGULATED IN NORMALS
markerGenes <- c("ATXN1", "MIR4666A", "HIST3H2A", "AUTS2", "LOC100507468", "HIST3H2BB", "MIR3681HG", "PAG1", "FRMD4A", "GAS7", "PCBP3", "MIR5093", "MYADM", "IGF1R", "TOX", "MIR3945", "MIR7843", "C17orf82", "FABP5")

p <- plotEmbedding(
    ArchRProj = archR_MBC,
    colorBy = "GeneScoreMatrix",
    name = markerGenes,
    embedding = "UMAP_Harmony_Batch_NoControls",
    imputeWeights = getImputeWeights(archR_MBC)
)

p$ATXN1
p$MIR4666A
p$HIST3H2A
p$AUTS2
p$LOC100507468
p$HIST3H2BB
p$MIR3681HG
p$PAG1
p$FRMD4A
p$GAS7
p$PCBP3
p$MIR5093
p$MYADM
p$IGF1R
p$TOX
p$MIR3945
p$MIR7843
p$C17orf82
p$FABP5

```

Motifs on normal vs. malignant MBC
```{r}
# Test marker peaks for heatmap
markersPeaks <- getMarkerFeatures(
    ArchRProj = archR_MBC, 
    useMatrix = "PeakMatrix", 
    groupBy = "cluster_names",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

# Heatmap
heatmapPeaks <- markerHeatmap(
  seMarker = markersPeaks, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.5",
  transpose = TRUE,
  plotLog2FC = TRUE
)

draw(heatmapPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")

archR_MBC <- addMotifAnnotations(ArchRProj = archR_MBC, motifSet = "cisbp", name = "Motif")

motifsUp <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = archR_MBC,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )

df.up <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df.up <- df.up[order(df.up$mlog10Padj, decreasing = TRUE),]
df.up$rank <- seq_len(nrow(df.up))

ggUp <- ggplot(df.up, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

#########

motifsDo <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = archR_MBC,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.5"
  )

df.down <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df.down <- df.down[order(df.down$mlog10Padj, decreasing = TRUE),]
df.down$rank <- seq_len(nrow(df.down))
head(df.down, n = 50)

ggDo <- ggplot(df.down, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo

plotPDF(ggUp, ggDo, name = "Normal-vs-Malignant_MBC-Markers-Motifs-Enriched", width = 5, height = 5, ArchRProj = archR, addDOC = FALSE)
```

# Now doing analysis with has PCs vs does not have PCs
```{r}
# Fix this metadata
archR$hasPC[archR$samplesClean == "WM25"] <- "-"
archR$hasPC[archR$samplesClean == "WM43"] <- "-"
archR$hasPC[archR$samplesClean == "WM46"] <- "-"
archR$hasPC[archR$samplesClean == "WM47"] <- "-"
archR$hasPC[archR$samplesClean == "WM54"] <- "+"
archR$hasPC[archR$samplesClean == "WM65"] <- "+"

idsRemove <- as.character(archR$cellNames[archR$cluster_names %in% c("Naive B", "Pre-B", "Pro-B", "Normal MBC")])
archR_malignant_only <- archR[!archR$cellNames %in% idsRemove,]

# Test marker peaks for heatmap
markersPeaks <- getMarkerFeatures(
    ArchRProj = archR_malignant_only, 
    useMatrix = "PeakMatrix", 
    groupBy = "hasPC",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

# Heatmap
heatmapPeaks <- markerHeatmap(
  seMarker = markersPeaks, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.5",
  transpose = TRUE,
  plotLog2FC = TRUE
)

draw(heatmapPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")

# Volcano plot
pv <- markerPlot(seMarker = markersPeaks, name = "+", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 1", plotAs = "Volcano")
pv

plotPDF(pv, name = "YesNo-PCs-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = archR, addDOC = FALSE)

```

```{r}
markerList <- getMarkers(markersPeaks, cutOff = "FDR <= 0.1 & Log2FC >= 0.5")
sort(table(markerList$`+`$seqnames), descending = T)
sort(table(markerList$`-`$seqnames), descending = T)
```
Transcription factors/Motifs
```{r}
archR_malignant_only <- addMotifAnnotations(ArchRProj = archR_malignant_only, motifSet = "cisbp", name = "Motif")

motifsUp <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = archR_malignant_only,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )

df.up <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df.up <- df.up[order(df.up$mlog10Padj, decreasing = TRUE),]
df.up$rank <- seq_len(nrow(df.up))

ggUp <- ggplot(df.up, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

#########

motifsDo <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = archR_malignant_only,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.5"
  )

df.down <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df.down <- df.down[order(df.down$mlog10Padj, decreasing = TRUE),]
df.down$rank <- seq_len(nrow(df.down))
head(df.down, n = 50)

ggDo <- ggplot(df.down, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo

plotPDF(ggUp, ggDo, name = "YesNoPCs_MalignantOnly-Markers-Motifs-Enriched", width = 5, height = 5, ArchRProj = archR, addDOC = FALSE)
```
For comparing malignants across patients...
```{r}
markersPeaks <- getMarkerFeatures(
    ArchRProj = archR, 
    useMatrix = "PeakMatrix", 
    groupBy = "celltypes",
    useGroups = "Malignant MBC",
    bgdGroups = "Normal MBC",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

# Heatmap
heatmapPeaks <- markerHeatmap(
  seMarker = markersPeaks, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.5",
  transpose = TRUE,
  plotLog2FC = TRUE
)

draw(heatmapPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")

enrichMotifs <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = archR_malignant_only,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
  )

heatmapEM <- plotEnrichHeatmap(enrichMotifs, n = 7, transpose = TRUE)

ComplexHeatmap::draw(heatmapEM, heatmap_legend_side = "bot", annotation_legend_side = "bot")
```
Gene Integration Data look
```{r}
markersGenes_GEX <- getMarkerFeatures(
    ArchRProj = archR_PC_MBC_only, 
    useMatrix = "GeneIntegrationMatrix", 
    groupBy = "celltypes",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

markerList_GEX <- getMarkers(markersGenes_GEX, cutOff = "FDR <= 0.1 & Log2FC >= 0.5")

malig.df.gex <- markerList_GEX$`Malignant MBC`
malig.df.gex <- as.data.frame(malig.df.gex)
head(malig.df.gex[order(-malig.df.gex$MeanDiff),], n = 20)

normal.df.gex <- markerList_GEX$`Normal MBC`
normal.df.gex <- as.data.frame(normal.df.gex)
head(normal.df.gex[order(-normal.df.gex$MeanDiff),], n = 20)

pv <- markerPlot(seMarker = markersPeaks_GEX, name = "Malignant MBC", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 1", plotAs = "Volcano")
pv

markerGenes <- c("ZNF804A", "PELI1", "MSI2", "PTPRG", "ETV6", "ZEB1", "EIF2AK3", "SYNE2", "CARMIL1", "JCHAIN", "PMAIP1", "EML6", "MAN1A1", "TLE4", "MBNL2", "IFNG-AS1", "SNX9", "CPEB4", "RASSF6", "LRCH1")

p2 <- plotEmbedding(
    ArchRProj = archR_PC_MBC_only,
    colorBy = "GeneIntegrationMatrix",
    name = markerGenes,
    embedding = "NoProgen_Recluster_UMAP",
    imputeWeights = getImputeWeights(archR_PC_MBC_only)
)

p2$ZNF804A
p2$PELI1
p2$MSI2
p2$PTPRG
p2$`IFNG-AS1`
p2$ETV6
p2$ZEB1
p2$EIF2AK3
p2$SYNE2
p2$CARMIL1
p2$JCHAIN
p2$PMAIP1
p2$EML6
p2$MAN1A1
p2$TLE4
p2$MBNL2
p2$SNX9
p2$CPEB4
p2$RASSF6
p2$LRCH1

p3 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 2),p))

p3
plotPDF(plotList = p3, 
    name = "N-vs-M_MBC_Reclustered_NoProgens_UpMalig_Top20_GEX_UMAPs.pdf", 
    ArchRProj = archR, 
    addDOC = FALSE, width = 5, height = 5)
```
Gene Score differential genes
```{r}
# Top malignant marker genes by GEX (sorted by MeanDiff)
markerGenes <- c("MIR4539", "MIR4269", "ZFAT-AS1", "PTPRG", "MIR5571", "MIR3139", "LINC01484", "ZNF804A", "GAB1", "CARMIL1", "GPRIN3", "TAS1R3", "SMIM6", "ELFN2", "MIR596", "MIR5700", "LOC100506271", "MIR548AE1", "KIF26B", "SNORA49")

p <- plotEmbedding(
    ArchRProj = archR_PC_MBC_only,
    colorBy = "GeneScoreMatrix",
    name = markerGenes,
    embedding = "NoProgen_Recluster_UMAP",
    imputeWeights = getImputeWeights(archR_PC_MBC_only)
)

p$MIR4539
p$MIR4269
p$`ZFAT-AS1`
p$PTPRG
p$MIR5571
p$MIR3139
p$LINC01484
p$ZNF804A
p$GAB1
p$CARMIL1
p$GPRIN3
p$TAS1R3
p$SMIM6
p$ELFN2
p$MIR596
p$MIR5700
p$LOC100506271
p$MIR548AE1
p$KIF26B
p$SNORA49

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 2),p))

p2
plotPDF(plotList = p2, 
    name = "N-vs-M_MBC_Reclustered_NoProgens_UpMalig_Top20_GeneScore_UMAPs.pdf", 
    ArchRProj = archR, 
    addDOC = FALSE, width = 5, height = 5)
```

