---
title: "WM_Figures"
output: html_document
date: "2024-03-06"
---
Code used to generate main figures. Bulk RNA-seq heatmap processing included in additional code (see bulkRNA_Analysis.Rmd). 
# Load libraries and object
```{r}
library(ArchR)
library(Seurat)
library(SummarizedExperiment)
library(ggpp)
library(ggrepel)
library(cowplot)
library(venn)
library(VennDiagram)
library(RColorBrewer)
library(svglite)
addArchRThreads(threads = 1)
addArchRGenome("hg38")
options(future.globals.maxSize=1000000000000000000000000000) 
archR <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_DeviationsAdded_BatchCorrected")
```
# Patient UMAP Figure 1E
```{r}
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
patient_colors <- gg_color_hue(length(table(archR$samplesRefCombined)))
names(patient_colors) <- names(table(archR$samplesRefCombined))

plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesRefCombined", embedding = "UMAP_Harmony", pal = patient_colors, labelMeans = F, labelSize = 4) + ggtitle("") + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure1/Figure1E_MainPatientUMAP.pdf", height = 6, width = 6)

plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesRefCombined", embedding = "UMAP_Harmony", pal = patient_colors, labelAsFactors = F, labelSize = 4) + ggtitle("") + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure1/Figure1E_MainPatientUMAP_Labels.pdf", height = 6, width = 6)

plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesRefCombined", embedding = "UMAP_Harmony", pal = patient_colors, 
              labelMeans = F,
              baseSize = 10,
              labelSize = 5.5,
              fgColor = "black",
              bgWidth = 0) + NoLegend() + theme(title = element_blank(),
                                                axis.title.x = element_text(size = 15),
                                                axis.title.y = element_text(size = 15))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure1/Figure1E_MainPatientUMAP.pdf", height = 6, width = 6)
```
# Celltype barplot for Figure 1 -
# Dirty way to get celltype relative abundance by patient dataframe for stacked barplots
```{r}
#new column for having new name for WM43
archR$barplot_annotation <- archR$final_annotation
archR$barplot_annotation[archR$barplot_annotation == "WM43"] <- "Lymphoplasmacyte"

#new column for having new name for WM43
archR$barplot_samples <- archR$samplesClean
archR$barplot_samples[archR$samplesClean == "BCP003"] <- "Ref1"
archR$barplot_samples[archR$samplesClean == "BCP004"] <- "Ref2"
archR$barplot_samples[archR$samplesClean == "BCP005"] <- "Ref3"
archR$barplot_samples[archR$samplesClean == "BCP006"] <- "Ref4"
archR$barplot_samples[archR$samplesClean == "stanford_1"] <- "Ref5"
archR$barplot_samples[archR$samplesClean == "stanford_2"] <- "Ref6"
archR$barplot_samples[archR$samplesClean == "stanford_3"] <- "Ref7"


############### NYU SAMPLES
p01.076.ids <- BiocGenerics::which(archR$barplot_samples %in% "01-076")
p01.076.celltypes <- archR$barplot_annotation[p01.076.ids]
p01.115.ids <- BiocGenerics::which(archR$barplot_samples %in% "01-115")
p01.115.celltypes <- archR$barplot_annotation[p01.115.ids]
p01.131.ids <- BiocGenerics::which(archR$barplot_samples %in% "01-131")
p01.131.celltypes <- archR$barplot_annotation[p01.131.ids]
p01.163.ids <- BiocGenerics::which(archR$barplot_samples %in% "01-163")
p01.163.celltypes <- archR$barplot_annotation[p01.163.ids]
p01.190.ids <- BiocGenerics::which(archR$barplot_samples %in% "01-190")
p01.190.celltypes <- archR$barplot_annotation[p01.190.ids]
p04.003.ids <- BiocGenerics::which(archR$barplot_samples %in% "04-003")
p04.003.celltypes <- archR$barplot_annotation[p04.003.ids]
p04.006.ids <- BiocGenerics::which(archR$barplot_samples %in% "04-006")
p04.006.celltypes <- archR$barplot_annotation[p04.006.ids]

############### GUSTAVE ROUSSY SAMPLES

wm25.ids <- BiocGenerics::which(archR$barplot_samples %in% "WM25")
wm25.celltypes <- archR$barplot_annotation[wm25.ids]
wm43.ids <- BiocGenerics::which(archR$barplot_samples %in% "WM43")
wm43.celltypes <- archR$barplot_annotation[wm43.ids]
wm46.ids <- BiocGenerics::which(archR$barplot_samples %in% "WM46")
wm46.celltypes <- archR$barplot_annotation[wm46.ids]
wm47.ids <- BiocGenerics::which(archR$barplot_samples %in% "WM47")
wm47.celltypes <- archR$barplot_annotation[wm47.ids]
wm54.ids <- BiocGenerics::which(archR$barplot_samples %in% "WM54")
wm54.celltypes <- archR$barplot_annotation[wm54.ids]
wm65.ids <- BiocGenerics::which(archR$barplot_samples %in% "WM65")
wm65.celltypes <- archR$barplot_annotation[wm65.ids]
###############

bcp003.ids <- BiocGenerics::which(archR$barplot_samples %in% "Ref1")
bcp003.celltypes <- archR$barplot_annotation[bcp003.ids]
bcp004.ids <- BiocGenerics::which(archR$barplot_samples %in% "Ref2")
bcp004.celltypes <- archR$barplot_annotation[bcp004.ids]
bcp005.ids <- BiocGenerics::which(archR$barplot_samples %in% "Ref3")
bcp005.celltypes <- archR$barplot_annotation[bcp005.ids]
bcp006.ids <- BiocGenerics::which(archR$barplot_samples %in% "Ref4")
bcp006.celltypes <- archR$barplot_annotation[bcp006.ids]

###############
stanford1.ids <- BiocGenerics::which(archR$barplot_samples %in% "Ref5")
stanford1.celltypes <- archR$barplot_annotation[stanford1.ids]
stanford2.ids <- BiocGenerics::which(archR$barplot_samples %in% "Ref6")
stanford2.celltypes <- archR$barplot_annotation[stanford2.ids]
stanford3.ids <- BiocGenerics::which(archR$barplot_samples %in% "Ref7")
stanford3.celltypes <- archR$barplot_annotation[stanford3.ids]

library(tidyverse)

# Relative abundance by patient...requires a lot of fennegling bcuz different patients have different cell populations
p01.076_relabun <- data.frame(table(p01.076.celltypes)/sum(table(p01.076.celltypes)) * 100)
p01.076_relabun$Patient <- "01-076"
colnames(p01.076_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, p01.076_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "01-076"
p01.076.mergedf <- rbind(p01.076_relabun, missing.df)

p01.115_relabun <- data.frame(table(p01.115.celltypes)/sum(table(p01.115.celltypes)) * 100)
p01.115_relabun$Patient <- "01-115"
colnames(p01.115_relabun)[1] <- "Celltype"
# missing.celltypes <- setdiff(archR$barplot_annotation, p01.115_relabun$Celltype)
# missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
# colnames(missing.df) <- c("Celltype", "Freq", "Patient")
# missing.df$Celltype <- missing.celltypes
# missing.df$Patient <- "01-115"
# p01.115.mergedf <- rbind(p01.115_relabun, missing.df)
p01.115.mergedf <- p01.115_relabun # because it has all celltypes!

p01.131_relabun <- data.frame(table(p01.131.celltypes)/sum(table(p01.131.celltypes)) * 100)
p01.131_relabun$Patient <- "01-131"
colnames(p01.131_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, p01.131_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "01-131"
p01.131.mergedf <- rbind(p01.131_relabun, missing.df)

p01.163_relabun <- data.frame(table(p01.163.celltypes)/sum(table(p01.163.celltypes)) * 100)
p01.163_relabun$Patient <- "01-163"
colnames(p01.163_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, p01.163_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "01-163"
p01.163.mergedf <- rbind(p01.163_relabun, missing.df)

p01.190_relabun <- data.frame(table(p01.190.celltypes)/sum(table(p01.190.celltypes)) * 100)
p01.190_relabun$Patient <- "01-190"
colnames(p01.190_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, p01.190_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "01-190"
p01.190.mergedf <- rbind(p01.190_relabun, missing.df)

p04.003_relabun <- data.frame(table(p04.003.celltypes)/sum(table(p04.003.celltypes)) * 100)
p04.003_relabun$Patient <- "04-003"
colnames(p04.003_relabun)[1] <- "Celltype"
# missing.celltypes <- setdiff(archR$barplot_annotation, p04.003_relabun$Celltype)
# missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
# colnames(missing.df) <- c("Celltype", "Freq", "Patient")
# missing.df$Celltype <- missing.celltypes
# missing.df$Patient <- "04-003"
# p04.003.mergedf <- rbind(p04.003_relabun, missing.df)
p04.003.mergedf <- p04.003_relabun # because it has all celltypes!

p04.006_relabun <- data.frame(table(p04.006.celltypes)/sum(table(p04.006.celltypes)) * 100)
p04.006_relabun$Patient <- "04-006"
colnames(p04.006_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, p04.006_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "04-006"
p04.006.mergedf <- rbind(p04.006_relabun, missing.df)

###################################### GOUSTAV ROUSSY SAMPLES

wm25_relabun <- data.frame(table(wm25.celltypes)/sum(table(wm25.celltypes)) * 100)
wm25_relabun$Patient <- "WM25"
colnames(wm25_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, wm25_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "WM25"
wm25.mergedf <- rbind(wm25_relabun, missing.df)

wm43_relabun <- data.frame(table(wm43.celltypes)/sum(table(wm43.celltypes)) * 100)
wm43_relabun$Patient <- "WM43"
colnames(wm43_relabun)[1] <- "Celltype"
# missing.celltypes <- setdiff(archR$barplot_annotation, wm43_relabun$Celltype)
# missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
# colnames(missing.df) <- c("Celltype", "Freq", "Patient")
# missing.df$Celltype <- missing.celltypes
# missing.df$Patient <- "Wm43"
# wm43.mergedf <- rbind(wm43_relabun, missing.df)
wm43.mergedf <- wm43_relabun # because it has all celltypes

wm46_relabun <- data.frame(table(wm46.celltypes)/sum(table(wm46.celltypes)) * 100)
wm46_relabun$Patient <- "WM46"
colnames(wm46_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, wm46_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "WM46"
wm46.mergedf <- rbind(wm46_relabun, missing.df)

wm47_relabun <- data.frame(table(wm47.celltypes)/sum(table(wm47.celltypes)) * 100)
wm47_relabun$Patient <- "WM47"
colnames(wm47_relabun)[1] <- "Celltype"
# missing.celltypes <- setdiff(archR$barplot_annotation, wm47_relabun$Celltype)
# missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
# colnames(missing.df) <- c("Celltype", "Freq", "Patient")
# missing.df$Celltype <- missing.celltypes
# missing.df$Patient <- "WM47"
# wm47.mergedf <- rbind(wm47_relabun, missing.df)
wm47.mergedf <- wm47_relabun # because it has all celltypes

wm54_relabun <- data.frame(table(wm54.celltypes)/sum(table(wm54.celltypes)) * 100)
wm54_relabun$Patient <- "WM54"
colnames(wm54_relabun)[1] <- "Celltype"
# missing.celltypes <- setdiff(archR$barplot_annotation, wm54_relabun$Celltype)
# missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
# colnames(missing.df) <- c("Celltype", "Freq", "Patient")
# missing.df$Celltype <- missing.celltypes
# missing.df$Patient <- "WM54"
# wm54.mergedf <- rbind(wm54_relabun, missing.df)
wm54.mergedf <- wm54_relabun # because it has all celltypes

wm65_relabun <- data.frame(table(wm65.celltypes)/sum(table(wm65.celltypes)) * 100)
wm65_relabun$Patient <- "WM65"
colnames(wm65_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, wm65_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
missing.df$Patient <- "WM65"
wm65.mergedf <- rbind(wm65_relabun, missing.df)

###################################### REFERENCE DATA

bcp003_relabun <- data.frame(table(bcp003.celltypes)/sum(table(bcp003.celltypes)) * 100)
bcp003_relabun$Patient <- "Ref1"
colnames(bcp003_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, bcp003_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
#missing.df$Patient <- "BCP003"
bcp003.mergedf <- rbind(bcp003_relabun, missing.df)
bcp003.mergedf <- bcp003_relabun # bcuz there are no missing types

bcp004_relabun <- data.frame(table(bcp004.celltypes)/sum(table(bcp004.celltypes)) * 100)
bcp004_relabun$Patient <- "Ref2"
colnames(bcp004_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, bcp004_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
#missing.df$Patient <- "BCP004"
bcp004.mergedf <- rbind(bcp004_relabun, missing.df)
bcp004.mergedf <- bcp004_relabun # bcuz there are no missing types

bcp005_relabun <- data.frame(table(bcp005.celltypes)/sum(table(bcp005.celltypes)) * 100)
bcp005_relabun$Patient <- "Ref3"
colnames(bcp005_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, bcp005_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
#missing.df$Patient <- "BCP005"
bcp005.mergedf <- rbind(bcp005_relabun, missing.df)
bcp005.mergedf <- bcp005_relabun # bcuz there are no missing types

bcp006_relabun <- data.frame(table(bcp006.celltypes)/sum(table(bcp006.celltypes)) * 100)
bcp006_relabun$Patient <- "Ref4"
colnames(bcp006_relabun)[1] <- "Celltype"
missing.celltypes <- setdiff(archR$barplot_annotation, bcp006_relabun$Celltype)
missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
colnames(missing.df) <- c("Celltype", "Freq", "Patient")
missing.df$Celltype <- missing.celltypes
#missing.df$Patient <- "BCP006"
bcp006.mergedf <- rbind(bcp006_relabun, missing.df)
bcp006.mergedf <- bcp006_relabun # bcuz there are no missing types

#########################################

stanford1_relabun <- data.frame(table(stanford1.celltypes)/sum(table(stanford1.celltypes)) * 100)
stanford1_relabun$Patient <- "Ref5"
colnames(stanford1_relabun)[1] <- "Celltype"
# missing.celltypes <- setdiff(archR$barplot_annotation, stanford1_relabun$Celltype)
# missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
# colnames(missing.df) <- c("Celltype", "Freq", "Patient")
# missing.df$Celltype <- missing.celltypes
# missing.df$Patient <- "stanford1"
# stanford1.mergedf <- rbind(stanford1_relabun, missing.df)
stanford1.mergedf <- stanford1_relabun # because it has all celltypes

stanford2_relabun <- data.frame(table(stanford2.celltypes)/sum(table(stanford2.celltypes)) * 100)
stanford2_relabun$Patient <- "Ref6"
colnames(stanford2_relabun)[1] <- "Celltype"
# missing.celltypes <- setdiff(archR$barplot_annotation, stanford2_relabun$Celltype)
# missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
# colnames(missing.df) <- c("Celltype", "Freq", "Patient")
# missing.df$Celltype <- missing.celltypes
# missing.df$Patient <- "stanford2"
# stanford2.mergedf <- rbind(stanford2_relabun, missing.df)
stanford2.mergedf <- stanford2_relabun # because it has all celltypes

stanford3_relabun <- data.frame(table(stanford3.celltypes)/sum(table(stanford3.celltypes)) * 100)
stanford3_relabun$Patient <- "Ref7"
colnames(stanford3_relabun)[1] <- "Celltype"
#missing.celltypes <- setdiff(archR$barplot_annotation, stanford3_relabun$Celltype)
#missing.df <- data.frame(matrix(NA, nrow = length(missing.celltypes), ncol = 3))
#colnames(missing.df) <- c("Celltype", "Freq", "Patient")
#missing.df$Celltype <- missing.celltypes # not missing any types
#missing.df$Patient <- "stanford3"
#stanford3.mergedf <- rbind(stanford3_relabun)
stanford3.mergedf <- stanford3_relabun # because it has all celltypes

allpatient_df <- rbind(p01.076.mergedf, p01.115.mergedf, p01.131.mergedf, p01.163.mergedf, p01.190.mergedf, p04.003.mergedf, p04.006.mergedf, 
                       wm25.mergedf, wm43.mergedf, wm46.mergedf, wm47.mergedf, wm54.mergedf, wm65.mergedf,
                       bcp003.mergedf, bcp004.mergedf, bcp005.mergedf, bcp006.mergedf, 
                       stanford1.mergedf, stanford2.mergedf, stanford3.mergedf)

pc.df <- allpatient_df[allpatient_df$Celltype == "Clonal PCs",]
```
# Stacked bar plot of cell type relative abundances - FIGURE 1C
```{r}
plot_order <- c("MBC-like MBC", "PC-like MBC", "Clonal PC", "Polyclonal MBC", "Polyclonal PC", "Lymphoplasmacyte", "Reference MBC", "Reference PC", "Reference Naive", "Reference GC")

# factor
allpatient_df$Celltype <- factor(allpatient_df$Celltype, levels = plot_order)

# similar w patients
#allpatient_df$Patient <- factor(allpatient_df$Patient, levels = c("01-076", "01-115", "01-131", "01-163", "01-190", "04-003", "04-006",
#                                                                  "WM25", "WM43", "WM46", "WM47", "WM54", "WM65",
#                                                                  "Ref1", "Ref2", "Ref3", "Ref4",
#                                                                  "Ref5", "Ref6", "Ref7"
#                                                                 ))

allpatient_df$Patient <- factor(allpatient_df$Patient, levels = c("WM46", "WM47", "WM65", "01-190", "01-076", "WM54", "04-006", "01-115",
                                                                  "01-131", "01-163", "04-003", "WM25", "WM43",
                                                                  "Ref1", "Ref2", "Ref3", "Ref4", "Ref5", "Ref6", "Ref7"))

allpatient_df$Patient <- droplevels(allpatient_df$Patient)

colors_barplot <- c("MBC-like MBC" = "firebrick", 
                    "PC-like MBC" = "slateblue",
                    "Clonal PC" = "lightseagreen",
                    "Polyclonal MBC" = "thistle2",
                    "Polyclonal PC" = "orange1",
                    "Lymphoplasmacyte" = "forestgreen",
                    "Reference MBC" = "goldenrod",
                    "Reference PC" = "darkcyan",
                    "Reference Naive" = "black",
                    "Reference GC" = "darkmagenta")

colors <- c("firebrick", "slateblue", "lightseagreen", "thistle2", "orange1", "forestgreen", "goldenrod", "darkcyan", "black", "darkmagenta")
# Stacked barplot
ggplot(allpatient_df, aes(fill=Celltype, y=Freq, x=Patient)) + 
    geom_bar(position="stack", stat="identity", width = 0.5) + scale_fill_manual(values=colors) +
  xlab(label = "Patient") + ylab("Relative Abundance") + theme_minimal() + 
  guides(fill = guide_legend(nrow = 1)) +
  theme(axis.text.y = element_text(size = "10", vjust = 0.5, hjust = 0.5),
        axis.text.x = element_text(size = "10", vjust = 0.5, hjust = 0.5, angle = 90),
        axis.title.x = element_text(size = "15", face = "bold"),
        axis.title.y = element_text(size = "15", face = "bold"),
        legend.position = "top")
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure1/WM_Fig1B_CellTypeBarplot.pdf", width = 8, height = 4) 
```
# Stacked violin plot of major marker genes Fig 1D
```{r}
colors_barplot <- c("MBC-like MBC" = "firebrick", 
                    "PC-like MBC" = "slateblue",
                    "Clonal PC" = "lightseagreen",
                    "Polyclonal MBC" = "thistle2",
                    "Polyclonal PC" = "orange1",
                    "WM43" = "forestgreen",
                    "Reference MBC" = "goldenrod",
                    "Reference PC" = "darkcyan",
                    "Reference Naive" = "black",
                    "Reference GC" = "darkmagenta")

genes <- c("PRDM1", "XBP1", "IRF4", 
              "MZB1", "CD38", "CD27", 
              "CD74", "IRF8", "BACH2",
              "MS4A1", "CD24", "PAX5")

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "GeneIntegrationMatrix", 
    name = genes,
    plotAs = "violin",
    imputeWeights = getImputeWeights(archR), 
    alpha = 0.4,
    addBoxPlot = T,
    groupOrder = c("MBC-like MBC", "PC-like MBC", "Polyclonal MBC", "WM43", "Clonal PC", "Polyclonal PC", "Reference PC", "Reference MBC", "Reference Naive", "Reference GC"),
    pal = colors_barplot,
    quantCut = c(0.975, 0.025)
   ) 
p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0.15, 0, 0.15), "cm")) +
    theme(
        panel.spacing = unit(0, "lines"),
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "10", face = "bold")
    )
})
plot_grid(plotlist = p2, ncol = 1, align = "v")
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure1/Fig1_MarkerViolins.pdf", height = 6, width = 3)
```
############################################################################
################ FIGURE 2 - SUBTYPES VS. REFERENCE #########################
############################################################################
# Subsetting MBC-like and PC-like cells for recluster
```{r}
illegal.test <- archR[mbc.like.ref.mbc.cells] # figure out which patients have fewer than 100 cells. these will mess up downstream iterative LSI
table(illegal.test$samplesClean)

mbc.like.ref.mbc.cells <- archR$cellNames[archR$final_annotation %in% c("MBC-like MBC", "Reference MBC", "Polyclonal MBC") & archR$samplesClean %ni% c("01-163", "WM43", "04-006")] # add these patients to the filtering

# Subset
mbc.like.ref.sub <- subsetArchRProject(
  ArchRProj = archR,
  cells = mbc.like.ref.mbc.cells,
  outputDirectory = "MBC-like_RefMBC_Subset",
  dropCells = TRUE,
  logFile = NULL,
  threads = getArchRThreads(),
  force = FALSE
)

# Recluster
mbc.like.ref.sub <- addIterativeLSI(
  ArchRProj = mbc.like.ref.sub,
  useMatrix = "PeakMatrix",
  name = "IterativeLSI_MBC_like",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

# Batch correct
mbc.like.ref.sub <- addHarmony(
  mbc.like.ref.sub,
  reducedDims = "IterativeLSI_MBC_like",
  name = "MBClike_Harmony",
  groupBy = "batch"
)

# Add UMAP
mbc.like.ref.sub <- addUMAP(
    ArchRProj = mbc.like.ref.sub, 
    reducedDims = "MBClike_Harmony", 
    name = "MBClike_Harmony_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

# Save it out
saveArchRProject(ArchRProj = mbc.like.ref.sub,
                 outputDirectory = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/MBC-like_RefMBC_Subset",
                 overwrite = TRUE)

### NOW PC
pc.like.ref.mbc.cells <- archR$cellNames[archR$final_annotation %in% c("PC-like MBC", "Reference MBC")]

illegal.test <- archR[pc.like.ref.mbc.cells] # figure out which patients have fewer than 100 cells. these will mess up downstream iterative LSI
table(illegal.test$samplesClean)

pc.like.ref.sub <- subsetArchRProject(
  ArchRProj = archR,
  cells = pc.like.ref.mbc.cells,
  outputDirectory = "PC-like_RefMBC_Subset",
  dropCells = TRUE,
  logFile = NULL,
  threads = getArchRThreads(),
  force = TRUE
)

### Recluster
pc.like.ref.sub <- addIterativeLSI(
  ArchRProj = pc.like.ref.sub,
  useMatrix = "PeakMatrix",
  name = "IterativeLSI_PC_like",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

pc.like.ref.sub <- addHarmony(
  pc.like.ref.sub,
  reducedDims = "IterativeLSI_PC_like",
  name = "PClike_Harmony",
  groupBy = "batch"
)


pc.like.ref.sub <- addUMAP(
    ArchRProj = pc.like.ref.sub, 
    reducedDims = "PClike_Harmony", 
    name = "PC_Harmony_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

saveArchRProject(ArchRProj = pc.like.ref.sub,
                 outputDirectory = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/PC-like_RefMBC_Subset",
                 overwrite = TRUE)
```
# Fig 2A - MBC-like UMAP
```{r}
mbc.like.ref.sub <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/MBC-like_RefMBC_Subset")

#plotEmbedding(ArchRProj = mbc.like.ref.sub, colorBy = "cellColData", name = "final_annotation", embedding = "MBClike_Harmony_UMAP")
#plotEmbedding(ArchRProj = mbc.like.ref.sub, colorBy = "cellColData", name = "samplesRefCombined", embedding = "MBClike_Harmony_UMAP")
#plotEmbedding(ArchRProj = mbc.like.ref.sub, colorBy = "cellColData", name = "hasTopCDR3", embedding = "MBClike_Harmony_UMAP")
#plotEmbedding(ArchRProj = mbc.like.ref.sub, colorBy = "cellColData", name = "MBClike_Annos", embedding = "MBClike_Harmony_UMAP")

# Make new column for this specific UMAP
mbc.like.ref.sub$MBClike_Annos <- mbc.like.ref.sub$final_annotation
mbc.like.ref.sub$MBClike_Annos[mbc.like.ref.sub$samplesClean == "WM65"] <- "WM65"
mbc.like.ref.sub$MBClike_Annos[mbc.like.ref.sub$samplesClean == "WM47"] <- "WM47"
mbc.like.ref.sub$MBClike_Annos[mbc.like.ref.sub$samplesClean == "WM46"] <- "WM46"
mbc.like.ref.sub$MBClike_Annos[mbc.like.ref.sub$samplesClean == "WM54"] <- "WM54"
mbc.like.ref.sub$MBClike_Annos[mbc.like.ref.sub$samplesClean == "01-076" & mbc.like.ref.sub$final_annotation == "MBC-like MBC"] <- "01-076"
mbc.like.ref.sub$MBClike_Annos[mbc.like.ref.sub$samplesClean == "01-190"] <- "01-190"
mbc.like.ref.sub$MBClike_Annos[mbc.like.ref.sub$MBClike_Annos == "MBC-like MBC"] <- "Polyclonal MBC"


cols <- ArchRPalettes$paired[1:length(table(mbc.like.ref.sub$MBClike_Annos))]
names(cols) <- names(table(mbc.like.ref.sub$MBClike_Annos)) # THIS IS HOW YOU USE PAL! NEED TO SET NUMBERS TO THE PROPER CLUSTER LABEL!!!! YEAAAAYUH!
cols[4] <- "ivory2"
cols[3] <- "thistle2"

plotEmbedding(ArchRProj = mbc.like.ref.sub, colorBy = "cellColData", name = "MBClike_Annos", embedding = "MBClike_Harmony_UMAP", pal = cols, 
              labelMeans = F,
              baseSize = 10,
              labelSize = 5.5,
              fgColor = "black",
              bgWidth = 0) + NoLegend() + theme(title = element_blank(),
                                                axis.title.x = element_text(size = 15, face = "bold"),
                                                axis.title.y = element_text(size = 15, face = "bold"))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure2/Figure2_MBC-like_UMAP.pdf", height = 6, width = 6)
```
# Fig 2B - MBC-like vs. Reference MBC - GEX VOLCANO PLOT
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "MBC-like MBC",
  bgdGroups = c("Reference MBC")
)

markerList <- getMarkers(markerTest, cutOff = "FDR < 100")
df.motifs <- as.data.frame(markerList$`MBC-like`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1


# Do it again with lower threshold for better volcano plot
motifs.up <- df.motifs[df.motifs$Log2FC > 0.25,]
motifs.do <- df.motifs[df.motifs$Log2FC < (-0.25),]
sorted.up <- motifs.up[order(motifs.up$FDR),]
sorted.do <- motifs.do[order(motifs.do$FDR),]

combined.df <- rbind(motifs.up, motifs.do)
combined.df$category <- NA
combined.df$category[combined.df$Log2FC > 0.5 & combined.df$FDR < 0.000000005] <- "POS"
combined.df$category[combined.df$Log2FC < (-0.5) & combined.df$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df[combined.df$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_RefMBC_Wilcoxon_MBC_Upregulated_GEX.csv")
pc.upreg.df <- na.omit(combined.df[combined.df$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_RefMBC_Wilcoxon_PC_Upregulated_GEX.csv")

up.labels <- c("ARHGAP24", "RHEX", "SOX5", "VOPP1", "OSBPL10", "PLCG2", "IL7", "PYHIN1", "VOPP1", "PTPRG", "CD24", "BCL2",  "AFF3", "IL2RA", "HCK",  "CDK1", "MKI67", "ZNF804A", "SGSM1", "EPHB1", "AKT3", "FUT8", "PTPRJ", "HDAC9", "PLEKHG1", "SYK", "CARD11", "CD79B", "TCF4", "ITGB2", "TRIO", "IL2RA", "FCRL1", "MAP3K5")
do.labels <- c("SSBP2", "FOS", "CD96", "SCML4", "CDK6", "SATB1", "FOSB", "NRIP1", "TPT1", "SLAMF1", "PAG1", "CD86", "BCL7A", "GLIPR1", "ITGB2", "PTGER4", "TRIM38", "CR1", "CR2", "CD96", "PAG1", "BCL7A", "LMO2", "ARNTL", "EGR2", "IGF1R", "NRG1", "CCL3", "ACTN1", "PLAG1")
comb.labels <- c(up.labels, do.labels)

combined.df$labels <- combined.df$name
combined.df$labels[combined.df$labels %ni% comb.labels] <- ""
table(combined.df$labels)

ggplot(combined.df, aes(Log2FC, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.8, y = 0.1, center_x = 0, direction = "split")) +
  theme_ArchR() + theme(text = element_text(size = 15),
                        axis.title.y = element_text(size = 15, face = "bold"),
                        axis.title.x = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="firebrick3", "NEG"="black", "NONE"="grey")) +
  xlab("Log2FC") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-4, 4))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure2/Figure2-MBC-like_vs_RefMBC_GEX_Volcano.pdf", height = 9, width = 8)
```
# Fig 2C - MBC-like comparison MOTIF VOLCANO
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "MBC-like MBC",
  bgdGroups = c("Reference MBC")
)

# Up in MBC-like
markerList.motifs <- getMarkers(markerTest, cutOff = "FDR <= 100")
df.motifs <- as.data.frame(markerList.motifs$`MBC-like MBC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1

# Subset out some dfs for manipulation
motifs.up <- df.motifs[df.motifs$MeanDiff > 0,]
motifs.do <- df.motifs[df.motifs$MeanDiff < -(0),]
sorted.motifs.up <- motifs.up[order(-motifs.up$log10.fdr),]
sorted.motifs.do <- motifs.do[order(-motifs.do$log10.fdr),]

# Combine and add a category for plot labels
combined.df.motifs <- rbind(sorted.motifs.up, sorted.motifs.do)
combined.df.motifs$category <- NA
combined.df.motifs$category[combined.df.motifs$MeanDiff > 0.025 & combined.df.motifs$FDR < 0.000000005] <- "POS"
combined.df.motifs$category[combined.df.motifs$MeanDiff < (-0.025) & combined.df.motifs$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/DifferentialTestResults/MBC-like_vs_RefMBC_MOTIFS_UpMBClikeMBC.csv") # write out the results while you're at it
pc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/DifferentialTestResults/MBC-like_vs_RefMBC_MOTIFS_UpRefMBC.csv")

# Filter out accidentially included labels
motifs.labels.up <- sorted.motifs.up$name[1:25]
markerMotifs.labels.up <- getFeatures(archR, select = paste(motifs.labels.up, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.up <- grep("z:", markerMotifs.labels.up, value = TRUE)
markerMotifs.labels.up <- markerMotifs.labels.up[markerMotifs.labels.up %ni% c("z:AC0021266_638")]

motifs.labels.do <- sorted.motifs.do$name[1:25]
markerMotifs.labels.do <- getFeatures(archR, select = paste(motifs.labels.do, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.do <- grep("z:", markerMotifs.labels.do, value = TRUE)

# Add a few others we want labels for
comb.labels.motifs <- c(motifs.labels.up, motifs.labels.do)
to_add <- c("SP8_226", "SP7_241", "SP9_283",  "MEF2A_639", "TCF4_97", "ID3_38", "ID4_75", "REL_721",
            "JUN_143", "STAT6_776", "ETV6_335", "SMAD4_739", "BATF_129",  "BACH2_113")
comb.labels.motifs <- c(to_add, comb.labels.motifs)
comb.labels.motifs.sub <- comb.labels.motifs[comb.labels.motifs %ni% c("ENSG00000250096_734", "DDIT3_141", "NFYC_800")] # remove ones we don't want

# Fix names
combined.df.motifs$labels <- combined.df.motifs$name
combined.df.motifs$labels[combined.df.motifs$labels %ni% comb.labels.motifs.sub] <- ""
combined.df.motifs$labels <- gsub("_*[^_]*$", "", combined.df.motifs$labels)

# Plot
ggplot(combined.df.motifs, aes(MeanDiff, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1,
                   max.overlaps = Inf, 
                   position = position_nudge_center(x = 0.06, y = 0.001, center_x = 0, direction = "split")) + 
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="goldenrod", "NEG"="black", "NONE"="grey")) +
  xlab("MeanDiff") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-0.45, 0.45)) + scale_y_continuous(limits = c(0, 140))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure2/Fig2-MBC-like-MBC_vs_RefMBC_MOTIF_Volcano.pdf", height = 9, width = 8)
```
# Fig 2D - PC-like UMAP
```{r}
pc.like.ref.sub <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/PC-like_RefMBC_Subset")

pc.like.ref.sub$PClike_Annos <- pc.like.ref.sub$samplesRefCombined
pc.like.ref.sub$PClike_Annos[pc.like.ref.sub$samplesRefCombined == "Reference"] <- "Reference MBC"

cols <- ArchRPalettes$bear[1:length(table(pc.like.ref.sub$PClike_Annos))]
names(cols) <- names(table(pc.like.ref.sub$PClike_Annos)) # THIS IS HOW YOU USE PAL! NEED TO SET NUMBERS TO THE PROPER CLUSTER LABEL!!!! YEAAAAYUH!
cols[6] <- "ivory2"
cols[7] <- "burlywood"


plotEmbedding(ArchRProj = pc.like.ref.sub, colorBy = "cellColData", name = "PClike_Annos", embedding = "PC_Harmony_UMAP", pal = cols, 
              labelMeans = F,
              baseSize = 10,
              labelSize = 6,
              fgColor = "black",
              bgWidth = 0) + NoLegend() + theme(title = element_blank(),
                                                axis.title.x = element_text(size = 15, face = "bold"),
                                                axis.title.y = element_text(size = 15, face = "bold"))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure2/Figure2_PC-like_ReclusteredUMAP.pdf", height = 6, width = 6)
```
# Fig 2E - PC LIKE MBC VS. REF MBC - GEX VOLCANO
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "PC-like MBC",
  bgdGroups = c("Reference MBC")
)

markerList <- getMarkers(markerTest, cutOff = "FDR < 100")
df.motifs <- as.data.frame(markerList$`PC-like MBC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1

# Do it again with lower threshold for better volcano plot
motifs.up <- df.motifs[df.motifs$Log2FC > 0.25,]
motifs.do <- df.motifs[df.motifs$Log2FC < (-0.25),]
sorted.up <- motifs.up[order(motifs.up$FDR),]
sorted.do <- motifs.do[order(motifs.do$FDR),]

combined.df <- rbind(motifs.up, motifs.do)
combined.df$category <- NA
combined.df$category[combined.df$Log2FC > 0.5 & combined.df$FDR < 0.000000005] <- "POS"
combined.df$category[combined.df$Log2FC < (-0.5) & combined.df$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df[combined.df$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/PC-like_vs_RefMBC_Wilcoxon_MBC_Upregulated_GEX.csv")
pc.upreg.df <- na.omit(combined.df[combined.df$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/PC-like_vs_RefMBC_Wilcoxon_PC_Upregulated_GEX.csv")

up.labels <- c("PELI1", "CARMIL1", "TRIO", "PRDM2", "PTPRG", "FBXW7", "EML6", "TLE4", "COBLL1", "EIF2AK3", "CDK14", "IFNG-AS1", "CPEB4", "MSI2", "GAB1", "ZEB1", "BCL2", "DUSP22", "VOPP1", "PMAIP1", "FOXP1", "ITM2B", "OSBPL10", "CD9", "JCHAIN", "XBP1", "HIVEP1", "FNBP1", "PIK3CA", "ATF6")
####
do.labels <- c("SORL1", "SEPT6", "SMIM14", "PAG1", "SEPT7", "NRIP1", "CLEC2B", "CHD3", "SATB1", "CD96", "TMSB4X", "CR2", "SLAMF1", "SCML4", "SLAMF1", "CHI3L2", "LMO2", "FOS", "SCML4","ACTB", "HIF1A", "CD48", "CR1", "BCL6", "RIPOR2", "TPM3", "MUM1", "CD72", "SCML1", "FCRL4", "ITGAX", "S100A8")

comb.labels <- c(up.labels, do.labels)

combined.df$labels <- combined.df$name
combined.df$labels[combined.df$labels %ni% comb.labels] <- ""
table(combined.df$labels)

ggplot(combined.df, aes(Log2FC, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.8, y = 0.1, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="slateblue1", "NEG"="black", "NONE"="grey")) +
  xlab("Log2FC") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-4.5, 4.5)) + scale_y_continuous(limits = c(0, 140))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure2/Fig2-PC-like-MBC_vs_RefMBC_GEX_Volcano.pdf", height = 9, width = 8)
```
# Fig 2F - PC-like comparison MOTIF VOLCANO
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "PC-like MBC",
  bgdGroups = c("Reference MBC")
)

# Up in MBC-like
markerList.motifs <- getMarkers(markerTest, cutOff = "FDR <= 100")
df.motifs <- as.data.frame(markerList.motifs$`PC-like MBC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1

motifs.up <- df.motifs[df.motifs$MeanDiff > 0,]
motifs.do <- df.motifs[df.motifs$MeanDiff < -(0),]
sorted.motifs.up <- motifs.up[order(-motifs.up$log10.fdr),]
sorted.motifs.do <- motifs.do[order(-motifs.do$log10.fdr),]

combined.df.motifs <- rbind(sorted.motifs.up, sorted.motifs.do)
combined.df.motifs$category <- NA
combined.df.motifs$category[combined.df.motifs$MeanDiff > 0.025 & combined.df.motifs$FDR < 0.000000005] <- "POS"
combined.df.motifs$category[combined.df.motifs$MeanDiff < (-0.025) & combined.df.motifs$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/DifferentialTestResults/PC-like_vs_RefMBC_MOTIFS_UpPClikeMBC.csv")
pc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/DifferentialTestResults/PC-like_vs_RefMBC_MOTIFS_UpReferenceMBC.csv")

motifs.labels.up <- sorted.motifs.up$name[1:30]
markerMotifs.labels.up <- getFeatures(archR, select = paste(motifs.labels.up, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.up <- grep("z:", markerMotifs.labels.up, value = TRUE)

motifs.labels.do <- sorted.motifs.do$name[1:30]
markerMotifs.labels.do <- getFeatures(archR, select = paste(motifs.labels.do, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.do <- grep("z:", markerMotifs.labels.do, value = TRUE)

comb.labels.motifs <- c(motifs.labels.up, motifs.labels.do)
comb.labels.motifs.sub <- comb.labels.motifs[comb.labels.motifs %ni% c("AC0021266_638", "ENSG00000250096_734", "KLF15_223", "MECP2_645", "DDIT3_141", "MECP2_645", "MESP2_94", "MEF2D_642", "MEF2B_643")]
to_add <- c("SNAI1_199", "LMO2_808", "REL_721", "RELA_722", "TWIST1_42", "NFKB1_719", "NFKB2_714", "ESR1_661", 
            "HIVEP1_170", "IRF8_633", "SMAD4_739", "SMAD1_744")
comb.labels.motifs.sub <- c(comb.labels.motifs.sub, to_add)

combined.df.motifs$labels <- combined.df.motifs$name
combined.df.motifs$labels[combined.df.motifs$labels %ni% comb.labels.motifs.sub] <- ""
combined.df.motifs$labels <- gsub("_*[^_]*$", "", combined.df.motifs$labels)

# Plot
ggplot(combined.df.motifs, aes(MeanDiff, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1.5,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.08, y = 0.001, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="darkcyan", "NEG"="black", "NONE"="grey")) +
  xlab("MeanDiff") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-0.35, 0.35)) + scale_y_continuous(limits = c(0, 140))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure2/Fig2-PC-like-MBC_vs_RefMBC_MOTIF_Volcano.pdf", height = 9, width = 8)
```
# FIG 2G - VENN DIAGRAM OVERLAP ANALYSIS - GENE EXPRESSION
```{r}
# Get lists
pc.like_markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "PC-like MBC",
  bgdGroups = c("Reference MBC")
)

# Subset and whatnot
pc.like_markerList <- getMarkers(pc.like_markerTest, cutOff = "FDR < 100")
pc.like_df.gex <- as.data.frame(pc.like_markerList$`PC-like MBC`)
pc.like_gex.up <- pc.like_df.gex[pc.like_df.gex$Log2FC > 0.5 & pc.like_df.gex$FDR < 0.000000005,]
pc.like_gex.do <- pc.like_df.gex[pc.like_df.gex$Log2FC < (-0.5) & pc.like_df.gex$FDR < 0.000000005,]
pc.like_sorted.up <- pc.like_gex.up[order(pc.like_gex.up$FDR),]
pc.like_sorted.do <- pc.like_gex.do[order(pc.like_gex.do$FDR),]
pc.like_gex_genes_up <- pc.like_gex.up$name
pc.like_gex_genes_do <- pc.like_gex.do$name

# Now do it for MBC-like
mbc.like_markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "MBC-like MBC",
  bgdGroups = c("Reference MBC")
)

# Subset and whatnot
mbc.like_markerList <- getMarkers(mbc.like_markerTest, cutOff = "FDR < 100")
mbc.like_df.gex <- as.data.frame(mbc.like_markerList$`MBC-like MBC`)
mbc.like_gex.up <- mbc.like_df.gex[mbc.like_df.gex$Log2FC > 0.5 & mbc.like_df.gex$FDR < 0.000000005,]
mbc.like_gex.do <- mbc.like_df.gex[mbc.like_df.gex$Log2FC < (-0.5) & mbc.like_df.gex$FDR < 0.000000005,]
mbc.like_sorted.up <- mbc.like_gex.up[order(mbc.like_gex.up$FDR),]
mbc.like_sorted.do <- mbc.like_gex.do[order(mbc.like_gex.do$FDR),]
mbc.like_gex_genes_up <- mbc.like_gex.up$name
mbc.like_gex_genes_do <- mbc.like_gex.do$name

# Get nice colors
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
gg_color_hue(8)

# Make venn
venn.gex <- venn.diagram(
        x = list(pc.like_gex_genes_up, mbc.like_gex_genes_up),
        category.names = c("", ""),
          imagetype="svg" ,
          inverted = F,
          #main.pos = c("MBC-like", "PC-like"),
          height = 480 , 
          width = 480 , 
          resolution = 300,
          compression = "lzw",
          lwd = 1,
          col=c("#00A9FF", '#F8766D'),
          fill = c(alpha("#00A9FF",0.3), alpha('#F8766D',0.3)),
          cex = 0.4,
          fontfamily = "sans",
          cat.cex = 0.3,
          cat.default.pos = "outer",
          cat.pos = c(-27, 27),
          cat.dist = c(0.055, 0.055),
          cat.fontfamily = "sans",
          cat.col = c("#00A9FF", '#F8766D'),
          #rotation = 1,
        filename = NULL,
        output=TRUE,
          # Circles
)

ggsave(venn.gex, file="/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/MBClike_vs_PClike_VennDiagram_GEX.svg", device = "svg", height = 8, width = 8)

# Writing out venn results to csvs
# Getting some top genes to include for each side
pc.like.top.names <- pc.like_sorted.up$name
mbc.like.top.names <- mbc.like_sorted.up$name

# Subset
shared <- Reduce(intersect, list(pc.like.top.names, mbc.like.top.names))
pc.unique <- setdiff(pc.like.top.names, mbc.like.top.names)
mbc.unique <- setdiff(mbc.like.top.names, pc.like.top.names)

# Write out
write.csv(shared, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/Subtype_Shared_UpRegGEX.csv")
write.csv(pc.unique, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/PC-Unique_UpRegGEX.csv")
write.csv(mbc.unique, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-Unique_UpRegGEX.csv")

# Now for downreg
pc.like.do.names <- pc.like_sorted.do$name
mbc.like.do.names <- mbc.like_sorted.do$name

# Subset
shared.do <- Reduce(intersect, list(pc.like.do.names, mbc.like.do.names))
pc.unique.do <- setdiff(pc.like.do.names, mbc.like.do.names)
mbc.unique.do <- setdiff(mbc.like.do.names, pc.like.do.names)

# Write out
write.csv(shared.do, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/Subtype_Shared_DownRegGEX.csv")
write.csv(pc.unique.do, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/PC-Unique_DownRegGEX.csv")
write.csv(mbc.unique.do, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-Unique_DownRegGEX.csv")
```
# FIG 2H - VENN DIAGRAM OVERLAP ANALYSIS - MOTIFS
```{r}
# PC-like MBC motif
pc.like_markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "PC-like MBC",
  bgdGroups = c("Reference MBC")
)

# Subset
pc.like_markerList <- getMarkers(pc.like_markerTest, cutOff = "FDR < 100")
pc.like_df.motif <- as.data.frame(pc.like_markerList$`PC-like MBC`)
pc.like_motif.up <- pc.like_df.motif[pc.like_df.motif$MeanDiff > 0.025 & pc.like_df.motif$FDR < 0.000000005,]
pc.like_motif.up.stdev <- pc.like_df.motif[abs(pc.like_df.motif$MeanDiff - mean(pc.like_df.motif$MeanDiff)) > sd(pc.like_df.motif$MeanDiff),]
pc.like_motif.do <- pc.like_df.motif[pc.like_df.motif$MeanDiff < (-0.025) & pc.like_df.motif$FDR < 0.000000005,]
pc.like_sorted.up <- pc.like_motif.up[order(pc.like_motif.up$FDR),]
pc.like_sorted.up.stdev <- pc.like_motif.up.stdev[order(pc.like_motif.up.stdev$FDR),]
pc.like_sorted.do <- pc.like_motif.do[order(pc.like_motif.do$FDR),]

pc.like_motif_genes_up <- pc.like_motif.up$name
pc.like_motif_genes_do <- pc.like_motif.do$name

# Now for MBC-like
mbc.like_markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "MBC-like MBC",
  bgdGroups = c("Reference MBC")
)

mbc.like_markerList <- getMarkers(mbc.like_markerTest, cutOff = "FDR < 100")
mbc.like_df.motif <- as.data.frame(mbc.like_markerList$`MBC-like MBC`)
mbc.like_motif.up <- mbc.like_df.motif[mbc.like_df.motif$MeanDiff > 0.025 & mbc.like_df.motif$FDR < 0.000000005,]
mbc.like_motif.up.stdev <- mbc.like_df.motif[abs(mbc.like_df.motif$MeanDiff - mean(mbc.like_df.motif$MeanDiff)) > sd(mbc.like_df.motif$MeanDiff),]
mbc.like_motif.do <- mbc.like_df.motif[mbc.like_df.motif$MeanDiff < (-0.025) & mbc.like_df.motif$FDR < 0.000000005,]
mbc.like_sorted.up <- mbc.like_motif.up[order(mbc.like_motif.up$FDR),]
mbc.like_sorted.up.stdev <- mbc.like_motif.up.stdev[order(mbc.like_motif.up.stdev$FDR),]
mbc.like_sorted.do <- mbc.like_motif.do[order(mbc.like_motif.do$FDR),]

mbc.like_motif_genes_up <- mbc.like_motif.up$name
mbc.like_motif_genes_do <- mbc.like_motif.do$name

gg_color_hue(8)
venn.diagram(
        x = list(pc.like_motif_genes_up, mbc.like_motif_genes_up),
        category.names = c("", ""),
          imagetype="png" ,
          inverted = T,
          height = 480 , 
          width = 480 , 
          resolution = 300,
          compression = "lzw",
          lwd = 1,
          col=c("#00BFC4", '#00A9FF'),
          fill = c(alpha("#00BFC4",0.3), alpha('#00A9FF',0.3)),
          cex = 0.5,
          fontfamily = "sans",
          cat.cex = 0.3,
          cat.default.pos = "outer",
          cat.pos = c(-27, 27),
          cat.dist = c(0.055, 0.055),
          cat.fontfamily = "sans",
          cat.col = c("#00BFC4", '#00A9FF'),
          #rotation = 180,
        filename = 'MBClike_PClike_MOTIFS_venn_diagramm_Up.png',
        output=TRUE,
          # Circles
)

# Getting some top genes to include for each side
pc.like.top.names <- gsub("_.*", "", pc.like_sorted.up$name)
mbc.like.top.names <- gsub("_.*", "", mbc.like_sorted.up$name)

# Get overlaps and uniques
shared <- Reduce(intersect, list(pc.like.top.names, mbc.like.top.names))
pc.unique <- setdiff(pc.like.top.names, mbc.like.top.names)
mbc.unique <- setdiff(mbc.like.top.names, pc.like.top.names)

# Write out
write.csv(shared, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/Venn_Subtype_Shared_UpReg_Motif.csv")
write.csv(pc.unique, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/Venn_PC-Unique_UpReg_Motif.csv")
write.csv(mbc.unique, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/Venn_MBC-Unique_UpReg_Motif.csv")
```
############################################################################
################ FIGURE 3 - SUBTYPES VS. EACH OTHER ########################
############################################################################
# Fig 3A - GEX volcano
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "MBC-like MBC",
  bgdGroups = c("PC-like MBC")
)

markerList <- getMarkers(markerTest, cutOff = "FDR < 100")
df.motifs <- as.data.frame(markerList$`MBC-like MBC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1
motifs.up <- df.motifs[df.motifs$Log2FC > 0.5 & df.motifs$FDR < 0.000000005,]
motifs.do <- df.motifs[df.motifs$Log2FC < (-0.5) & df.motifs$FDR < 0.000000005,]
sorted.up <- motifs.up[order(motifs.up$FDR),]
write.csv(sorted.up, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_GEX_UpMBC-like.csv")
sorted.do <- motifs.do[order(motifs.do$FDR),]
write.csv(sorted.do, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_GEX_UpPC-like.csv")

# Do it again with lower threshold for better volcano plot
motifs.up <- df.motifs[df.motifs$Log2FC > 0.25,]
motifs.do <- df.motifs[df.motifs$Log2FC < (-0.25),]
sorted.up <- motifs.up[order(motifs.up$FDR),]
sorted.do <- motifs.do[order(motifs.do$FDR),]

combined.df <- rbind(motifs.up, motifs.do)
combined.df$category <- NA
combined.df$category[combined.df$Log2FC > 0.5 & combined.df$FDR < 0.000000005] <- "POS"
combined.df$category[combined.df$Log2FC < (-0.5) & combined.df$FDR < 0.000000005] <- "NEG"

mbc.upreg.df <- na.omit(combined.df[combined.df$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_Motifs_MBC_Upregulated_GEX.csv")
pc.upreg.df <- na.omit(combined.df[combined.df$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_Motifs_PC_Upregulated_GEX.csv")


up.labels <- c("MARCH1", "RIPOR2", "SMIM14", "PLD5", "FCRL1", "AFF3", "MYO1F", "SESN3", "SGSM1","HLA-DRA", "FCHSD2", "CR2", "CD52", "ACTB", "MAP4K4",?
  "ZEB2", "CD1C", "CD24", "HIF1A", "SOX5", "IRAK3", "VAV3", "LILRB1", "BCL11A", "SPI1", "CR1", "SYK", "IRF8", "CD74", "EPHB1", "FCRL4", "PAX5", "SPIB", "BCL6")

do.labels <- c("PELI1", "EIF2AK3", "CARMIL1", "PMAIP1", "MSI2", "DUSP22", "PRDM5", "TLE", "CPEB4", "ELL2",  "LRCH1", "ERN1", "IFNG-AS1", "JCHAIN", "ZEB1", "XBP1", "CD9", "ITM2B", "AFF1",  "MYO6", "TOP1", "RRAS2", "CD9", "CXXC4-AS1", "IL15", "PRDM2", "GAB1", "CD27", "IGF1", "PIK3CA", "CADPS2", "IL6ST", "PIEZO2", "ETV6", "SMAD1", "IL15", "FOXO1", "JUND", "CD38", "RELB", "NFKB2", "REL")

comb.labels <- c(up.labels, do.labels)

combined.df$labels <- combined.df$name
combined.df$labels[combined.df$labels %ni% comb.labels] <- ""
table(combined.df$labels)

ggplot(combined.df, aes(Log2FC, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.8, y = 0.1, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="firebrick3", "NEG"="slateblue1", "NONE"="grey")) +
  xlab("Log2FC") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-6, 6)) #+ coord_flip()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure3/Figure3_MBC-like_vs_PC-like_GEX_Volcano.pdf", height = 8, width = 7)
```
# Fig 3B - UMAPs of key genes
```{r}
genes <- c("SPI1", "BCL11A", "FCHSD2", "VAV3",
           "NFKB1", "REL", "PELI1", "CD27",
           "BCL2", "PLCG2", "FOXP1", "OSBPL10")

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.025, 0.975),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(archR)
)
p

p2 <- lapply(p, function(x){ 
    x +
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    ) 
}) + NoLegend()

do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure3/Fig3_SubtypeComparison_GEX_UMAPs.pdf", height = 9, width = 12)

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x +
    #guides(color = FALSE, fill = FALSE) +
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(legend.title = element_text(size = 6),
          legend.text = element_text(size = 6),
          legend.key.size = unit(0.45, "cm"),
          #legend.title.position = "left",
          legend.position = c(0.88, 0.775),#"right",
          legend.direction = "vertical",
          #legend.margin = margin(0,0,0,0),
          legend.box.margin = margin(-10,-10,-10,-10)) +
    theme(
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank(),
        axis.text.x=element_blank(),
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure3/Fig3_SubtypeComparison_GEX_UMAPs_LegendSave.pdf", height = 9, width = 12)
```
# Fig 3C - MOTIF volcano
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "MBC-like MBC",
  bgdGroups = c("PC-like MBC")
)

# Up in MBC-like
markerList.motifs <- getMarkers(markerTest, cutOff = "FDR <= 100")
df.motifs <- as.data.frame(markerList.motifs$`MBC-like MBC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1

up <- df.motifs[df.motifs$MeanDiff > 0.025 & df.motifs$FDR < 0.000000005,]
do <- df.motifs[df.motifs$MeanDiff < -(0.025) & df.motifs$FDR < 0.000000005,]
sorted.up <- up[order(-up$log10.fdr),]
write.csv(sorted.up, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_Motifs_UpMBC-like.csv")
sorted.do <- do[order(-do$log10.fdr),]
write.csv(sorted.do, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_Motifs_UpPC-like.csv")

motifs.up <- df.motifs[df.motifs$MeanDiff > 0.01,]
motifs.do <- df.motifs[df.motifs$MeanDiff < -(0.01),]
sorted.motifs.up <- motifs.up[order(-motifs.up$log10.fdr),]
sorted.motifs.do <- motifs.do[order(-motifs.do$log10.fdr),]

combined.df.motifs <- rbind(sorted.motifs.up, sorted.motifs.do)
combined.df.motifs$category <- NA
combined.df.motifs$category[combined.df.motifs$MeanDiff > 0.025 & combined.df.motifs$FDR < 0.000000005] <- "POS"
combined.df.motifs$category[combined.df.motifs$MeanDiff < (-0.025) & combined.df.motifs$FDR < 0.000000005] <- "NEG"

mbc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_Motifs_MBC_Upregulated.csv")
pc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/MBC-like_vs_PC-like_Wilcoxon_Motifs_PC_Upregulated.csv")

motifs.labels.up <- combined.df.motifs$name[combined.df.motifs$category == "POS"]
markerMotifs.labels.up <- getFeatures(archR, select = paste(motifs.labels.up, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.up <- grep("z:", markerMotifs.labels.up, value = TRUE)

motifs.labels.do <- sorted.motifs.do$name[1:30]
markerMotifs.labels.do <- getFeatures(archR, select = paste(motifs.labels.do, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.do <- grep("z:", markerMotifs.labels.do, value = TRUE)


comb.labels.motifs <- c(markerMotifs.labels.up, markerMotifs.labels.do)
to_add <- c("IRF8_633", "BATF_129", "NFKB2_714", "BACH2_113", "REL_721", "NFKB1_719", "ID1_47", "ID2_35", "XBP1_109", "EBF1_67", "FOXO1_361", "IRF5_631")
comb.labels.motifs <- c(to_add, comb.labels.motifs)
comb.labels.motifs <- gsub("z:", "", comb.labels.motifs)
comb.labels.motifs <- comb.labels.motifs[comb.labels.motifs %ni% c("NANOGP1_835", "ATOH8_71", "SMARCC1_651", "NFE2_119", "NFIX_738", "ENSG00000235187_346", "NANOG_433", "SNAI3_268", "ARID2_11", "NFE2L2_115", "SREBF1_22", "EBF1_67", "NAIF1_637", "ID1_47", "ID2_35", "FOXO1_361", "XBP1_109", "IRF5_631")]

combined.df.motifs$labels <- combined.df.motifs$name
combined.df.motifs$labels[combined.df.motifs$labels %ni% comb.labels.motifs] <- ""
combined.df.motifs$labels <- gsub("_*[^_]*$", "", combined.df.motifs$labels)

# Plot
ggplot(combined.df.motifs, aes(MeanDiff, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1.5,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.08, y = 0.001, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="goldenrod", "NEG"="darkorchid4", "NONE"="grey")) +
  xlab("MeanDiff") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-0.25, 0.25)) + scale_y_continuous(limits = c(0, 60))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure3/Figure3_MBClike_vs_PClike_MOTIF_Volcano.pdf", height = 8, width = 7)
```
# FIG D - UMAPs of key motifs
```{r}
motifs <- c("SPI1", "SPIB", "BCL11A", "ELF5",
            "NFKB1", "REL", "ZEB1", "POU2F1",
            "ARID3A", "FUBP1", "HOXD8", "PITX2")

motif.labels <- getFeatures(archR, select = paste(motifs, collapse="|"), useMatrix = "MotifMatrix")
motif.labels <- grep("z:", motif.labels, value = TRUE)
motif.labels <- motif.labels[c(11,9,12,10,
                               4,3,13,6,
                               14,1,7,8)]
p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "MotifMatrix", 
    name = motif.labels, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.025, 0.975),
    pal = ArchRPalettes$horizonExtra,
    imputeWeights = getImputeWeights(archR)
)

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x + 
    #guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(legend.title = element_text(size = 6),
          legend.text = element_text(size = 6),
          legend.key.size = unit(0.45, "cm"), 
          #legend.title.position = "left",
          legend.position = c(0.88, 0.775),#"right",
          legend.direction = "vertical",
          #legend.margin = margin(0,0,0,0),
          legend.box.margin = margin(-10,-10,-10,-10)) +
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    ) + labs(fill = "DevScore")
})

do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure3/WM_Figure3_SubtypeComparison_MOTIF_UMAPs.pdf", height = 9, width = 12)
```
# Fig 3E/F - GSEA
```{r}
DE.markers <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = c("MBC-like MBC"),
  bgdGroups = c("PC-like MBC")
)

DE.markers_list <- getMarkers(DE.markers) # DO NOT FILTER GENES!!!! FOR GSERA!!!!! FEKEKEEEE

df.gex <- DE.markers_list$`MBC-like MBC`
df.gex <- as.data.frame(df.gex)
df.gex <- df.gex[order(-df.gex$MeanDiff),]

# Generate rank score by mulitplying avg log FC by negative log10 of p value.
rank.score <- sign(df.gex$Log2FC) * -log10(df.gex$FDR)
#rank.score <- normaldf.gex$MeanDiff

# Generate gene list
genes <- df.gex$name

# Put into df
rank.df <- data.frame(genes, rank.score)
rownames(rank.df) <- rank.df$genes
rank.df <- rank.df[order(-rank.df$rank.score),]

# Save out
write.table(rank.df, "/Users/gagled01/morganLab/Waldenstroms/GSEA/ArchR_MBC-likeMBC_vs_PC-likeMBC_forGSEA_UpMBC-like.rnk",
            sep = "\t",
            quote = F,
            row.names = F,
            col.names = F)
```
# AT THIS POINT YOU NEED TO TAKE THE RANK LISTS AND THEN RUN GSEA PRERANKED ON THEM!
# IT'S EASY! GOOGLE IT!
# Load GSEA results and plot
```{r}
# load GSEA outputs
mbc.like_vs_pc.like_gsea <- read.table("/Users/gagled01/morganLab/Waldenstroms/singlecell/MBClike_vs_PClike_RankedList_upMBC.tsv", sep = "\t", header = T)
pc.like_vs_mbc.like_gsea <- read.table("/Users/gagled01/morganLab/Waldenstroms/singlecell/MBClike_vs_PClike_RankedList_upPC.tsv", sep = "\t", header = T)

# Create df
plotting.df <- data.frame(mbc.like_vs_pc.like_gsea$NAME, mbc.like_vs_pc.like_gsea$NES, mbc.like_vs_pc.like_gsea$FDR.q.val, mbc.like_vs_pc.like_gsea$SIZE)
plotting.df <- head(plotting.df, n = 25)
colnames(plotting.df) <- c("Pathway", "NES", "FDR.q.val", "Count")
plotting.df$Pathway <- gsub("HALLMARK_", "", plotting.df$Pathway)
plotting.df$Pathway <- c("Allograft Rejection", "Apical Junction", "G2M Checkpoint", "E2F Targets", "Mitotic Spindle", "Interferon Gamma Response", "IL2/STAT5 Signaling", 
                         "Myogenesis", "PI3K/AKT/MTOR Signaling", "MYC Targets v1", "Complement", "MTORC1 Signaling", "p53 Pathway", "KRAS Signaling", "Apoptosis", "DNA Repair", "Oxidative Phosphorylation",
                         "Peroxisome", "UV Response Down", "Interferon Alpha Response", "Cholesterol Homeostasis", "Estrogen Response Late", "Androgen Response", "Bile Acid Metabolism", "Coagulation" 
                         )
plotting.df$log10FDR.qval <- -log10(plotting.df$FDR.q.val)

plotting.df2 <- data.frame(pc.like_vs_mbc.like_gsea$NAME, pc.like_vs_mbc.like_gsea$NES, pc.like_vs_mbc.like_gsea$FDR.q.val, pc.like_vs_mbc.like_gsea$SIZE)
plotting.df2 <- head(plotting.df2, n = 25)
colnames(plotting.df2) <- c("Pathway", "NES", "FDR.q.val", "Count")
#plotting.df2$FDR.q.val[1:2] <- 0.0001 # Manually giving some v low Q values because 0 does not work for plotting
plotting.df2$Pathway <- gsub("HALLMARK_", "", plotting.df2$Pathway)
plotting.df2$Pathway <- c("Heme Metabolism", "Estrogen Response Early", "Estrogen Response Late", "Apoptosis", "Androgen Response", "MTORC1 Signaling", "Inflammatory Response",
                         "IL2/STAT5 Signaling", "Adipogenesis", "TNFa Signaling via NF-kB", "KRAS Signaling", "Complement", "Xenobiotic Metabolism", "Interferon Gamma Response",
                         "Glycolysis", "Hypoxia", "Myogenesis", "UV Response Down", "UV Response UP", "p53 Pathway", "Epithelial-Mesenchymal Transition", "Apical Junction")
plotting.df2$log10FDR.qval <- -log10(plotting.df2$FDR.q.val)

# And plot
ggplot(plotting.df, aes(x=NES, y=reorder(Pathway,-FDR.q.val), size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("Clonal PC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/ClonalPC_vs_RefPC_GSEA_Dotplot_upClonalPC_FDRqval.png", height = 8, width = 8)

ggplot(plotting.df2, aes(x=NES, y=reorder(Pathway,-FDR.q.val),size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("Refrence PC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/ClonalPC_vs_RefPC_GSEA_Dotplot_upRefPC_FDRqval.png", height = 8, width = 8)
```
############################################################################
#################### FIGURE 4 - TRAJECTORY ANALYSIS ########################
############################################################################
# Fig 4A - Pseudotime trajectory heatmap
```{r}
# Set the trajectory route
trajectory <- c("Polyclonal MBC", "MBC-like MBC", "PC-like MBC", "Clonal PC")

# Add it
archR <- addTrajectory(
    ArchRProj = archR, 
    name = "PartialTrajectory", 
    groupBy = "final_annotation",
    reducedDims = "FinalHarmony",
    trajectory = trajectory, 
    embedding = "UMAP_Harmony", 
    force = TRUE
)

# Plot it
p <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "cellColData", name = "PartialTrajectory", continuousSet = "blueYellow",  embedding = "UMAP_Harmony", baseSize = 12)
p[[1]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/PartialTrajectory_UMAP.png", height = 6, width = 6)
```
# Correlate trajectories
```{r}
# As per ArchR's correlation methodology; see https://www.archrproject.com/bookdown/trajectory-analysis-with-archr.html

# Get trajectory for each
trajMM  <- getTrajectory(ArchRProj = archR, name = "PartialTrajectory", useMatrix = "MotifMatrix", log2Norm = FALSE)
trajGIM  <- getTrajectory(ArchRProj = archR, name = "PartialTrajectory", useMatrix = "GeneIntegrationMatrix", log2Norm = FALSE)

# Correlate and find the correlated genes
corGIM_MM <- correlateTrajectories(trajGIM, trajMM)
corGIM_MM[[1]]$matchname1
trajGIM2 <- trajGIM[corGIM_MM[[1]]$name1, ]
trajMM2 <- trajMM[corGIM_MM[[1]]$name2, ]
trajCombined <- trajGIM2
assay(trajCombined, withDimnames = F) <- t(apply(assay(trajGIM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))
rowOrder <- match(rownames(combinedMat), rownames(trajGIM2))
```
# Fig 4C/E - Heatmaps
```{r}
# Plot 4C heatmap 
p1 <- plotTrajectoryHeatmap(trajCombined, pal = paletteContinuous(set = "horizonExtra"))
p1
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure4C_Trajectory_GEX_Heatmap.png", height = 6, width = 6)

# Plot 4E heatmap
p2 <- plotTrajectoryHeatmap(trajGIM, pal = paletteContinuous(set = "solarExtra"))
p2
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure4E_Trajectory_GEX_Heatmap.png", height = 6, width = 6)
```
# Fig 4D - GEX as a function of pseudotime for key correlated genes
```{r}
p1 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "SPI1", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p2 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "SPIB", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p3 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "BCL11A", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p4 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "STAT3", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p5 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "NFKB1", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p6 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "RELB", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p7 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "XBP1", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p8 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "ATF6", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p9 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "FOXO1", embedding = "UMAP_Harmony",continuousSet = "solarExtra")
p10 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "GeneIntegrationMatrix", name = "TGIF2", embedding = "UMAP_Harmony",continuousSet = "solarExtra")

p1[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_SPI1.png", width = 4, height = 3)
p2[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_SPIB.png", width = 4, height = 3)
p3[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_BCL11A.png", width = 4, height = 3)
p4[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_STAT3.png", width = 4, height = 3)
p5[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_NFKB1.png", width = 4, height = 3)
p6[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_RELB.png", width = 4, height = 3)
p7[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_XPB1.png", width = 4, height = 3)
p8[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_ATF6.png", width = 4, height = 3)
p9[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_FOXO1.png", width = 4, height = 3)
p10[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_GEX_TGIF2.png", width = 4, height = 3)
```
# Fig 4F - Motif enrichment as a function of pseudotime for key correlated genes
```{r}
motif.genes <- c("SPI1", "SPIB", "BCL11A", "STAT3", "NFKB1", "RELB", "XBP1", "ATF6", "FOXO1", "TGIF2")
motif.labels <- getFeatures(archR, select = paste(motif.genes, collapse="|"), useMatrix = "MotifMatrix")
motif.labels <- grep("z:", motif.labels, value = TRUE)
motif.labels

p1 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "SPI1", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p2 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "SPIB", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p3 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "BCL11A", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p4 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "STAT3", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p5 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "NFKB1", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p6 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "RELB", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p7 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "XBP1", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p8 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "ATF6", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p9 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "FOXO1", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")
p10 <- plotTrajectory(archR, trajectory = "PartialTrajectory", colorBy = "MotifMatrix", name = "TGIF2", embedding = "UMAP_Harmony",continuousSet = "horizonExtra")

p1[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_SPI1.png", width = 4, height = 3)
p2[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_SPIB.png", width = 4, height = 3)
p3[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_BCL11A.png", width = 4, height = 3)
p4[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_STAT3.png", width = 4, height = 3)
p5[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_NFKB1.png", width = 4, height = 3)
p6[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_RELB.png", width = 4, height = 3)
p7[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_XPB1.png", width = 4, height = 3)
p8[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_ATF6.png", width = 4, height = 3)
p9[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_FOXO1.png", width = 4, height = 3)
p10[[2]]
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Pseudotime_Motif_TGIF2.png", width = 4, height = 3)
```
