---
title: "SupplementaryFigures"
output: html_document
date: "2024-08-28"
---

# Load Seurat object for supp figures
```{r}
library(Seurat)
library(ggplot2)
library(SeuratWrappers)

options(future.globals.maxSize=10000000000000000000) # Set max global size so we don't run out of memory
scrna <- readRDS("/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/SeuratAnalysis_ReclusteredObject_12.3.2024.rds")
```
# Supp figure 1
```{r}
# Supp figure 1A
DimPlot(scrna.subby, raster = F, group.by = "majority_voting") + ggtitle("") + theme(axis.text.x = element_blank(),
                                                                       axis.text.y = element_blank(),
                                                                       axis.title.x = element_blank(),
                                                                       axis.title.y = element_blank(),
                                                                       axis.ticks.y = element_blank(),
                                                                       axis.ticks.x = element_blank()) + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/scRNA_PatientsRef_UMAP_CellTypes.png", height = 6, width = 6)

genes <- c("MS4A1", "CD74", "HLA-DRA",
           "JCHAIN", "MZB1", "XBP1",
           "MME", "MKI67", "VPREB1",
           "IGLL1", "CD24", "CD79B")

# Supp Figure 1C - Violin plots
VlnPlot(scrna.subby, features = genes, group.by = "majority_voting", pt.size = 0) & theme(axis.text.x = element_blank(),
                                                                             axis.title.x = element_blank())
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFig1_scRNA_MarkerGene_ViolinPlots.pdf", height = 9, width = 12)

# Supp Figure 1D - UMAP expression
FeaturePlot(scrna.subby, features = genes) & theme(axis.text.x = element_blank(),
                                                   axis.text.y = element_blank(),
                                                   axis.title.x = element_blank(),
                                                   axis.title.y = element_blank())
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFig1_scRNA_MarkerGene_FeaturePlots.pdf", height = 9, width = 12)
```
# Supp figure 2 - MBC surface markers 
```{r}
Idents(scrna) <- "majority_voting"
scrna.mbc.sub <- subset(scrna, idents = c("Memory B cells"))

DefaultAssay(scrna.mbc.sub) <- "integrated"
scrna.mbc.sub <- ScaleData(scrna.mbc.sub)
scrna.mbc.sub <- RunPCA(scrna.mbc.sub, npcs = 20)
scrna.mbc.sub <- FindNeighbors(scrna.mbc.sub, dims = 1:20)
scrna.mbc.sub <- FindClusters(scrna.mbc.sub, resolution = 0.5)
scrna.mbc.sub <- RunUMAP(scrna.mbc.sub, dims = 1:20)

mbc.surfacemarker.genes <- c("IGHM", "IGHG1", "IGHD", "IGHA1", "CD27", "FCRL4", "FCRL5", "RUNX2",
           "ITGAX", "SOX5", "DLL1", "AICDA", "CR2", "CXCR5")

# Feature Plots
DefaultAssay(scrna.mbc.sub) <- "RNA"
FeaturePlot(scrna.mbc.sub, features = mbc.surfacemarker.genes) & theme(axis.text.x = element_blank(),
                                                   axis.text.y = element_blank(),
                                                   axis.title.x = element_blank(),
                                                   axis.title.y = element_blank())
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure2/scRNA_Seurat_FeaturePlots.pdf", height = 9, width = 12)

# Violin Plots
DefaultAssay(scrna.mbc.sub) <- "RNA"
VlnPlot(scrna.mbc.sub, features = mbc.surfacemarker.genes, group.by = "patient", pt.size = 0) & theme(axis.text.x = element_blank(),
                                                                             axis.title.x = element_blank())
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure2/scRNA_ViolinPlots.pdf", height = 9, width = 12)

# UMAP
DimPlot(scrna.mbc.sub, raster = F, group.by = "patient") + ggtitle("") + theme(axis.text.x = element_blank(),
                                                                       axis.text.y = element_blank(),
                                                                       axis.title.x = element_blank(),
                                                                       axis.title.y = element_blank(),
                                                                       axis.ticks.y = element_blank(),
                                                                       axis.ticks.x = element_blank()) + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure2/scRNA_Patients_MBCRecluster_UMAPs.pdf", height = 6, width = 6)

# Heatmap unscaled
DefaultAssay(scrna.mbc.sub) <- "RNA"
DoHeatmap(scrna.mbc.sub, features = mbc.surfacemarker.genes, group.by = "patient", slot = "data")  # Unscaled
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure2/scRNA_MBC_SurfaceMarkers_Heatmap_Unscaled.png", height = 6, width = 6)
```
# load up ArchR object
```{r}
library(ArchR)
library(Seurat)
library(ggplot2)
library(ggrepel)
library(ggpp)
library(SummarizedExperiment)
addArchRThreads(threads = 1)
addArchRGenome("hg38")
options(future.globals.maxSize=1000000000000000000000000000) 
archR <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_DeviationsAdded_BatchCorrected")

# Fix annotations
archR$final_annotation[archR$final_annotation == "Non-clonal MBC"] <- "Polyclonal MBC"
archR$final_annotation[archR$final_annotation == "Non-clonal PC"] <- "Polyclonal PC"
archR$final_annotation[archR$final_annotation == "Intermediate MBC"] <- "WM43"
table(archR$final_annotation)

# Make new col for patient UMAP annotations
archR$samplesRefCombined <- archR$samplesClean
archR$samplesRefCombined[archR$condition == "HD"] <- "Reference"

# Set base colors and plot orders
# Plot order for violin plots etc.
plot_order <- c("MBC-like MBC", "PC-like MBC", "Clonal PC", "Polyclonal MBC", "Polyclonal PC", "WM43", "Reference MBC", "Reference PC", "Reference Naive", "Reference GC")

cols <- table(archR$final_annotation)
names(cols) <- names(table(archR$final_annotation)) 
cols[1] <- "lightseagreen"
cols[2] <- "forestgreen"
cols[3] <- "firebrick"
cols[4] <- "slateblue"
cols[5] <- "thistle2"
cols[6] <- "orange1"
cols[7] <- "darkmagenta"
cols[8] <- "goldenrod"
cols[9] <- "black"
cols[10] <- "darkcyan"
  
colors_barplot <- c("MBC-like MBC" = "firebrick", 
                    "PC-like MBC" = "slateblue",
                    "Clonal PC" = "lightseagreen",
                    "Polyclonal MBC" = "thistle2",
                    "Polyclonal PC" = "orange1",
                    "WM43" = "forestgreen",
                    "Reference MBC" = "goldenrod",
                    "Reference PC" = "darkcyan",
                    "Reference Naive" = "black",
                    "Reference GC" = "darkmagenta")


plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "final_annotation", embedding = "UMAP_Harmony", pal = colors_barplot, labelAsFactors = F, labelMeans = F) + ggtitle("") + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Figure1/Fig1E_WM_Annotation_UMAP_Final.pdf", width = 6, height = 6) 
```
# Supplementary Figure 3 - TRUST 4 patient reclusters
# Patient subset function
```{r}
patient_subset <- function(archR_obj, id_str) {
  idx <- BiocGenerics::which(archR$samplesClean == id_str)
  cells <- archR_obj$cellNames[idx]
  sub_archR <- archR_obj[cells,]
  return(sub_archR)
}
```
# Run it on em
```{r}
p01_076_archR <- patient_subset(archR, "01-076")
p01_115_archR <- patient_subset(archR, "01-115")
p01_131_archR <- patient_subset(archR, "01-131")
p01_163_archR <- patient_subset(archR, "01-163")
p01_190_archR <- patient_subset(archR, "01-190")
p04_003_archR <- patient_subset(archR, "04-003")
p04_006_archR <- patient_subset(archR, "04-006")
wm25_archR <- patient_subset(archR, "WM25")
wm43_archR <- patient_subset(archR, "WM43")
wm46_archR <- patient_subset(archR, "WM46")
wm47_archR <- patient_subset(archR, "WM47")
wm54_archR <- patient_subset(archR, "WM54")
wm65_archR <- patient_subset(archR, "WM65")
```
Reclustering individual patients and plot
```{r}
patient_recluster <- function(archR_obj, sample_str) {
  print(paste0("Reclustering patient ", sample_str, "..."))
  print("Running Iterative LSI...")
  
  # Add iterative LSI
  archR_obj <- addIterativeLSI(
  ArchRProj = archR_obj,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

  print("Adding UMAP...")
# Add UMAP
  archR_obj <- addUMAP(
      ArchRProj = archR_obj, 
      reducedDims = "Patient_Recluster", 
      name = "Patient_Recluster_UMAP", 
      nNeighbors = 30, 
      minDist = 0.5, 
      metric = "cosine",
      force = T
  )
  return(archR_obj)
}
```
# Alternative clustering for WM46 and 65 (bcuz they are too homogenous!)
```{r}
patient_recluster2 <- function(archR_obj, sample_str) {
  print(paste0("Reclustering patient ", sample_str, "..."))
  print("Running Iterative LSI...")
  
  # Add iterative LSI
  archR_obj <- addIterativeLSI(
  ArchRProj = archR_obj,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:10,
  force = T
)

  print("Adding UMAP...")
# Add UMAP
  archR_obj <- addUMAP(
      ArchRProj = archR_obj, 
      reducedDims = "Patient_Recluster", 
      name = "Patient_Recluster_UMAP", 
      nNeighbors = 30, 
      minDist = 0.5, 
      metric = "cosine",
      force = T
  )
  return(archR_obj)
}
```
# Recluster them
```{r}
p01_076_archR <- patient_recluster(p01_076_archR, "01-076")
p01_115_archR <- patient_recluster(p01_115_archR, "01-115")
p01_131_archR <- patient_recluster(p01_131_archR, "01-131")
p01_163_archR <- patient_recluster(p01_163_archR, "01-163")
p01_190_archR <- patient_recluster(p01_190_archR, "01-190")
p04_003_archR <- patient_recluster(p04_003_archR, "04-003")
p04_006_archR <- patient_recluster(p04_006_archR, "04-006")
wm25_archR <- patient_recluster(wm25_archR, "WM25")
wm43_archR <- patient_recluster(wm43_archR, "WM43")
wm46_archR <- patient_recluster2(wm46_archR, "WM46")
wm47_archR <- patient_recluster(wm47_archR, "WM47")
wm54_archR <- patient_recluster(wm54_archR, "WM54")
wm65_archR <- patient_recluster2(wm65_archR, "WM65")
```
Read in Immunarch style reports
```{r}
# Load barcode files
p01_076_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-076_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_115_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-115_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_131_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-131_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_163_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-163_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_190_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-190_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p04_003_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_04-003_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p04_006_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_04-006_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm25_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM25_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm43_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM43_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm46_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM46_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm47_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM47_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm54_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM54_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm65_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM65_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
```
Check patient barcodes match
```{r}
cell_id_fixed <- gsub("0.-..._", "", archR$cellNames_clean)
archR$cell_id_fixed <- cell_id_fixed
cell_id_fixed2 <- gsub("-1", "", archR$cell_id_fixed)
archR$cell_id_fixed <- cell_id_fixed2
cell_id_fixed3 <- gsub("WM..", "", archR$cell_id_fixed)
archR$cell_id_fixed <- cell_id_fixed3

# Wait actually need this
patient_barcode <- gsub("-1$", "", archR$cellNames_clean) # need to use $ to specify end of string bcuz otherwise the -1 in 01-115 (and the 3 other patients like this) will get replaced!!! fucking up matching!!!
archR$patient_barcode_clean <- patient_barcode
```
# Fixing TRUST4 barcodes in the TRUST4 output for each patient. Otherwise you run the risk of overmatching cells due to the non-uniqueness of cleaned barcodes between patients
```{r}
# Fix cell ids
p01_076_barcodes$cell_id_fixed <- gsub("-1", "", p01_076_barcodes$cell_id) # remove the -1
p01_076_barcodes$patient_barcode_clean <- paste0("01-076_", p01_076_barcodes$cell_id_fixed) # add the patient as prefix. if you do not fix cell types like this you run the risk of overmatching due to NON-UNIQUENESS OF CELL BARCODES between patients
p01_076_barcodes$sample <- "01-076"

p01_115_barcodes$cell_id_fixed <- gsub("-1$", "", p01_115_barcodes$cell_id)
p01_115_barcodes$patient_barcode_clean <- paste0("01-115_", p01_115_barcodes$cell_id_fixed)
p01_115_barcodes$sample <- "01-115"

p01_131_barcodes$cell_id_fixed <- gsub("-1$", "", p01_131_barcodes$cell_id)
p01_131_barcodes$patient_barcode_clean <- paste0("01-131_", p01_131_barcodes$cell_id_fixed)
p01_131_barcodes$sample <- "01-131"

p01_163_barcodes$cell_id_fixed <- gsub("-1$", "", p01_163_barcodes$cell_id)
p01_163_barcodes$patient_barcode_clean <- paste0("01-163_", p01_163_barcodes$cell_id_fixed)
p01_163_barcodes$sample <- "01-163"

p01_190_barcodes$cell_id_fixed <- gsub("-1$", "", p01_190_barcodes$cell_id)
p01_190_barcodes$patient_barcode_clean <- paste0("01-190_", p01_190_barcodes$cell_id_fixed)
p01_190_barcodes$sample <- "01-190"

p04_003_barcodes$cell_id_fixed <- gsub("-1$", "", p04_003_barcodes$cell_id)
p04_003_barcodes$patient_barcode_clean <- paste0("04-003_", p04_003_barcodes$cell_id_fixed)
p04_003_barcodes$sample <- "04-003"

p04_006_barcodes$cell_id_fixed <- gsub("-1$", "", p04_006_barcodes$cell_id)
p04_006_barcodes$patient_barcode_clean <- paste0("04-006_", p04_006_barcodes$cell_id_fixed)
p04_006_barcodes$sample <- "04-006"

wm25_barcodes$cell_id_fixed <- gsub("-1$", "", wm25_barcodes$cell_id)
wm25_barcodes$patient_barcode_clean <- paste0("WM25_", wm25_barcodes$cell_id_fixed)
wm25_barcodes$sample <- "WM25"

wm43_barcodes$cell_id_fixed <- gsub("-1$", "", wm43_barcodes$cell_id)
wm43_barcodes$patient_barcode_clean <- paste0("WM43_", wm43_barcodes$cell_id_fixed)
wm43_barcodes$sample <- "WM43"

wm46_barcodes$cell_id_fixed <- gsub("-1$", "", wm46_barcodes$cell_id)
wm46_barcodes$patient_barcode_clean <- paste0("WM46_", wm46_barcodes$cell_id_fixed)
wm46_barcodes$sample <- "WM46"

wm47_barcodes$cell_id_fixed <- gsub("-1$", "", wm47_barcodes$cell_id)
wm47_barcodes$patient_barcode_clean <- paste0("WM47_", wm47_barcodes$cell_id_fixed)
wm47_barcodes$sample <- "WM47"

wm54_barcodes$cell_id_fixed <- gsub("-1$", "", wm54_barcodes$cell_id)
wm54_barcodes$patient_barcode_clean <- paste0("WM54_", wm54_barcodes$cell_id_fixed)
wm54_barcodes$sample <- "WM54"

wm65_barcodes$cell_id_fixed <- gsub("-1$", "", wm65_barcodes$cell_id)
wm65_barcodes$patient_barcode_clean <- paste0("WM65_", wm65_barcodes$cell_id_fixed)
wm65_barcodes$sample <- "WM65"

length(intersect(archR$patient_barcode_clean, p01_076_barcodes$patient_barcode_clean)) # they all match nice
length(intersect(archR$patient_barcode_clean, p01_115_barcodes$patient_barcode_clean)) 
length(intersect(archR$patient_barcode_clean, p01_131_barcodes$patient_barcode_clean)) 
length(intersect(archR$patient_barcode_clean, p01_163_barcodes$patient_barcode_clean)) 
length(intersect(archR$patient_barcode_clean, p01_190_barcodes$patient_barcode_clean)) 
length(intersect(archR$patient_barcode_clean, p04_003_barcodes$patient_barcode_clean))
length(intersect(archR$patient_barcode_clean, p04_006_barcodes$patient_barcode_clean))
length(intersect(archR$patient_barcode_clean, wm25_barcodes$patient_barcode_clean))
length(intersect(archR$patient_barcode_clean, wm43_barcodes$patient_barcode_clean))
length(intersect(archR$patient_barcode_clean, wm46_barcodes$patient_barcode_clean))
length(intersect(archR$patient_barcode_clean, wm47_barcodes$patient_barcode_clean))
length(intersect(archR$patient_barcode_clean, wm54_barcodes$patient_barcode_clean))
length(intersect(archR$patient_barcode_clean, wm65_barcodes$patient_barcode_clean))

# Merge into single df
all_barcodes <- rbind(p01_076_barcodes, p01_115_barcodes, p01_131_barcodes, p01_163_barcodes,p01_190_barcodes, p04_003_barcodes, p04_006_barcodes,
                      wm25_barcodes, wm43_barcodes, wm46_barcodes, wm47_barcodes, wm54_barcodes, wm65_barcodes)

all_barcodes
wm54_barcodes
```
# Identify clones in patients
```{r}
# 01-076
p01_076_archR$primaryClone <- NA
p01_076_archR$secondaryClone <- NA
p01_076_archR$tertiaryClone <- NA
p01_076.cells.with <- p01_076_barcodes$patient_barcode_clean[p01_076_barcodes$junction_aa == "CQQFDSSWTF"]
p01_076_archR$primaryClone[p01_076_archR$patient_barcode_clean %in% p01_076.cells.with] <- "CQQFDSSWTF"

p01_076.cells.with <- p01_076_barcodes$patient_barcode_clean[p01_076_barcodes$junction_aa == "CQAWDSNSHVF"]
p01_076_archR$secondaryClone[p01_076_archR$patient_barcode_clean %in% p01_076.cells.with] <- "CQAWDSNSHVF"

p01_076.cells.with <- p01_076_barcodes$patient_barcode_clean[p01_076_barcodes$junction_aa == "CSRNKRLGQPAGPFFDYW"]
p01_076_archR$tertiaryClone[p01_076_archR$patient_barcode_clean %in% p01_076.cells.with] <- "CSRNKRLGQPAGPFFDYW"

# 01-115
p01_115_archR$primaryClone <- NA
p01_115_archR$secondaryClone <- NA
p01_115_archR$tertiaryClone <- NA
temp <- c("CLQRGNWPLTF", "CQKYNSYFIF", "CCSYAGTWTWLF")
p01_115.cells.with <- p01_115_barcodes$patient_barcode_clean[p01_115_barcodes$junction_aa == "CLQRGNWPLTF"]
p01_115_archR$primaryClone[p01_115_archR$patient_barcode_clean %in% p01_115.cells.with] <- "CLQRGNWPLTF"

p01_115.cells.with <- p01_115_barcodes$patient_barcode_clean[p01_115_barcodes$junction_aa == "CCSYAGTWTWLF"]
p01_115_archR$secondaryClone[p01_115_archR$patient_barcode_clean %in% p01_115.cells.with] <- "CCSYAGTWTWLF"

p01_115.cells.with <- p01_115_barcodes$patient_barcode_clean[p01_115_barcodes$junction_aa == "CQKYNSYFIF"]
p01_115_archR$tertiaryClone[p01_115_archR$patient_barcode_clean %in% p01_115.cells.with] <- "CQKYNSYFIF"

# 01-131
p01_131_archR$primaryClone <- NA
p01_131_archR$secondaryClone <- NA
p01_131.cells.with <- p01_131_barcodes$patient_barcode_clean[p01_131_barcodes$junction_aa == "CQQYYSIPTF"]
p01_131_archR$primaryClone[p01_131_archR$patient_barcode_clean %in% p01_131.cells.with] <- "CQQYYSIPTF"

p01_131.cells.with <- p01_131_barcodes$patient_barcode_clean[p01_131_barcodes$junction_aa == "CNSRDSSGNHAVF"]
p01_131_archR$secondaryClone[p01_131_archR$patient_barcode_clean %in% p01_131.cells.with] <- "CNSRDSSGNHAVF"

# 01-163
p01_163_archR$primaryClone <- NA
p01_163_archR$secondaryClone <- NA
p01_163.cells.with <- p01_163_barcodes$patient_barcode_clean[p01_163_barcodes$junction_aa == "CQQYNNWPPLTF"]
p01_163_archR$primaryClone[p01_163_archR$patient_barcode_clean %in% p01_163.cells.with] <- "CQQYNNWPPLTF"

p01_163.cells.with <- p01_163_barcodes$patient_barcode_clean[p01_163_barcodes$junction_aa == "CLQRGNWPLTF"]
p01_163_archR$secondaryClone[p01_163_archR$patient_barcode_clean %in% p01_163.cells.with] <- "CLQRGNWPLTF"


# 01-190
p01_190_archR$primaryClone <- NA
p01_190_archR$secondaryClone <- NA
p01_190.cells.with <- p01_190_barcodes$patient_barcode_clean[p01_190_barcodes$junction_aa == "CQQYNNWPITF"]
p01_190_archR$primaryClone[p01_190_archR$patient_barcode_clean %in% p01_190.cells.with] <- "CQQYNNWPITF"

p01_190.cells.with <- p01_190_barcodes$patient_barcode_clean[p01_190_barcodes$junction_aa == "CQHYDNLPLTF"]
p01_190_archR$secondaryClone[p01_190_archR$patient_barcode_clean %in% p01_190.cells.with] <- "CQHYDNLPLTF"

# 04-003
p04_003_archR$primaryClone <- NA
p04_003_archR$secondaryClone <- NA
p04_003_archR$tertiaryClone <- NA
p04_003.cells.with <- p04_003_barcodes$patient_barcode_clean[p04_003_barcodes$junction_aa == "CQQYDSYPYTF"]
p04_003_archR$primaryClone[p04_003_archR$patient_barcode_clean %in% p04_003.cells.with] <- "CQQYDSYPYTF"

p04_003.cells.with <- p04_003_barcodes$patient_barcode_clean[p04_003_barcodes$junction_aa == "CQQTFDYRTF"]
p04_003_archR$secondaryClone[p04_003_archR$patient_barcode_clean %in% p04_003.cells.with] <- "CQQTFDYRTF"

p04_003.cells.with <- p04_003_barcodes$patient_barcode_clean[p04_003_barcodes$junction_aa == "CQAWDSNTVIF"]
p04_003_archR$tertiaryClone[p04_003_archR$patient_barcode_clean %in% p04_003.cells.with] <- "CQAWDSNTVIF"

# 04-006
p04_006_archR$primaryClone <- NA
p04_006_archR$secondaryClone <- NA
p04_006.cells.with <- p04_006_barcodes$patient_barcode_clean[p04_006_barcodes$junction_aa == "CQQYGTSSLTF"]
p04_006_archR$primaryClone[p04_006_archR$patient_barcode_clean %in% p04_006.cells.with] <- "CQQYGTSSLTF"

p04_006.cells.with <- p04_006_barcodes$patient_barcode_clean[p04_006_barcodes$junction_aa == "CQQYNNWPPLTF"]
p04_006_archR$secondaryClone[p04_006_archR$patient_barcode_clean %in% p04_006.cells.with] <- "CQQYNNWPPLTF"

# WM25
wm25_archR$primaryClone <- NA
wm25_archR$secondaryClone <- NA
wm25.cells.with <- wm25_barcodes$patient_barcode_clean[wm25_barcodes$junction_aa == "CMQALQTPRTF"]
wm25_archR$primaryClone[wm25_archR$patient_barcode_clean %in% wm25.cells.with] <- "CMQALQTPRTF"

wm25.cells.with <- wm25_barcodes$patient_barcode_clean[wm25_barcodes$junction_aa == "CQAWDSNTLVF"]
wm25_archR$secondaryClone[wm25_archR$patient_barcode_clean %in% wm25.cells.with] <- "CQAWDSNTLVF"

# wm43
wm43_archR$primaryClone <- NA
wm43_archR$secondaryClone <- NA
wm43_archR$tertiaryClone <- NA
wm43.cells.with <- wm43_barcodes$patient_barcode_clean[wm43_barcodes$junction_aa == "CQQYNTYPWTF"]
wm43_archR$primaryClone[wm43_archR$patient_barcode_clean %in% wm43.cells.with] <- "CQQYNTYPWTF"

wm43.cells.with <- wm43_barcodes$patient_barcode_clean[wm43_barcodes$junction_aa == "CQAWDSSAVLF"]
wm43_archR$secondaryClone[wm43_archR$patient_barcode_clean %in% wm43.cells.with] <- "CQAWDSSAVLF"

# wm46
wm46_archR$primaryClone <- NA
wm46_archR$secondaryClone <- NA
wm46_archR$tertiaryClone <- NA
wm46.cells.with <- wm46_barcodes$patient_barcode_clean[wm46_barcodes$junction_aa == "CQQYNNWPRTF"]
wm46_archR$primaryClone[wm46_archR$patient_barcode_clean %in% wm46.cells.with] <- "CQQYNNWPRTF"

wm46.cells.with <- wm46_barcodes$patient_barcode_clean[wm46_barcodes$junction_aa == "CQQYNSWPRTF"]
wm46_archR$secondaryClone[wm46_archR$patient_barcode_clean %in% wm46.cells.with] <- "CQQYNSWPRTF"

# wm46
wm47_archR$primaryClone <- NA
wm47_archR$secondaryClone <- NA
wm47_archR$tertiaryClone <- NA
wm47.cells.with <- wm47_barcodes$patient_barcode_clean[wm47_barcodes$junction_aa == "CQQYGSSSSYTF"]
wm47_archR$primaryClone[wm47_archR$patient_barcode_clean %in% wm47.cells.with] <- "CQQYGSSSSYTF"

wm47.cells.with <- wm47_barcodes$patient_barcode_clean[wm47_barcodes$junction_aa == "CHQGVF"]
wm47_archR$secondaryClone[wm47_archR$patient_barcode_clean %in% wm47.cells.with] <- "CHQGVF"

# Secondary for WM54
wm54_archR$primaryClone <- NA
wm54_archR$secondaryClone <- NA
wm54.cells.with <- wm54_barcodes$patient_barcode_clean[wm54_barcodes$junction_aa == "CQQCGDWPLTF"]
wm54_archR$primaryClone[wm54_archR$patient_barcode_clean %in% wm54.cells.with] <- "CQQCGDWPLTF"
wm54.cells.with <- wm54_barcodes$patient_barcode_clean[wm54_barcodes$junction_aa == "CQQYYSTPRTF"]
wm54_archR$secondaryClone[wm54_archR$patient_barcode_clean %in% wm54.cells.with] <- "CQQYYSTPRTF"

# wm65
wm65_archR$primaryClone <- NA
wm65_archR$secondaryClone <- NA
wm65_archR$tertiaryClone <- NA
wm65.cells.with <- wm65_barcodes$patient_barcode_clean[wm65_barcodes$junction_aa == "CQQYNSYPYTF"]
wm65_archR$primaryClone[wm65_archR$patient_barcode_clean %in% wm65.cells.with] <- "CQQYNSYPYTF"

wm65.cells.with <- wm65_barcodes$patient_barcode_clean[wm65_barcodes$junction_aa == "CQQYYSTPRTF"]
wm65_archR$secondaryClone[wm65_archR$patient_barcode_clean %in% wm65.cells.with] <- "CQQYYSTPRTF"
```
# Supp figure 3 - TRUST4 clonality reclusters
```{r}
cols <- c("gray", "violetred")
names(cols) <- names(table(p01_076_archR$hasTopCDR3))
pdf("/Users/gagled01/morganLab/Waldenstroms/scRNA_ATAC/figures/Supplementary/SuppFigure3/p01-076_Reclsuter_TRUST4_SuppFigure3.pdf", height = 6, width = 8)
p1 <- plotEmbedding(p01_076_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("01-076")
p1
dev.off()
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/scRNA_ATAC/figures/Supplementary/p01-076_Reclsuter_TRUST4_SuppFigure3.png", height = 6, width = 6)

names(cols) <- names(table(p01_115_archR$hasTopCDR3)) 
p2 <- plotEmbedding(p01_115_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("01-115") + NoLegend()

names(cols) <- names(table(p01_131_archR$hasTopCDR3)) 
p3 <- plotEmbedding(p01_131_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("01-131") + NoLegend()

names(cols) <- names(table(p01_163_archR$hasTopCDR3)) 
p4 <- plotEmbedding(p01_163_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("01-163") + NoLegend()

names(cols) <- names(table(p01_190_archR$hasTopCDR3)) 
p5 <- plotEmbedding(p01_190_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("01-190") + NoLegend()

names(cols) <- names(table(p04_003_archR$hasTopCDR3)) 
p6 <- plotEmbedding(p04_003_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("04-003") + NoLegend()

names(cols) <- names(table(p04_006_archR$hasTopCDR3)) 
p7 <- plotEmbedding(p04_006_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("04-006") + NoLegend()

names(cols) <- names(table(wm25_archR$hasTopCDR3)) 
p8 <- plotEmbedding(wm25_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("WM25") + NoLegend()

names(cols) <- names(table(wm43_archR$hasTopCDR3)) 
p9 <- plotEmbedding(wm43_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("WM43") + NoLegend()

names(cols) <- names(table(wm46_archR$hasTopCDR3)) 
p10 <- plotEmbedding(wm46_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("WM46") + NoLegend()

names(cols) <- names(table(wm47_archR$hasTopCDR3)) 
p11 <- plotEmbedding(wm47_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("WM47")+ NoLegend()

wm54_archR$trust4_anno <- wm54_archR$hasTopCDR3
wm54_archR$trust4_anno[wm54_archR$secondaryClone == "CQQYYSTPRTF"] <- "Secondary Clone"
cols <- c("gray", "slateblue", "violetred")
names(cols) <- names(table(wm54_archR$trust4_anno)) 
p12 <- plotEmbedding(wm54_archR, embedding = "Patient_Recluster_UMAP", name = "trust4_anno", pal = cols, labelMeans = F) + ggtitle("WM54") +  NoLegend()

cols <- c("gray", "violetred")
names(cols) <- names(table(wm65_archR$hasTopCDR3)) 
p13 <- plotEmbedding(wm65_archR, embedding = "Patient_Recluster_UMAP", name = "hasTopCDR3", pal = cols, labelMeans = F) + ggtitle("WM65") + NoLegend()

library(patchwork)
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/SuppFigure3_PatientRecluster_TRUST4_UMAPs.pdf", width = 16, height = 12)
(p1 + p2 + p3 + p4 + p5 + p6 + p7 + p8 + p9 + p10 + p11 + p12 + p13)
dev.off()
```
# Supp 3B - Reclusterd patients by celltype
```{r}
colors <- c("slateblue", "firebrick", "thistle2", "orange1", "lightseagreen", "forestgreen") # MBC-like, PClike, polyclonal MBC, polyclonal PC, clonal PC, WM43

cols <- c(colors[5], colors[1], colors[3], colors[4])
names(cols) <- names(table(p01_076_archR$final_annotation)) 
pdf("/Users/gagled01/morganLab/Waldenstroms/scRNA_ATAC/figures/Supplementary/SuppFigure3/p01-076_Reclsuter_CellType_SuppFigure3.pdf", height = 6, width = 8)
p1 <- plotEmbedding(p01_076_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("01-076")
p1
dev.off()

cols <- c(colors[5], colors[1], colors[2], colors[3], colors[4])
names(cols) <- names(table(p01_115_archR$final_annotation)) 
p2 <- plotEmbedding(p01_115_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("01-115") + NoLegend()

cols <- c(colors[5], colors[1], colors[2], colors[3], colors[4])
names(cols) <- names(table(p01_131_archR$final_annotation)) 
p3 <- plotEmbedding(p01_131_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("01-131") + NoLegend()

cols <- c(colors[5], colors[1], colors[2], colors[3])
names(cols) <- names(table(p01_163_archR$final_annotation)) 
p4 <- plotEmbedding(p01_163_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("01-163") + NoLegend()

cols <- c(colors[5], colors[1], colors[3], colors[4])
names(cols) <- names(table(p01_190_archR$final_annotation)) 
p5 <- plotEmbedding(p01_190_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("01-190") + NoLegend()

cols <- c(colors[5], colors[1], colors[2], colors[3], colors[4])
names(cols) <- names(table(p04_003_archR$final_annotation)) 
p6 <- plotEmbedding(p04_003_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("04-003") + NoLegend()

cols <- c(colors[5], colors[1], colors[2], colors[3], colors[4])
names(cols) <- names(table(p04_006_archR$final_annotation)) 
p7 <- plotEmbedding(p04_006_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("04-006") + NoLegend()

cols <- c(colors[5], colors[1], colors[2], colors[3], colors[4])
names(cols) <- names(table(wm25_archR$final_annotation)) 
p8 <- plotEmbedding(wm25_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("WM25") + NoLegend()

cols <- c(colors[1], colors[2], colors[6])
names(cols) <- names(table(wm43_archR$final_annotation)) 
p9 <- plotEmbedding(wm43_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("WM43") + NoLegend()

cols <- c(colors[5], colors[1], colors[3], colors[4])
names(cols) <- names(table(wm46_archR$final_annotation)) 
p10 <- plotEmbedding(wm46_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("WM46") + NoLegend()

cols <- c(colors[5], colors[1], colors[3], colors[4])
names(cols) <- names(table(wm47_archR$final_annotation)) 
p11 <- plotEmbedding(wm47_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("WM47") + NoLegend()

cols <- c(colors[5], colors[1], colors[2], colors[3], colors[4])
names(cols) <- names(table(wm54_archR$final_annotation)) 
p12 <- plotEmbedding(wm54_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("WM54") + NoLegend()

cols <- c(colors[5], colors[1], colors[3], colors[4])
names(cols) <- names(table(wm65_archR$final_annotation)) 
p13 <- plotEmbedding(wm65_archR, embedding = "Patient_Recluster_UMAP", name = "final_annotation", pal = cols, labelMeans = F) + ggtitle("WM65") + NoLegend()

pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/SuppFigure3_PatientRecluster_FinalAnnotation_UMAPs.pdf", width = 16, height = 12)
(p1 + p2 + p3 + p4 + p5 + p6 + p7 + p8 + p9 + p10 + p11 + p12 + p13)
dev.off()
```
# Supp figure 3 - part 2 - basic GEX - NOT USING!!!!
```{r}
genes <- c("JCHAIN", "XBP1", "PRDM1", "PAX5", "BACH2", "CD24")

p <- plotEmbedding( 
    ArchRProj = p01_076_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(p01_076_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})

p2
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/01-076_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# 01-115
p <- plotEmbedding( 
    ArchRProj = p01_115_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(p01_115_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/01-115_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# 01-131
p <- plotEmbedding( 
    ArchRProj = p01_131_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.9),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(p01_131_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/01-131_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# 01-163
p <- plotEmbedding( 
    ArchRProj = p01_163_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(p01_163_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/01-163_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# 01-190
p <- plotEmbedding( 
    ArchRProj = p01_190_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(p01_190_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/01-190_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# 04-003
p <- plotEmbedding( 
    ArchRProj = p04_003_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(p04_003_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/04-003_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# 04-006
p <- plotEmbedding( 
    ArchRProj = p04_006_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(p04_006_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/04-006_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# WM25
p <- plotEmbedding( 
    ArchRProj = wm25_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(wm25_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/WM25_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# WM43
p <- plotEmbedding( 
    ArchRProj = wm43_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(wm43_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/WM43_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# WM46
p <- plotEmbedding( 
    ArchRProj = wm46_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(wm46_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/WM46_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# WM47
p <- plotEmbedding( 
    ArchRProj = wm47_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    #quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(wm47_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/WM47_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# WM54
p <- plotEmbedding( 
    ArchRProj = wm54_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(wm54_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/WM54_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()

# WM65
p <- plotEmbedding( 
    ArchRProj = wm65_archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "Patient_Recluster_UMAP",
    quantCut = c(0.01, 0.99),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(wm65_archR)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank()
    )
})
pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/WM65_Reclustered_Genes.pdf", width = 24, height = 4)
do.call(cowplot::plot_grid, c(list(ncol = 6),p2))
dev.off()


pdf("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure3/Temp_GetLEgend.pdf", width = 4, height = 4)
p$JCHAIN
dev.off()

p2
```
# Supplementary Figure 4 - Batch Correction comparison
```{r}
anno_cols <- table(archR$final_annotation)
names(anno_cols) <- names(table(archR$final_annotation)) 
anno_cols[1] <- "lightseagreen"
anno_cols[2] <- "forestgreen"
anno_cols[3] <- "firebrick"
anno_cols[4] <- "slateblue"
anno_cols[5] <- "thistle2"
anno_cols[6] <- "orange1"
anno_cols[7] <- "darkmagenta"
anno_cols[8] <- "goldenrod"
anno_cols[9] <- "black"
anno_cols[10] <- "darkcyan"
  
colors_barplot

# Emulate R default palette
gg_color_hue <- function(n) {
  hues = seq(15, 375, length = n + 1)
  hcl(h = hues, l = 65, c = 100)[1:n]
}
patient_colors <- gg_color_hue(length(table(archR$samplesRefCombined)))

# Get trust4 colors
trust4_cols <- table(archR$hasTopCDR3)
names(trust4_cols) <- names(table(archR$hasTopCDR3)) 
trust4_cols[1] <- "gray"
trust4_cols[2] <- "violetred"

patient_umap <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesRefCombined", embedding = "UMAP_Harmony", pal = patient_colors, labelAsFactors = F, labelSize = 4, labelMeans = F) + ggtitle("") + NoLegend()
trust4_umap <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "hasTopCDR3", embedding = "UMAP_Harmony", pal = trust4_cols, labelMeans = F) + ggtitle("") + NoLegend()
celltype_umap <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "final_annotation", embedding = "UMAP_Harmony", pal = colors_barplot, labelAsFactors = F, labelMeans = F) + ggtitle("") + NoLegend()

# Now for non-batch corrected
patient_umap_noBC <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesRefCombined", embedding = "BLineage_NoProgen_Recluster_UMAP", pal = patient_colors, labelAsFactors = F, labelSize = 4, labelMeans = F) + ggtitle("") + NoLegend()
trust4_umap_noBC <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "hasTopCDR3", embedding = "BLineage_NoProgen_Recluster_UMAP", pal = trust4_cols, labelMeans = F) + ggtitle("") + NoLegend()
celltype_umap_noBC <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "final_annotation", embedding = "BLineage_NoProgen_Recluster_UMAP", pal = colors_barplot, labelAsFactors = F, labelMeans = F) + ggtitle("") + NoLegend()

(patient_umap + trust4_umap + celltype_umap) / (patient_umap_noBC + trust4_umap_noBC + celltype_umap_noBC)
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure4/SuppFigure4_BatchCorrect.pdf", height = 6, width = 9)
```
# Supplementary Figure 5 - GEX Feature Plots and violins used for Subtype annotation
```{r}
# Violin plots
plot_order <- c("MBC-like MBC", "PC-like MBC", "Clonal PC", "Polyclonal MBC", "Polyclonal PC", "WM43", "Reference MBC", "Reference PC", "Reference Naive", "Reference GC")

# Other section of SUpp Fig 8 - Zoom in on Subtypes
cellsKeep <- archR$cellNames[archR$final_annotation %in% c("MBC-like MBC", "PC-like MBC")]
illegal_archR <- archR[cellsKeep,]

zoom_cols <- table(illegal_archR$final_annotation)
names(zoom_cols) <- names(table(illegal_archR$final_annotation)) 
zoom_cols[1] <- "firebrick"
zoom_cols[2] <- "slateblue"

p <- plotGroups(
    ArchRProj = illegal_archR,
    groupBy = "final_annotation",
    colorBy = "GeneIntegrationMatrix",
    name = genes_mbc,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = zoom_cols 
   )


p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "12", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 2),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure5/SuppFigure5_MBCViolin_SubtypeZoomIn.pdf", height = 6, width = 6)

p <- plotGroups(
    ArchRProj = illegal_archR,
    groupBy = "final_annotation",
    colorBy = "GeneIntegrationMatrix",
    name = genes_pc,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = zoom_cols 
   )


p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "12", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 2),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure5/SuppFigure5_PCViolin_SubtypeZoomIn.pdf", height = 6, width = 6)
```
# Supplementary Figure 6 - WM43 zoom in
```{r}
wm43_genes <- c("CADM2", "SDK1", "PCDH9",# adhesion genes
                "BDNF", "NRG1", "ASTN2", # neural genes
                "SOX5", "RHEX", "FCHSD2", # shared with MBC-like
                "MSI2", "CPEB4", "PMAIP1") # shared with PC like


p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = wm43_genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = NULL
)

p_cow <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),  
        axis.title.y = element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 6),p_cow))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure6/SuppFigure6_WM43_ZoomIn_Long.pdf", height = 12, width = 9)

p <- plotGroups(
    ArchRProj = archR,
    groupBy = "final_annotation",
    colorBy = "GeneIntegrationMatrix",
    name = wm43_genes,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = colors_barplot 
   )


p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.15, 0.15, 0.15, 0.15), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "12", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 2),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure6/SuppFigure6_WM43_ZoomIn_Violins.pdf", height = 12, width = 9)

```
# Supp Table 4  - get BACH2-BLIMP1 ratios
```{r}
# TRYING WITHOUT 04-006
cellsKeep <- archR$cellNames[archR$samplesClean %in% c("01-115", "01-131", "01-163", "04-003", "04-006", "WM25", "WM54")]
#cellsKeep <- archR$cellNames[archR$final_annotation %in% c("WM43")]

illegal_archR <- archR[cellsKeep,]

mbc.matrix <- getMatrixFromProject(
  ArchRProj = illegal_archR,
  useMatrix = "GeneIntegrationMatrix",
  useSeqnames = NULL,
  verbose = TRUE,
  binarize = FALSE,
  threads = getArchRThreads(),
  logFile = createLogFile("getMatrixFromProject")
)

geneintedata=assay(mbc.matrix)
genedata=rowData(mbc.matrix)
geneinteinfo=cbind(genedata, geneintedata)
geneinteinfo2=as.data.frame(t(as.data.frame(geneinteinfo[,7:length(colnames(geneinteinfo))])))
colnames(geneinteinfo2)=genedata$name
rownames(geneinteinfo2)=colnames(geneinteinfo)[7:length(colnames(geneinteinfo))]

mean.bach2 <- mean(geneinteinfo2$BACH2)
mean.blimp1 <- mean(geneinteinfo2$PRDM1)
ratio <- mean.blimp1/mean.bach2
mean.blimp1
mean.bach2
ratio


STOP
rm(geneintedata, genedata, geneinteinfo, geneinteinfo2, mbc.matrix, illegal_archR)
```
# Supp figure 8 - GC motif enrichment
```{r}
# Motifs
`%ni%` <- Negate(`%in%`)

genes <- c("POU2F2", "POU2F1", "LMO2", "TCF3", "TCF4", "ID3", "BACH2", "IRF8", "FOXO1")
#prePlasmaBlast_genes <- c("FOS", "FOSB", "FOLS1", "FOSL2", "JUN", "JUNB", "JUND", "BATF", "BATF3")
#prePlasmaBlast_genes <- c("SPI1", "SPIB", "BCL11A", "BCL11B")
motif.labels <- getFeatures(archR, select = paste(genes, collapse="|"), useMatrix = "MotifMatrix")
motif.labels <- grep("z:", motif.labels, value = TRUE)
motif.labels <- motif.labels[motif.labels %ni% c("z:ARID3C_10", "z:ARID3B_8", "z:ARID3A_6")]

motif.labels <- c(motif.labels[3], motif.labels[4], motif.labels[1],
                  motif.labels[9], motif.labels[7], motif.labels[8],
                  motif.labels[6], motif.labels[2], motif.labels[5])

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "MotifMatrix", 
    name = motif.labels, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$horizonExtra,
    #scale_fill_continuous(name = "DevScore"),
    imputeWeights = getImputeWeights(archR)
)

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x + 
    #guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        #axis.title.y = element_blank()
    ) #+ labs(fill = "DevScore")
})

do.call(cowplot::plot_grid, c(list(ncol = 3),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure8/SupplementaryFigureX_PClike_GC_Motif_UMAPs.pdf", height = 9, width = 12)

# Get clean labels for plots
#clean.violin.labels <- gsub(".*:", "", motif.labels)
#clean.violin.labels <- gsub("_.*", "", clean.violin.labels)

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "MotifMatrix", 
    name = motif.labels,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = colors_barplot
   )

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure8/SuppFigure8_PClike_GCmotifs_Violins.pdf", height = 6, width = 7)

```
# Supplementary Figure 9 - clonal PC vs ref PC
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "Clonal PC",
  bgdGroups = c("Reference PC")
)

markerList <- getMarkers(markerTest, cutOff = "FDR < 100")

# Create df of both upreg and downreg and generate CSVs for reference
df.motifs <- as.data.frame(markerList$`Clonal PC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1
motifs.up <- df.motifs[df.motifs$Log2FC > 0,]
motifs.do <- df.motifs[df.motifs$Log2FC < (-0),]

# Create combined df and cut by significance
combined.df <- rbind(motifs.up, motifs.do)
combined.df$category <- NA
combined.df$category[combined.df$Log2FC > 0.5 & combined.df$FDR < 0.000000005] <- "POS" # annotate for volcano downstream
combined.df$category[combined.df$Log2FC < (-0.5) & combined.df$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df[combined.df$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_RefPC_FDR-Sorted_DifferentialTest_GEX_Up.csv")
pc.upreg.df <- na.omit(combined.df[combined.df$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_RefC_FDR-Sorted_DifferentialTest_GEX_Down.csv")

sorted.up # view to extract genes for volcano plot
sorted.do

# Plot
`%ni%` <- Negate(`%in%`)

up.labels <- c("LYN", "BANK1", "ZNF804A", "PRKCB", "PTPRJ",  "MEF2A", "EBF1", "CDK14", "ITPR2", "FOXP1", "PLCG2", "CARMIL1", "ZBTB20", "OSBPL10", "TRIO", "BCL2", "PLCG2", "CD79B", "PMAIP1", "PIK3CA", "RALGPS2", "SMCHD1", "SOX5", "TET2", "MAP3K1", "DUSP22", "NFKB1", "RRAS2", "MS4A1", "AFF3", "ZEB1", "REL", "TRAF3", "TCF4", "TCF12", "IL15", "FOXO1", "ETV6", "CD37", "PRDM5", "ZEB2", "HIVEP1")

do.labels <- c("PAG1", "FTL", "FOS", "PRDM1", "TPT1", "SLAMF7", "BCL7A", "XBP1", "SSR4", "BHLHE41", "SELENOK", "ZBTB38", "FOSB", "MZB1", "SAR1B", "BTG2", "TP53INP1", "SEPT1", "SEPT7", "BTG2", "FOSB", "SAR1B", "SSR3", "SYTL1", "MCL1", "MME", "LMAN1", "MUM1", "JUN", "CD63", "PIM2", "SORL1")

comb.labels <- c(up.labels, do.labels)

combined.df$labels <- combined.df$name
combined.df$labels[combined.df$labels %ni% comb.labels] <- ""
table(combined.df$labels)

ggplot(combined.df, aes(Log2FC, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.8, y = 0.1, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="lightseagreen", "NEG"="black", "NONE"="grey")) +
  xlab("Log2FC") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-5, 5)) + scale_y_continuous(limits = c(0, 100))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure9/SuppFigure9_MonoclonalPC-vs_RefPC_GEX_Volcano.pdf", height = 9, width = 8)

# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "Clonal PC",
  bgdGroups = c("Reference PC")
)

# Up in MBC-like
markerList.motifs <- getMarkers(markerTest, cutOff = "FDR <= 100")
df.motifs <- as.data.frame(markerList.motifs$`Clonal PC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1
#df.motifs$log2MeanDiff <- log2(df.motifs$MeanDiff)

motifs.up <- df.motifs[df.motifs$MeanDiff > 0,]
motifs.do <- df.motifs[df.motifs$MeanDiff < 0,]
sorted.motifs.up <- motifs.up[order(motifs.up$FDR),]
sorted.motifs.do <- motifs.do[order(motifs.do$FDR),]

combined.df.motifs <- rbind(sorted.motifs.up, sorted.motifs.do)
combined.df.motifs$category <- NA
combined.df.motifs$category[combined.df.motifs$MeanDiff > 0.025 & combined.df.motifs$FDR < 0.000000005] <- "POS"
combined.df.motifs$category[combined.df.motifs$MeanDiff < (-0.025) & combined.df.motifs$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_RefPC_FDR-Sorted_DifferentialTest_Motif_Up.csv")
pc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_RefC_FDR-Sorted_DifferentialTest_Motif_Down.csv")

motifs.labels.up <- sorted.motifs.up$name[1:30]
markerMotifs.labels.up <- getFeatures(archR, select = paste(motifs.labels.up, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.up <- grep("z:", markerMotifs.labels.up, value = TRUE)
#markerMotifs.labels.up <- markerMotifs.labels.up[markerMotifs.labels.up %ni% c("")]

motifs.labels.do <- sorted.motifs.do$name[1:30]
markerMotifs.labels.do <- getFeatures(archR, select = paste(motifs.labels.do, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.do <- grep("z:", markerMotifs.labels.do, value = TRUE)

comb.labels.motifs <- c(motifs.labels.up, motifs.labels.do)
comb.labels.motifs.sub <- comb.labels.motifs[comb.labels.motifs %ni% c("KLF15_223", "ZBTB7A_258", "ENSG00000187728_92", "ENSG00000250096_734", "PURA_813", "RUNX1_733")]

combined.df.motifs$labels <- combined.df.motifs$name
combined.df.motifs$labels[combined.df.motifs$labels %ni% comb.labels.motifs.sub] <- ""
combined.df.motifs$labels <- gsub("_*[^_]*$", "", combined.df.motifs$labels)

# Plot
ggplot(combined.df.motifs, aes(MeanDiff, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1.5,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.08, y = 0.001, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="darkcyan", "NEG"="black", "NONE"="grey")) +
  xlab("MeanDiff") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-0.4, 0.4)) + scale_y_continuous(limits = c(0, 145))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure9/SuppFigure9_MonoclonalPC_vs_RefPCs_MOTIF_Volcano.pdf", height = 9, width = 8)
```
Supplementary Figure 9 - GSEA
```{r}
gsea1 <- read.table("/Users/gagled01/morganLab/Waldenstroms/GSEA/ClonalPC_vs_RefPC_GSEA_UpClonal.tsv", sep = "\t", header = T)
gsea2 <- read.table("/Users/gagled01/morganLab/Waldenstroms/GSEA/ClonalPC_vs_RefPC_GSEA_UpRef.tsv", sep = "\t", header = T)

# Create plotting dfs
plotting.df <- data.frame(gsea1$NAME, gsea1$NES, gsea1$FDR.q.val, gsea1$SIZE)
plotting.df <- head(plotting.df, n = 30)
colnames(plotting.df) <- c("Pathway", "NES", "FDR.q.val", "Count")
plotting.df$Pathway <- gsub("HALLMARK_", "", plotting.df$Pathway)
plotting.df$log10FDR.qval <- -log10(plotting.df$FDR.q.val)

plotting.df2 <- data.frame(gsea2$NAME, gsea2$NES, gsea2$FDR.q.val, gsea2$SIZE)
plotting.df2 <- head(plotting.df2, n = 20)
colnames(plotting.df2) <- c("Pathway", "NES", "FDR.q.val", "Count")
#plotting.df2$FDR.q.val[1:2] <- 0.0001 # Manually giving some v low Q values because 0 does not work for plotting
plotting.df2$Pathway <- gsub("HALLMARK_", "", plotting.df2$Pathway)
plotting.df2$log10FDR.qval <- -log10(plotting.df2$FDR.q.val)

# Plot by regular FDR
# regular FDR
ggplot(plotting.df, aes(x=NES, y=reorder(Pathway,-FDR.q.val), size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("Clonal PC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure9/ClonalPC_vs_RefPC_GSEA_Dotplot_upClonal_FDRqval.pdf", height = 8, width = 8)

ggplot(plotting.df2, aes(x=NES, y=reorder(Pathway,-FDR.q.val),size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("Reference PC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure9/ClonalPC_vs_RefPC_GSEA_Dotplot_upReference_FDRqval.pdf", height = 8, width = 8)
```
# Supplementary Figure 10 - clonal PC vs. PC-like MBC
# GEX Volcano 
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "Clonal PC",
  bgdGroups = c("PC-like MBC")
)

markerList <- getMarkers(markerTest, cutOff = "FDR < 100")
df.motifs <- as.data.frame(markerList$`Clonal PC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1
motifs.up <- df.motifs[df.motifs$Log2FC > 0,]
motifs.do <- df.motifs[df.motifs$Log2FC < (0),]
sorted.up <- motifs.up[order(motifs.up$FDR),]
sorted.do <- motifs.do[order(motifs.do$FDR),]

sorted.up
sorted.do

# Create combined df and cut by significance
combined.df <- rbind(motifs.up, motifs.do)
combined.df$category <- NA
combined.df$category[combined.df$Log2FC > 0.5 & combined.df$FDR < 0.000000005] <- "POS" # annotate for volcano downstream
combined.df$category[combined.df$Log2FC < (-0.5) & combined.df$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df[combined.df$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PC-likeMBC_FDR-Sorted_DifferentialTest_GEX_UpClonalPC.csv")
pc.upreg.df <- na.omit(combined.df[combined.df$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PC-likeMBC_FDR-Sorted_DifferentialTest_GEX_UpPClikeMBC.csv")


`%ni%` <- Negate(`%in%`)

up.labels <- c("SSR4", "SSR3", "DERL3", "PRDX4", "FKBP11", "MZB1", "SEC11C", "MYDGF", "XBP1", "PIK3CG", "SSR1", "CD63", "TNFRSF17", "LMAN2", "CDK6", "LMAN1", "SELENOS", "PRDM1", "FCRL5", "SSR2", "BLNK", "CD38", "SDC1", "GAB1")
do.labels <- c("CD83", "AFF3", "REL", "PRDM2", "CD69", "CD37", "MGAT5", "CD74", "EZR", "CD52", "BANK1", "LYN", "SIK3", "ARID1B", "RRAS2", "BACH2", "PLCG2", "NFKBID",  "NFKB1", "ZEB1", "CXCR5", "JAK1", "BCL11A", "PAX5", "VAV3", "TRAF3", "TRAF5", "CD44", "ETV6", "RELB", "TCF4", "STAT3")

comb.labels <- c(up.labels, do.labels)

combined.df$labels <- combined.df$name
combined.df$labels[combined.df$labels %ni% comb.labels] <- ""
table(combined.df$labels)

ggplot(combined.df, aes(Log2FC, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.8, y = 0.1, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="indianred2", "NEG"="skyblue", "NONE"="grey")) +
  xlab("Log2FC") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-6, 6)) + scale_y_continuous(limits = c(0, 140))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure10/SuppFigure10-ClonalPC-vs_PClike_GEX_Volcano.pdf", height = 9, width = 8)

# Motifs

markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "Clonal PC",
  bgdGroups = c("PC-like MBC")
)

# Up in MBC-like
markerList.motifs <- getMarkers(markerTest, cutOff = "FDR <= 100")
df.motifs <- as.data.frame(markerList.motifs$`Clonal PC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1
#df.motifs$log2MeanDiff <- log2(df.motifs$MeanDiff)

motifs.up <- df.motifs[df.motifs$MeanDiff > 0,]
motifs.do <- df.motifs[df.motifs$MeanDiff < 0,]
sorted.motifs.up <- motifs.up[order(motifs.up$FDR),]
sorted.motifs.do <- motifs.do[order(motifs.do$FDR),]

combined.df.motifs <- rbind(sorted.motifs.up, sorted.motifs.do)
combined.df.motifs$category <- NA
combined.df.motifs$category[combined.df.motifs$MeanDiff > 0.025 & combined.df.motifs$FDR < 0.000000005] <- "POS"
combined.df.motifs$category[combined.df.motifs$MeanDiff < (-0.025) & combined.df.motifs$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PC-likeMBC_FDR-Sorted_DifferentialTest_MOTIFS_UpClonalPC.csv")
pc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PC-likeMBC_FDR-Sorted_DifferentialTest_MOTIFS_UpPClikeMBC.csv")

motifs.labels.up <- combined.df.motifs$name[combined.df.motifs$category == "POS"]
markerMotifs.labels.up <- getFeatures(archR, select = paste(motifs.labels.up, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.up <- grep("z:", markerMotifs.labels.up, value = TRUE)
#markerMotifs.labels.up <- markerMotifs.labels.up[markerMotifs.labels.up %ni% c("")]

motifs.labels.do <- combined.df.motifs$name[combined.df.motifs$category == "NEG"]
markerMotifs.labels.do <- getFeatures(archR, select = paste(motifs.labels.do, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.do <- grep("z:", markerMotifs.labels.do, value = TRUE)

comb.labels.motifs <- c(motifs.labels.up, motifs.labels.do)
comb.labels.motifs.sub <- comb.labels.motifs[comb.labels.motifs %ni% c("ENSG00000187728_92", "NEUROD6_821",  "NEUROD4_820", "NEUROG2_81", "NEUROG3_43", "C11orf9_654")]

combined.df.motifs$labels <- combined.df.motifs$name
combined.df.motifs$labels[combined.df.motifs$labels %ni% comb.labels.motifs.sub] <- ""
combined.df.motifs$labels <- gsub("_*[^_]*$", "", combined.df.motifs$labels)

# Plot
ggplot(combined.df.motifs, aes(MeanDiff, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1.5,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.08, y = 0.001, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="olivedrab", "NEG"="magenta3", "NONE"="grey")) +
  xlab("MeanDiff") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-0.45, 0.45)) + scale_y_continuous(limits = c(0, 45))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure10/SuppFig10_ClonalPC_vs_PClikeMBC_MOTIF_Volcano.pdf", height = 9, width = 8)
```
# Supp Figure 10 part 2 - Clonal PC vs PC-like MBC GSEA plots
```{r}
gsea1 <- read.table("/Users/gagled01/morganLab/Waldenstroms/GSEA/ClonalPC_vs_PClikeMBC_upClonalPC.tsv", sep = "\t", header = T)
gsea2 <- read.table("/Users/gagled01/morganLab/Waldenstroms/GSEA/ClonalPC_vs_PClikeMBC_upPClikeMBC.tsv", sep = "\t", header = T)

# Create plotting dfs
plotting.df <- data.frame(gsea1$NAME, gsea1$NES, gsea1$FDR.q.val, gsea1$SIZE)
plotting.df <- head(plotting.df, n = 30)
colnames(plotting.df) <- c("Pathway", "NES", "FDR.q.val", "Count")
plotting.df$Pathway <- gsub("HALLMARK_", "", plotting.df$Pathway)
plotting.df$log10FDR.qval <- -log10(plotting.df$FDR.q.val)

plotting.df2 <- data.frame(gsea2$NAME, gsea2$NES, gsea2$FDR.q.val, gsea2$SIZE)
plotting.df2 <- head(plotting.df2, n = 20)
colnames(plotting.df2) <- c("Pathway", "NES", "FDR.q.val", "Count")
#plotting.df2$FDR.q.val[1:2] <- 0.0001 # Manually giving some v low Q values because 0 does not work for plotting
plotting.df2$Pathway <- gsub("HALLMARK_", "", plotting.df2$Pathway)
plotting.df2$log10FDR.qval <- -log10(plotting.df2$FDR.q.val)

# Plot by regular FDR
# regular FDR
ggplot(plotting.df, aes(x=NES, y=reorder(Pathway,-FDR.q.val), size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("PC-like MBC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure10/PC-like_vs_ClonalPC_GSEA_Dotplot_upPClikeMBC_FDRqval.pdf", height = 8, width = 8)

ggplot(plotting.df2, aes(x=NES, y=reorder(Pathway,-FDR.q.val),size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("Clonal PC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure10/PC-like_vs_ClonalPC_GSEA_Dotplot_upClonalPC_FDRqval.pdf", height = 8, width = 8)
```
# Supplementary Figure 11 - clonal PC vs polyclonal PC - PT 1 GEX VOLCANO
```{r}
# Get DEG
markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "Clonal PC",
  bgdGroups = c("Polyclonal PC")
)

markerList <- getMarkers(markerTest, cutOff = "FDR < 100")
df.motifs <- as.data.frame(markerList$`Clonal PC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1
motifs.up <- df.motifs[df.motifs$Log2FC > 0,]
motifs.do <- df.motifs[df.motifs$Log2FC < (0),]
sorted.up <- motifs.up[order(motifs.up$FDR),]
sorted.do <- motifs.do[order(motifs.do$FDR),]

sorted.up
sorted.do

# Create combined df and cut by significance
combined.df <- rbind(motifs.up, motifs.do)
combined.df$category <- NA
combined.df$category[combined.df$Log2FC > 0.5 & combined.df$FDR < 0.000000005] <- "POS" # annotate for volcano downstream
combined.df$category[combined.df$Log2FC < (-0.5) & combined.df$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df[combined.df$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PolyclonalPC_FDR-Sorted_DifferentialTest_GEX_UpClonal.csv")
pc.upreg.df <- na.omit(combined.df[combined.df$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PolyclonalPC_FDR-Sorted_DifferentialTest_GEX_UpPolyclonal.csv")


`%ni%` <- Negate(`%in%`)

up.labels <- c("DUSP22", "ITPR2", "CDK14", "FOXP1", "ZNF804A", "CD79B", "MEF2A", "CAM2KD", "PTPJR", "HDAC9", "MS4A1", "PRDM2", "LYN", "BANK1", "PLCG2", "OSBPL10", "PIK3CA", "IL15", "BCL2", "RRAS2", "TCF12", "CD9", "ZEB1", "NFKB1")
do.labels <- c("PAG1", "SLAMf7", "HIST1H1D", "FOS", "PRDM1", "JUN", "SAT1", "SMIM14", "SORl1", "DUSP5", "SETP6", "SSR4", "IKZF1", "BCL7A", "LMO2", "NEAT1", "SELENOK")

comb.labels <- c(up.labels, do.labels)

combined.df$labels <- combined.df$name
combined.df$labels[combined.df$labels %ni% comb.labels] <- ""
table(combined.df$labels)

ggplot(combined.df, aes(Log2FC, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.8, y = 0.1, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="plum3", "NEG"="skyblue3", "NONE"="grey")) +
  xlab("Log2FC") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-6, 6)) + scale_y_continuous(limits = c(0, 100))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure11/SuppFigure11-ClonalPC_vs_PolyclonalPC_GEX_Volcano.pdf", height = 9, width = 8)

# Motifs

markerTest <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "MotifMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "Clonal PC",
  bgdGroups = c("Polyclonal PC")
)

# Up in MBC-like
markerList.motifs <- getMarkers(markerTest, cutOff = "FDR <= 100")
df.motifs <- as.data.frame(markerList.motifs$`Clonal PC`)
df.motifs$log10.fdr <- log10(df.motifs$FDR) * -1
#df.motifs$log2MeanDiff <- log2(df.motifs$MeanDiff)

motifs.up <- df.motifs[df.motifs$MeanDiff > 0,]
motifs.do <- df.motifs[df.motifs$MeanDiff < 0,]
sorted.motifs.up <- motifs.up[order(motifs.up$FDR),]
sorted.motifs.do <- motifs.do[order(motifs.do$FDR),]

combined.df.motifs <- rbind(sorted.motifs.up, sorted.motifs.do)
combined.df.motifs$category <- NA
combined.df.motifs$category[combined.df.motifs$MeanDiff > 0.025 & combined.df.motifs$FDR < 0.000000005] <- "POS"
combined.df.motifs$category[combined.df.motifs$MeanDiff < (-0.025) & combined.df.motifs$FDR < 0.000000005] <- "NEG"
mbc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "POS",])
write.csv(mbc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PolyclonalPC_FDR-Sorted_DifferentialTest_MOTIFS_UpClonal.csv")
pc.upreg.df <- na.omit(combined.df.motifs[combined.df.motifs$category == "NEG",])
write.csv(pc.upreg.df, "/Users/gagled01/morganLab/Waldenstroms/SupplementaryData/differential_tests/ClonalPC_vs_PolyclonalPC_FDR-Sorted_DifferentialTest_MOTIFS_UpPolyclonal.csv")

motifs.labels.up <- combined.df.motifs$name[combined.df.motifs$category == "POS"]
markerMotifs.labels.up <- getFeatures(archR, select = paste(motifs.labels.up, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.up <- grep("z:", markerMotifs.labels.up, value = TRUE)
#markerMotifs.labels.up <- markerMotifs.labels.up[markerMotifs.labels.up %ni% c("")]

motifs.labels.do <- combined.df.motifs$name[combined.df.motifs$category == "NEG"]
markerMotifs.labels.do <- getFeatures(archR, select = paste(motifs.labels.do, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs.labels.do <- grep("z:", markerMotifs.labels.do, value = TRUE)

comb.labels.motifs <- c(motifs.labels.up, motifs.labels.do)
comb.labels.motifs.sub <- comb.labels.motifs[comb.labels.motifs %ni% c("ENSG00000187728_92", "NEUROD6_821",  "NEUROD4_820", "NEUROG2_81", "NEUROG3_43", "C11orf9_654",
                                                                       "FOXN2_854", "FOXL2_855", "FOXK2_360", "FOXJ3_378", "FOXN3_829", "NFE2L2_115", "FOXP4_358", "FOXP2_356", "STAT5B_779", "IRF6_628", "IRF7_635")]

combined.df.motifs$labels <- combined.df.motifs$name
combined.df.motifs$labels[combined.df.motifs$labels %ni% comb.labels.motifs.sub] <- ""
combined.df.motifs$labels <- gsub("_*[^_]*$", "", combined.df.motifs$labels)

# Plot
ggplot(combined.df.motifs, aes(MeanDiff, log10.fdr, color = category, label = labels)) +
  geom_point() + 
  #geom_text(hjust = 0, vjust = 0) +
  geom_label_repel(box.padding = 1,
                   min.segment.length = 1.5,
                   max.overlaps = Inf,
                   position = position_nudge_center(x = 0.08, y = 0.001, center_x = 0, direction = "split")) +
   theme_ArchR() + theme(text = element_text(size = 15),
                         axis.title.x = element_text(size = 15, face = "bold"),
                         axis.title.y = element_text(size = 15, face = "bold")) +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("POS"="violetred", "NEG"="darkcyan", "NONE"="grey")) +
  xlab("MeanDiff") +
  ylab("-Log10(FDR)") + scale_x_continuous(limits = c(-0.45, 0.45)) + scale_y_continuous(limits = c(0, 60))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure11/SuppFig11_ClonalPC_vs_PolyclonalPC_MOTIF_Volcano.pdf", height = 9, width = 8)
```
# Supp Figure 11 GSEA
```{r}
gsea1 <- read.table("/Users/gagled01/morganLab/Waldenstroms/GSEA/ClonalPC_vs_PolyclonalPC_GSEA_UpClonal.tsv", sep = "\t", header = T)
gsea2 <- read.table("/Users/gagled01/morganLab/Waldenstroms/GSEA/ClonalPC_vs_PolyclonalPC_GSEA_UpPolyclonal.tsv", sep = "\t", header = T)

# Create plotting dfs
plotting.df <- data.frame(gsea1$NAME, gsea1$NES, gsea1$FDR.q.val, gsea1$SIZE)
plotting.df <- head(plotting.df, n = 30)
colnames(plotting.df) <- c("Pathway", "NES", "FDR.q.val", "Count")
plotting.df$Pathway <- gsub("HALLMARK_", "", plotting.df$Pathway)
plotting.df$log10FDR.qval <- -log10(plotting.df$FDR.q.val)

plotting.df2 <- data.frame(gsea2$NAME, gsea2$NES, gsea2$FDR.q.val, gsea2$SIZE)
plotting.df2 <- head(plotting.df2, n = 20)
colnames(plotting.df2) <- c("Pathway", "NES", "FDR.q.val", "Count")
#plotting.df2$FDR.q.val[1:2] <- 0.0001 # Manually giving some v low Q values because 0 does not work for plotting
plotting.df2$Pathway <- gsub("HALLMARK_", "", plotting.df2$Pathway)
plotting.df2$log10FDR.qval <- -log10(plotting.df2$FDR.q.val)

# Plot by regular FDR
# regular FDR
ggplot(plotting.df, aes(x=NES, y=reorder(Pathway,-FDR.q.val), size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("Clonal PC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure11/ClonalPC_vs_PolyclonalPC_GSEA_Dotplot_upClonal_FDRqval.pdf", height = 8, width = 8)

ggplot(plotting.df2, aes(x=NES, y=reorder(Pathway,-FDR.q.val),size = Size)) + geom_point(aes(size=Count, color = FDR.q.val)) + scale_color_gradient(low = "blue", high = "red", name = "FDR q-value") +
  ylab("") + ggtitle("Polyclonal PC Enriched Pathways") + theme_bw() + theme(legend.position = "right") 
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure11/ClonalPC_vs_Polyclonal_GSEA_Dotplot_upPolyclonal_FDRqval.pdf", height = 8, width = 8)
```
# Supp figure 12 - tumor mutational burden - found in WM_SNV_indels.Rmd
# Suppementary Figure 13 - Cell cycle genes
```{r}
genes <- c("MKI67", "CDK1", "TUBB", "NOTCH2", 
           "HMGB2","UBE2C", "CENPE", "CCNB2", 
           "E2F8", "MYO1E", "TOP2A", "TP53")

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(archR)
)

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x + 
    #guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure12/CellCycleGenes_UMAPs.pdf", height = 9, width = 16)

### and VIOLINS
cols <- table(archR$final_annotation)
names(cols) <- plot_order
cols[1] <- "firebrick"
cols[2] <- "slateblue"
cols[3] <- "lightseagreen"
cols[4] <- "thistle2"
cols[5] <- "orange1"
cols[6] <- "forestgreen"
cols[7] <- "goldenrod"
cols[8] <- "darkcyan"
cols[9] <- "black"
cols[10] <- "darkmagenta"

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "GeneIntegrationMatrix", 
    name = genes,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = colors_barplot
   )

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "10", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 4),p2))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure12/CellCycleGenes_Violins.pdf", height = 8, width = 12)
```
# Supplementary Figure 14 - NFkB signaling in B-cell differentiation
```{r}
nfkb_genes <- c("REL", "RELA", "RELB", "NFKB1", 
                "NFKB2", "NFKBIA", "NFKBIE", "CD40",  
           "TNFSF13B", "TNFRSF13C", "MAP3K14", "IKBKG")

nfkb_asc_model <- c("REL", "RELA", "PAX5", "BACH2", "BCL6", "PRDM1", "IRF4", "AICDA")
myd88_signaling <- c("HCK", "BTK", "PELI1", 
                      "IL6ST",  "PLCG2", 
                     "SYK", "CD79A", "CD79B", "LYN", "BLNK", "VAV3", "AFF3")

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = nfkb_genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(archR)
)

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x + 
    #guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 3),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure13/NFkB_GEX_UMAPs.pdf", height = 9, width = 16)

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "GeneIntegrationMatrix", 
    name = nfkb_genes,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = cols
   )

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "10", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure13/NFkB_Violins.pdf", height = 11, width = 9)
```
# Supp Figure 15 - BCR signaling genes
```{r}
genes <- c("CD79A", "CD79B", "LYN", "SYK", # CD79 BCR complex
           "VAV3","PLCG2", "BTK", "BLNK", # PLCG BCR signaling
           "CD19", "RAC1", "PIK3R1", "PIK3AP1", #  PI3K BCR signaling
           "PRKCB", "GAB1", "PDK1", "GRB2", # some other shit
           "TRAF6", "IRAK1", "IRAK4", "MYD88") # yep

# "rotating" it for the plot configuration
genes <- c("CD79A", "VAV3", "CD19", "PRKCB",
           "TRAF6", "CD79B", "PLCG2", "RAC1",
           "GAB1", "IRAK1", "LYN", "BTK",
           "PIK3R1", "PDK1", "IRAK4", "SYK", 
           "BLNK", "PIK3AP1", "GRB2", "MYD88")

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(archR)
)

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x + 
    #guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure14/BCRGenes_UMAPs.pdf", height = 20, width = 16)

### and VIOLINS
cols <- table(archR$final_annotation)
names(cols) <- plot_order
cols[1] <- "firebrick"
cols[2] <- "slateblue"
cols[3] <- "lightseagreen"
cols[4] <- "thistle2"
cols[5] <- "orange1"
cols[6] <- "forestgreen"
cols[7] <- "goldenrod"
cols[8] <- "darkcyan"
cols[9] <- "black"
cols[10] <- "darkmagenta"

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "GeneIntegrationMatrix", 
    name = genes,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = cols
   )

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "14", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 4),p2))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure14/BCRGenes_Violins.pdf", height = 14, width = 12)
```
# Supp figure 16 - inflammation/interferon/jak/stat signaling
```{r}
genes <- c("IFNGR2", "IFNG-AS1", "HIF1A", "SP110", # CD79 BCR complex
           "CREB1", "MAPK8", "MAPK1", "MAP3K1", # PLCG BCR signaling
           "IL15", "IL17RA", "IL6ST", "IL10RA",
            "STAT3", "STAT6", "JAK1", "IRF1") # yep

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(archR)
)
p

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x + 
    #guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure15/InterferonInflammation_UMAPs.pdf", height = 16, width = 16)

### and VIOLINS
cols <- table(archR$final_annotation)
names(cols) <- plot_order
cols[1] <- "firebrick"
cols[2] <- "slateblue"
cols[3] <- "lightseagreen"
cols[4] <- "thistle2"
cols[5] <- "orange1"
cols[6] <- "forestgreen"
cols[7] <- "goldenrod"
cols[8] <- "darkcyan"
cols[9] <- "black"
cols[10] <- "darkmagenta"

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "GeneIntegrationMatrix", 
    name = genes,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = colors_barplot
   )

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "14", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 4),p2))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure15/InterferonInflammation_Violins.pdf", height = 11, width = 12)
```
# Supp figure 17 - PI3K/AKT/MTOR signaling
```{r}
genes <- c("PIK3CA", "PIK3R1", "AKT1", "MTOR",
           "PTEN", "CDK1", "CFL1", "CDK4", 
           "PDK1", "PDPK1", "YWHAB", "GRK2",
           "FOXO1", "VAV3", "LCK", "PFN1")
p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(archR)
)

p2 <- lapply(p, function(x){ # insanely specific ggplot adjustments to fit the legend in here properly. jesus christ lol
    x + 
    #guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure16/PI3K-AKT-MTOR_UMAPs.pdf", height = 16, width = 16)

### and VIOLINS
cols <- table(archR$final_annotation)
names(cols) <- plot_order
cols[1] <- "firebrick"
cols[2] <- "slateblue"
cols[3] <- "lightseagreen"
cols[4] <- "thistle2"
cols[5] <- "orange1"
cols[6] <- "forestgreen"
cols[7] <- "goldenrod"
cols[8] <- "darkcyan"
cols[9] <- "black"
cols[10] <- "darkmagenta"

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "GeneIntegrationMatrix", 
    name = genes,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = cols
   )

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) +
    theme(
        #axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        #axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_text(size = "14", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 4),p2))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure16/PI3K-AKT-MTOR_Violins.pdf", height = 11, width = 12)
```
# Supp figure 18 - UPR and estrogen
```{r}
genes <- c("XBP1", "HERPUD1", "ERN1", "EIF2AK3",
           "ATF4", "ATF6", "SERP1", "PELI1",
           "CD63", "LMAN1", "TMED10", "ERGIC3",
           "ESR1", "ESR2", "GPER1", "B4GALT1")

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = getImputeWeights(archR)
)

p2 <- lapply(p, function(x){ 
    x + 
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) + # top, right, bottom, left
    theme(
        axis.ticks.x=element_blank(), 
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        title = element_blank(),
        axis.title.y = element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 4),p2))
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure17/ProteinHomeostasis_Estrogen_UMAPs.pdf", height = 16, width = 16)

### and VIOLINS
cols <- table(archR$final_annotation)
names(cols) <- plot_order
cols[1] <- "firebrick"
cols[2] <- "slateblue"
cols[3] <- "lightseagreen"
cols[4] <- "thistle2"
cols[5] <- "orange1"
cols[6] <- "forestgreen"
cols[7] <- "goldenrod"
cols[8] <- "darkcyan"
cols[9] <- "black"
cols[10] <- "darkmagenta"

p <- plotGroups(
    ArchRProj = archR, 
    groupBy = "final_annotation", 
    colorBy = "GeneIntegrationMatrix", 
    name = genes,
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    groupOrder = plot_order,
    pal = cols
   )

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    #theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0.25, 0.25, 0.25, 0.25), "cm")) +
    theme(
        axis.title.x = element_blank(),
        axis.text.x=element_blank(), 
        axis.text.y=element_text(size = "8"), 
        title = element_blank(),
        axis.title.y = element_text(size = "14", face = "bold")
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 4),p2))

ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Supplementary/SuppFigure17/ProteinHomeostasis_Estrogen_Violins.pdf", height = 11, width = 12)
```
# Testing
```{r}
genes <- c("CD79B", "CD79A", "BTK", "IRAK1", "IRAK3", "IRAK4", "IL1R1", "SYK", "LYN",
           "HCK", "IL6", "IL6ST")
cxcr4 <- c("CXCR4", "CXCL12")
del6q.genes <- c("PRDM1", "IBTK", "FOXO3", "BCL2", "TNFAIP3", "HIVEP2", "BACH2")
p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = del6q.genes, 
    embedding = "UMAP_Harmony",
    quantCut = c(0.01, 0.95),
    pal = ArchRPalettes$solarExtra,
    imputeWeights = NULL
)
p

```
# Checking subtypes vs. length of time having disesase...
```{r}
metadata <- read.csv("/Users/gagled01/morganLab/Waldenstroms/Data_for_Dylan_MorganTeam_clean_forR.csv", header = T)
metadata$Time.Marrow <- as.numeric(metadata$Time.Marrow) # Convert to numeric

# Subset out rows with NAs
metadata.sub <- subset(metadata, (!is.na(metadata$Subtype)) & (!is.na(metadata$Time.Marrow)))

# Check p-value for subtypes and time
chisq.test(metadata.sub$Time.Marrow, metadata.sub$Subtype)

# Visualize
ggplot(metadata, aes(x = Subtype, y = Time.Marrow)) + geom_point()


```

