---
title: "basic_VDJ_ClonotypeAnalysis"
output: html_document
date: "2023-09-25"
---

# doing basic VDJ clonotype analysis on WM samples.
# clonotype analysis performed with TRUST4 package using GEX bam from 10x multiome experiment on a per sample basis
# outputs fed into immunarch.
# main interest is to know which cells are clonotypic and which are not. this will allow us to know which cells are WM clones or not.

Load library
```{r}
library(immunarch)
library(Seurat)
library(ggplot2)
library(patchwork)
library(ArchR)
addArchRThreads(threads = 1)
addArchRGenome("hg38")
```
Load ArchR
```{r}
archR <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_BothSets_noControls_FullIntegration_Nov23")

# Get VERY clean cell names
veryCleanNames <- gsub("..-..._", "", archR$cellNames_clean)
veryVeryCleanNames <- gsub("-1", "", veryCleanNames)
archR$cellnames_VeryClean <- veryVeryCleanNames
```
# Reclustering
```{r}
archR <- addIterativeLSI(
  ArchRProj = archR,
  useMatrix = "TileMatrix",
  name = "IterativeLSI_noControls",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

archR <- addClusters(
  input = archR,
  reducedDims = "IterativeLSI_noControls",
  method = "Seurat",
  name = "NoControls_Reclustering",
  resolution = 1,
  force = T
)

archR <- addUMAP(
    ArchRProj = archR, 
    reducedDims = "IterativeLSI_noControls", 
    name = "NoControls_Reclustered_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesClean", embedding = "NoControls_Reclustered_UMAP")
p2 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "NoControls_Reclustering", embedding = "NoControls_Reclustered_UMAP")
p3 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "NoControls_Reclustered_UMAP") #+ #+ 
#+ #+ 
plotPDF(p1, p2, p3, name = "WM_NoControls_Reclustered_SamplesClustersAnnotations.pdf", ArchRProj = archR, addDOC = FALSE, width = 5, height = 5)
```
```{r}
saveArchRProject(ArchRProj = archR, outputDirectory = "WM_TRUST4_Updated_noControls_PeakReady", load = FALSE)
```

Add single-cell level metadata
```{r}
cell_id_fixed <- gsub("0.-..._", "", archR$cellNames_clean)
archR$cell_id_fixed <- cell_id_fixed
cell_id_fixed2 <- gsub("-1", "", archR$cell_id_fixed)
archR$cell_id_fixed <- cell_id_fixed2
cell_id_fixed3 <- gsub("WM..", "", archR$cell_id_fixed)
archR$cell_id_fixed <- cell_id_fixed3

# Load barcode files
p01_076_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-076_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_115_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-115_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_131_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-131_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_163_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-163_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_190_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-190_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p04_003_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_04-003_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p04_006_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_04-006_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm25_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM25_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm43_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM43_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm46_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM46_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm47_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM47_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm54_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM54_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
wm65_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_WM65_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)

# Fix cell ids
p01_076_barcodes$cell_id_fixed <- gsub("-1", "", p01_076_barcodes$cell_id)
p01_076_barcodes$sample <- "01-076"
p01_115_barcodes$cell_id_fixed <- gsub("-1", "", p01_115_barcodes$cell_id)
p01_115_barcodes$sample <- "01-115"
p01_131_barcodes$cell_id_fixed <- gsub("-1", "", p01_131_barcodes$cell_id)
p01_131_barcodes$sample <- "01-131"
p01_163_barcodes$cell_id_fixed <- gsub("-1", "", p01_163_barcodes$cell_id)
p01_163_barcodes$sample <- "01-163"
p01_190_barcodes$cell_id_fixed <- gsub("-1", "", p01_190_barcodes$cell_id)
p01_190_barcodes$sample <- "01-190"
p04_003_barcodes$cell_id_fixed <- gsub("-1", "", p04_003_barcodes$cell_id)
p04_003_barcodes$sample <- "04-003"
p04_006_barcodes$cell_id_fixed <- gsub("-1", "", p04_006_barcodes$cell_id)
p04_006_barcodes$sample <- "04-006"
wm25_barcodes$cell_id_fixed <- gsub("-1", "", wm25_barcodes$cell_id)
wm25_barcodes$sample <- "WM25"
wm43_barcodes$cell_id_fixed <- gsub("-1", "", wm43_barcodes$cell_id)
wm43_barcodes$sample <- "WM43"
wm46_barcodes$cell_id_fixed <- gsub("-1", "", wm46_barcodes$cell_id)
wm46_barcodes$sample <- "WM46"
wm47_barcodes$cell_id_fixed <- gsub("-1", "", wm47_barcodes$cell_id)
wm47_barcodes$sample <- "WM47"
wm54_barcodes$cell_id_fixed <- gsub("-1", "", wm54_barcodes$cell_id)
wm54_barcodes$sample <- "WM54"
wm65_barcodes$cell_id_fixed <- gsub("-1", "", wm65_barcodes$cell_id)
wm65_barcodes$sample <- "WM65"

#wm25_archR$cdr3 <- wm25_barcodes$junction_aa[match(wm25_archR$cell_id_fixed, wm25_barcodes$cell_id_fixed)]

# Merge into single df
all_barcodes <- rbind(p01_076_barcodes, p01_115_barcodes, p01_131_barcodes, p01_163_barcodes,p01_190_barcodes, p04_003_barcodes, p04_006_barcodes,
                      wm25_barcodes, wm43_barcodes, wm46_barcodes, wm47_barcodes, wm54_barcodes, wm65_barcodes)

# Update metadata
archR$v_call <- all_barcodes$v_call[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$d_call <- all_barcodes$d_call[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$j_call <- all_barcodes$j_call[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$c_call <- all_barcodes$j_call[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$cdr1 <- all_barcodes$cdr1[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$cdr2 <- all_barcodes$cdr2[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$cdr3 <- all_barcodes$junction_aa[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$complete_vdj <- all_barcodes$complete_vdj[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]
archR$consensus_count <- all_barcodes$consensus_count[match(archR$cell_id_fixed, all_barcodes$cell_id_fixed)]

# Make column of combined CDR3/V/J
combined_call1 <- paste(archR$cdr3, "/", archR$v_call)
combined_call2 <- paste(combined_call1, "/", archR$j_call)
archR$combined_call <- combined_call2

length(table(archR$combined_call))
```
# Load bam reports and get top 1-2 clones for each patient
```{r}
p01_076_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/01-076_report.tsv", sep = "\t", header = F)
p01_115_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/01-115_report.tsv", sep = "\t", header = F)
p01_131_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/01-131_report.tsv", sep = "\t", header = F)
p01_163_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/01-163_report.tsv", sep = "\t", header = F)
p01_190_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/01-190_report.tsv", sep = "\t", header = F)
p04_003_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/04-003_report.tsv", sep = "\t", header = F)
p04_006_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/04-006_report.tsv", sep = "\t", header = F)
wm25_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/TRUST_WM25_gex_possorted_bam_report.tsv", sep = "\t", header = F)
wm43_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/TRUST_WM43_gex_possorted_bam_report.tsv", sep = "\t", header = F)
wm46_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/TRUST_WM46_gex_possorted_bam_report.tsv", sep = "\t", header = F)
wm47_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/TRUST_WM47_gex_possorted_bam_report.tsv", sep = "\t", header = F)
wm54_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/TRUST_WM54_gex_possorted_bam_report.tsv", sep = "\t", header = F)
wm65_bamreport <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/TRUST_WM65_gex_possorted_bam_report.tsv", sep = "\t", header = F)

# Fix colnames
bamreport_colnames <- c("count", "frequency", "CDR3nt", "CDR3aa", "V", "D", "J", "C", "cid", "cid_full_length")

colnames(p01_076_bamreport) <- bamreport_colnames
colnames(p01_115_bamreport) <- bamreport_colnames
colnames(p01_131_bamreport) <- bamreport_colnames
colnames(p01_163_bamreport) <- bamreport_colnames
colnames(p01_190_bamreport) <- bamreport_colnames
colnames(p04_003_bamreport) <- bamreport_colnames
colnames(p04_006_bamreport) <- bamreport_colnames

colnames(wm25_bamreport) <- bamreport_colnames
colnames(wm43_bamreport) <- bamreport_colnames
colnames(wm46_bamreport) <- bamreport_colnames
colnames(wm47_bamreport) <- bamreport_colnames
colnames(wm54_bamreport) <- bamreport_colnames
colnames(wm65_bamreport) <- bamreport_colnames

# Make column of combined CDR3/V/J
combined_call1 <- paste(p01_076_bamreport$CDR3aa, "/", p01_076_bamreport$V)
combined_call2 <- paste(combined_call1, "/", p01_076_bamreport$J)
p01_076_bamreport$combined_call <- combined_call2

combined_call1 <- paste(p01_115_bamreport$CDR3aa, "/", p01_115_bamreport$V)
combined_call2 <- paste(combined_call1, "/", p01_115_bamreport$J)
p01_115_bamreport$combined_call <- combined_call2

combined_call1 <- paste(p01_131_bamreport$CDR3aa, "/", p01_131_bamreport$V)
combined_call2 <- paste(combined_call1, "/", p01_131_bamreport$J)
p01_131_bamreport$combined_call <- combined_call2

combined_call1 <- paste(p01_163_bamreport$CDR3aa, "/", p01_163_bamreport$V)
combined_call2 <- paste(combined_call1, "/", p01_163_bamreport$J)
p01_163_bamreport$combined_call <- combined_call2

combined_call1 <- paste(p01_190_bamreport$CDR3aa, "/", p01_190_bamreport$V)
combined_call2 <- paste(combined_call1, "/", p01_190_bamreport$J)
p01_190_bamreport$combined_call <- combined_call2

combined_call1 <- paste(p04_003_bamreport$CDR3aa, "/", p04_003_bamreport$V)
combined_call2 <- paste(combined_call1, "/", p04_003_bamreport$J)
p04_003_bamreport$combined_call <- combined_call2

combined_call1 <- paste(p04_006_bamreport$CDR3aa, "/", p04_006_bamreport$V)
combined_call2 <- paste(combined_call1, "/", p04_006_bamreport$J)
p04_006_bamreport$combined_call <- combined_call2

combined_call1 <- paste(wm25_bamreport$CDR3aa, "/", wm25_bamreport$V)
combined_call2 <- paste(combined_call1, "/", wm25_bamreport$J)
wm25_bamreport$combined_call <- combined_call2

combined_call1 <- paste(wm43_bamreport$CDR3aa, "/", wm43_bamreport$V)
combined_call2 <- paste(combined_call1, "/", wm43_bamreport$J)
wm43_bamreport$combined_call <- combined_call2

combined_call1 <- paste(wm46_bamreport$CDR3aa, "/", wm46_bamreport$V)
combined_call2 <- paste(combined_call1, "/", wm46_bamreport$J)
wm46_bamreport$combined_call <- combined_call2

combined_call1 <- paste(wm47_bamreport$CDR3aa, "/", wm47_bamreport$V)
combined_call2 <- paste(combined_call1, "/", wm47_bamreport$J)
wm47_bamreport$combined_call <- combined_call2

combined_call1 <- paste(wm54_bamreport$CDR3aa, "/", wm54_bamreport$V)
combined_call2 <- paste(combined_call1, "/", wm54_bamreport$J)
wm54_bamreport$combined_call <- combined_call2

combined_call1 <- paste(wm65_bamreport$CDR3aa, "/", wm65_bamreport$V)
combined_call2 <- paste(combined_call1, "/", wm65_bamreport$J)
wm65_bamreport$combined_call <- combined_call2

head(p01_076_bamreport)
```
Get top clones
```{r}
topNclone_UMAP <- function(archR_obj, immunarch_df_index, sample_str, n) {
  select_cdr3s <- unique(wm$data[[immunarch_df_index]]$CDR3.aa)[1:n]
  archR_obj$topCDR3s <- NA
  archR_obj$topCDR3s[archR_obj$cdr3 %in% select_cdr3s] <- select_cdr3s
  p1 <- plotEmbedding(ArchRProj = archR_obj, colorBy = "cellColData", name = "topCDR3s", embedding = "UMAP_Harmony_Batch")
  ggsave(p1, filename = paste0("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/", sample_str, "Top2CloneUMAP.png"), height = 6, width = 10)
  return(archR_obj)
}

  
# Get top CDR3s
topCDR3s <- c(p01_076_bamreport$CDR3aa[1],
              p01_115_bamreport$CDR3aa[1],
              p01_131_bamreport$CDR3aa[1],
              p01_163_bamreport$CDR3aa[1],
              p01_190_bamreport$CDR3aa[1],
              p04_003_bamreport$CDR3aa[1],
              p04_006_bamreport$CDR3aa[1],
              wm25_bamreport$CDR3aa[1],
              wm43_bamreport$CDR3aa[1],
              wm46_bamreport$CDR3aa[1],
              wm47_bamreport$CDR3aa[1],
              wm54_bamreport$CDR3aa[1],
              wm65_bamreport$CDR3aa[1])
topCDR3s
table(topCDR3s)
archR$topCDR3s <- NA
archR$topCDR3s[archR$cdr3 %in% topCDR3s] <- topCDR3s
table(archR$topCDR3s)

plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "topCDR3s", embedding = "NoProgen_Recluster_UMAP")

# Get top combined calls
topCalls <- c(p01_076_bamreport$combined_call[1],
              p01_115_bamreport$combined_call[1],
              p01_131_bamreport$combined_call[1],
              p01_163_bamreport$combined_call[1],
              p01_190_bamreport$combined_call[1],
              p04_003_bamreport$combined_call[1],
              p04_006_bamreport$combined_call[1],
              wm25_bamreport$combined_call[1],
              wm43_bamreport$combined_call[1],
              wm46_bamreport$combined_call[1],
              wm47_bamreport$combined_call[1],
              wm54_bamreport$combined_call[1],
              wm65_bamreport$combined_call[1])

archR$topClone <- NA
archR$topClone[archR$combined_call %in% topCalls] <- topCalls

table(archR$topClone) # WHY THIS LIKE THIS???

plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "topClone", embedding = "NoProgen_Recluster_UMAP")

```
How much do we lose w lambdas?
```{r}
all_calls <- table(all_barcodes$c_call)
all_barcodes_ighm_igkc_only <- subset(all_barcodes, subset = c_call %in% c("IGHM", "IGKC"))

difference <- as.numeric(table(all_barcodes$sample)) - as.numeric(table(all_barcodes_ighm_igkc_only$sample))
perc.loss <- difference/as.numeric(table(all_barcodes$sample))

pre <- table(all_barcodes$sample)
post <- table(all_barcodes_ighm_igkc_only$sample)
loss <- (perc.loss * 100)

df <- data.frame(total_calls = pre, igkc_ighm_only = post, perc.loss = loss)
colnames(df) <- c("Sample", "total_calls", "fleep", "igkc_ighm_only", "perc.loss")
df <- df[-3]

table(all_barcodes$c_call)
```
Renaming clusters
```{r}
table(archR$Harmony_Batch_Clusters_NoControls)

archR$cluster_names <- archR$Harmony_Batch_Clusters_NoControls
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C1"] <- "Pre-B"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C2"] <- "Pro-B"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C3"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C4"] <- "Normal MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C5"] <- "Normal MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C6"] <- "Normal MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C7"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C8"] <- "Naive B"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C9"] <- "Naive B"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C10"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C11"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C12"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C13"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C14"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C15"] <- "Plasma cells"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C16"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C17"] <- "Plasma cells"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C18"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C19"] <- "Plasma cells"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C20"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C21"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C22"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C23"] <- "Malignant MBC"
archR$cluster_names[archR$Harmony_Batch_Clusters_NoControls == "C24"] <- "Malignant MBC"

table(archR$cluster_names)
plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "cluster_names", embedding = "UMAP_Harmony_Batch_NoControls")
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/Updated_ClusterAnnotations_UMAP.png", height = 6, width = 6)
```

Make Immunarch metadata file
```{r}
# Sample <- c("01-076", "01-115", "01-131", "01-163", "01-190", "04-003", "04-003", "WM25", "WM43", "WM46", "WM47", "WM54", "WM65")
# Batch <- c("2", "2", "2", "2", "1", "2","1", "3", "3", "3", "3", "3", "3")
# MYD88 <- c("+", "+", "+", "+", "+", "+", "+", "+", "+", "+", "+", "+", "+")
# CXCR4 <- c("-", "+", "-", "-", "+", "-", "-", NA, NA, NA, NA, NA, NA)
# Clonotype <- c("kappa", "kappa", "kappa", "kappa", "kappa", "kappa", "kappa", "kappa", "kappa", "kappa", "kappa", "kappa", "kappa")
# 
# metadata <- data.frame(Sample, Batch, MYD88, CXCR4)
# metadata

# write.csv(metadata, "/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/metadata.txt", row.names = F) # THIS WILL STILL NEED MANUAL EDITING! FiLE must be tab sep and have no quotations
```
Load aggregated TRUST4 BCR data
```{r}
library(immunarch)
wm <- repLoad("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports")
wm$data
```
# Patient subset function
```{r}
patient_subset <- function(archR_obj, id_str) {
  idx <- BiocGenerics::which(archR$samplesClean == id_str)
  cells <- archR_obj$cellNames[idx]
  sub_archR <- archR_obj[cells,]
  return(sub_archR)
}
```
# Run it on em
```{r}
p01_076_archR <- patient_subset(archR, "01-076")
p01_115_archR <- patient_subset(archR, "01-115")
p01_131_archR <- patient_subset(archR, "01-131")
p01_163_archR <- patient_subset(archR, "01-163")
p01_190_archR <- patient_subset(archR, "01-190")
p04_003_archR <- patient_subset(archR, "04-003")
p04_006_archR <- patient_subset(archR, "04-006")
wm25_archR <- patient_subset(archR, "WM25")
wm43_archR <- patient_subset(archR, "WM43")
wm46_archR <- patient_subset(archR, "WM46")
wm47_archR <- patient_subset(archR, "WM47")
wm54_archR <- patient_subset(archR, "WM54")
wm65_archR <- patient_subset(archR, "WM65")
```
# function to get top N clones in an ArchR Object (preferably a patient subset one) and create UMAP
```{r}
topNclone_UMAP <- function(archR_obj, immunarch_df_index, sample_str, n) {
  select_cdr3s <- unique(wm$data[[immunarch_df_index]]$CDR3.aa)[1:n]
  archR_obj$topCDR3s <- NA
  archR_obj$topCDR3s[archR_obj$cdr3 %in% select_cdr3s] <- select_cdr3s
  p1 <- plotEmbedding(ArchRProj = archR_obj, colorBy = "cellColData", name = "topCDR3s", embedding = "UMAP_Harmony_Batch")
  ggsave(p1, filename = paste0("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/", sample_str, "Top2CloneUMAP.png"), height = 6, width = 10)
  return(archR_obj)
}
```
# Now run this
```{r}
p01_076_archR <- topNclone_UMAP(p01_076_archR, 1, "01-076", 2)
p01_115_archR <- topNclone_UMAP(p01_115_archR, 2, "01-115", 2)
p01_131_archR <- topNclone_UMAP(p01_131_archR, 3, "01-131", 2)
p01_163_archR <- topNclone_UMAP(p01_163_archR, 4, "01-163", 2)
p01_190_archR <- topNclone_UMAP(p01_190_archR, 5, "01-190", 2)
p04_003_archR <- topNclone_UMAP(p04_003_archR, 6, "04-003", 2)
p04_006_archR <- topNclone_UMAP(p04_006_archR, 7, "04-006", 2)
wm25_archR <- topNclone_UMAP(wm25_archR, 8, "WM25", 2)
wm43_archR <- topNclone_UMAP(wm43_archR, 9, "WM43", 2)
wm46_archR <- topNclone_UMAP(wm46_archR, 10, "WM46", 2)
wm47_archR <- topNclone_UMAP(wm47_archR, 11, "WM47", 2)
wm54_archR <- topNclone_UMAP(wm54_archR, 12, "WM54", 2)
wm65_archR <- topNclone_UMAP(wm65_archR, 13, "WM65", 2)
```
Reclustering individual patients and plot
```{r}
patient_recluster <- function(archR_obj, sample_str) {
  print(paste0("Reclustering patient ", sample_str, "..."))
  print("Running Iterative LSI...")
  
  # Add iterative LSI
  archR_obj <- addIterativeLSI(
  ArchRProj = archR_obj,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)
  print("Adding UMAP...")
# Add UMAP
  archR_obj <- addUMAP(
      ArchRProj = archR_obj, 
      reducedDims = "Patient_Recluster", 
      name = "Patient_Recluster_UMAP", 
      nNeighbors = 30, 
      minDist = 0.5, 
      metric = "cosine",
      force = T
  )
  return(archR_obj)
}
```

```{r}
p01_076_archR <- patient_recluster(p01_076_archR, "01-076")
p01_115_archR <- patient_recluster(p01_115_archR, "01-115")
p01_131_archR <- patient_recluster(p01_131_archR, "01-131")
p01_163_archR <- patient_recluster(p01_163_archR, "01-163")
p01_190_archR <- patient_recluster(p01_190_archR, "01-190")
p04_003_archR <- patient_recluster(p04_003_archR, "04-003")
p04_006_archR <- patient_recluster(p04_006_archR, "04-006")
wm25_archR <- patient_recluster(wm25_archR, "WM25")
wm43_archR <- patient_recluster(wm43_archR, "WM43")
wm46_archR <- patient_recluster(wm46_archR, "WM46")
wm47_archR <- patient_recluster(wm47_archR, "WM47")
wm54_archR <- patient_recluster(wm54_archR, "WM54")
wm65_archR <- patient_recluster(wm65_archR, "WM65")
```
# Manual clustering for WM46 bcuz error...
# Needed to raise the resolution from 0.25 to 0.5
```{r}
wm46_archR <- addIterativeLSI(
ArchRProj = wm46_archR,
useMatrix = "TileMatrix",
name = "Patient_Recluster",
iterations = 2,
clusterParams = list( # based on Seurat's FindClusters()
  resolution = c(0.5), # had to raise to 0.5 for WM46 specifically
  sampleCells = 2000,
  n.start = 10
),
varFeatures = 20000,
dimsToUse = 1:10,
force = T
)

wm46_archR <- addUMAP(
      ArchRProj = wm46_archR, 
      reducedDims = "Patient_Recluster", 
      name = "Patient_Recluster_UMAP", 
      nNeighbors = 30, 
      minDist = 0.5, 
      metric = "cosine",
      force = T
  )

wm65_archR <- addIterativeLSI(
ArchRProj = wm65_archR,
useMatrix = "TileMatrix",
name = "Patient_Recluster",
iterations = 2,
clusterParams = list( # based on Seurat's FindClusters()
  resolution = c(0.5), # had to raise to 0.5 for WM46 specifically
  sampleCells = 2000,
  n.start = 10
),
varFeatures = 20000,
dimsToUse = 1:10,
force = T
)

wm65_archR <- addUMAP(
      ArchRProj = wm65_archR, 
      reducedDims = "Patient_Recluster", 
      name = "Patient_Recluster_UMAP", 
      nNeighbors = 30, 
      minDist = 0.5, 
      metric = "cosine",
      force = T
  )
```
# Plot
```{r}
patientRecluster_plots <- function(archR_obj, sample_str) {
  print(paste0("Making plots for sample: ", sample_str))
  p1 <- plotEmbedding(ArchRProj = archR_obj, colorBy = "cellColData", name = "topCDR3s", embedding = "Patient_Recluster_UMAP")
  p2 <- plotEmbedding(ArchRProj = archR_obj, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
  ggsave(p1, filename = paste0("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/", sample_str, "_PatientReclustered_TopClones_UMAP.png"), height = 6, width = 10)
  ggsave(p2, filename = paste0("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/", sample_str, "_PatientReclustered_CellTypist_UMAP.png"), height = 6, width = 10)
}
```

```{r}
patientRecluster_plots(p01_076_archR, "01-076")
patientRecluster_plots(p01_115_archR, "01-115")
patientRecluster_plots(p01_131_archR, "01-131")
patientRecluster_plots(p01_163_archR, "01-163")
patientRecluster_plots(p01_190_archR, "01-190")
patientRecluster_plots(p04_003_archR, "04-003")
patientRecluster_plots(p04_006_archR, "04-006")
patientRecluster_plots(wm25_archR, "WM25")
patientRecluster_plots(wm43_archR, "WM43")
patientRecluster_plots(wm46_archR, "WM46")
patientRecluster_plots(wm47_archR, "WM47")
patientRecluster_plots(wm54_archR, "WM54")
patientRecluster_plots(wm65_archR, "WM65")
```

# Better table viewing
```{r}
table(p04_006_archR$cdr3) %>% 
        as.data.frame() %>% 
        arrange(desc(Freq))

length(p04_006_archR$cellNames) - sum(table(p04_006_archR$cdr3))
(length(p04_006_archR$cellNames) - sum(table(p04_006_archR$cdr3)))/length(p04_006_archR$cellNames)

```
# Basic bar plots
```{r}
library(ggplot2)

count.table <- as.data.frame(table(p01_076_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/01-076_CellType_Barplots.png", height = 6, width = 8)

############################

count.table <- as.data.frame(table(p01_115_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/01-115_CellType_Barplots.png", height = 6, width = 8)

############################

count.table <- as.data.frame(table(p01_131_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/01-131_CellType_Barplots.png", height = 6, width = 8)

############################

count.table <- as.data.frame(table(p01_163_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/01-163_CellType_Barplots.png", height = 6, width = 8)


############################

count.table <- as.data.frame(table(p01_190_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/01-190_CellType_Barplots.png", height = 6, width = 8)

############################

count.table <- as.data.frame(table(p04_003_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/04_003_CellType_Barplots.png", height = 6, width = 8)

############################

count.table <- as.data.frame(table(p04_006_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/04_006_CellType_Barplots.png", height = 6, width = 8)
```
Next patinet batch
```{r}
count.table <- as.data.frame(table(wm25_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 5))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5) + 
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM25_CellType_Barplots.png", height = 6, width = 8)

##########################################

count.table <- as.data.frame(table(wm43_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM43_CellType_Barplots.png", height = 6, width = 8)

##########################################

count.table <- as.data.frame(table(wm46_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 4))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM46_CellType_Barplots.png", height = 6, width = 8)

##########################################

count.table <- as.data.frame(table(wm47_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM47_CellType_Barplots.png", height = 6, width = 8)

##########################################

count.table <- as.data.frame(table(wm54_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM54_CellType_Barplots.png", height = 6, width = 8)

##########################################

count.table <- as.data.frame(table(wm65_archR$predictedGroup_Co))
df <- as.data.frame(matrix(ncol = 2, nrow = 6))
colnames(df) <- c("Celltype", "Count")
df$Celltype <- count.table$Var1
df$Count <- count.table$Freq

# Basic barplot
ggplot(data=df, aes(x=Celltype, y=Count, fill = Celltype)) +
  geom_bar(stat="identity") + 
  geom_text(aes(label=Count), vjust=-0.3, size=5)+
  theme_minimal() + NoLegend()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM65_CellType_Barplots.png", height = 6, width = 8)
```



```{r}
p01_115_archR <- addIterativeLSI(
  ArchRProj = p01_115_archR,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

p01_115_archR <- addClusters(
  input = p01_115_archR,
  reducedDims = "Patient_Recluster",
  method = "Seurat",
  name = "Patient_Reclustering",
  resolution = 1,
  force = T
)

p01_115_archR <- addUMAP(
    ArchRProj = p01_115_archR, 
    reducedDims = "Patient_Recluster", 
    name = "Patient_Recluster_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = p01_115_archR, colorBy = "cellColData", name = "topCDR3", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-115_TopClone_PatientReclustered_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = p01_115_archR, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-115_TopClone_PatientReclustered_CellTypist_UMAP.png", height = 6, width = 10)
```

```{r}
p01_131_archR <- addIterativeLSI(
  ArchRProj = p01_131_archR,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

p01_131_archR <- addClusters(
  input = p01_131_archR,
  reducedDims = "Patient_Recluster",
  method = "Seurat",
  name = "Patient_Reclustering",
  resolution = 1,
  force = T
)

p01_131_archR <- addUMAP(
    ArchRProj = p01_131_archR, 
    reducedDims = "Patient_Recluster", 
    name = "Patient_Recluster_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = p01_131_archR, colorBy = "cellColData", name = "topCDR3", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-131_TopClone_PatientReclustered_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = p01_131_archR, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-131_TopClone_PatientReclustered_CellTypist_UMAP.png", height = 6, width = 10)
```

```{r}
p01_163_archR <- addIterativeLSI(
  ArchRProj = p01_163_archR,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

p01_163_archR <- addClusters(
  input = p01_163_archR,
  reducedDims = "Patient_Recluster",
  method = "Seurat",
  name = "Patient_Reclustering",
  resolution = 1,
  force = T
)

p01_163_archR <- addUMAP(
    ArchRProj = p01_163_archR, 
    reducedDims = "Patient_Recluster", 
    name = "Patient_Recluster_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = p01_163_archR, colorBy = "cellColData", name = "topCDR3", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-163_TopClone_PatientReclustered_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = p01_163_archR, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-163_TopClone_PatientReclustered_CellTypist_UMAP.png", height = 6, width = 10)
```
```{r}
p01_190_archR <- addIterativeLSI(
  ArchRProj = p01_190_archR,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

p01_190_archR <- addClusters(
  input = p01_190_archR,
  reducedDims = "Patient_Recluster",
  method = "Seurat",
  name = "Patient_Reclustering",
  resolution = 1,
  force = T
)

p01_190_archR <- addUMAP(
    ArchRProj = p01_190_archR, 
    reducedDims = "Patient_Recluster", 
    name = "Patient_Recluster_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = p01_190_archR, colorBy = "cellColData", name = "topCDR3", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-190_TopClone_PatientReclustered_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = p01_190_archR, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/01-190_TopClone_PatientReclustered_CellTypist_UMAP.png", height = 6, width = 10)
```

```{r}
p04_003_archR <- addIterativeLSI(
  ArchRProj = p04_003_archR,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

p04_003_archR <- addClusters(
  input = p04_003_archR,
  reducedDims = "Patient_Recluster",
  method = "Seurat",
  name = "Patient_Reclustering",
  resolution = 1,
  force = T
)

p04_003_archR <- addUMAP(
    ArchRProj = p04_003_archR, 
    reducedDims = "Patient_Recluster", 
    name = "Patient_Recluster_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = p04_003_archR, colorBy = "cellColData", name = "topCDR3", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/04-003_TopClone_PatientReclustered_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = p04_003_archR, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/04-003_TopClone_PatientReclustered_CellTypist_UMAP.png", height = 6, width = 10)
```

```{r}
p04_006_archR <- addIterativeLSI(
  ArchRProj = p04_006_archR,
  useMatrix = "TileMatrix",
  name = "Patient_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

p04_006_archR <- addClusters(
  input = p04_006_archR,
  reducedDims = "Patient_Recluster",
  method = "Seurat",
  name = "Patient_Reclustering",
  resolution = 1,
  force = T
)

p04_006_archR <- addUMAP(
    ArchRProj = p04_006_archR, 
    reducedDims = "Patient_Recluster", 
    name = "Patient_Recluster_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = p04_006_archR, colorBy = "cellColData", name = "topCDR3", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/04-006_TopClone_PatientReclustered_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = p04_006_archR, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "Patient_Recluster_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/04-006_TopClone_PatientReclustered_CellTypist_UMAP.png", height = 6, width = 10)
```
Get clonal proportions by cell type
```{r}
p01_076_top_cdr3 <- unique(wm$data[[1]]$CDR3.aa)[1]
p01_115_top_cdr3 <- unique(wm$data[[2]]$CDR3.aa)[1]
p01_131_top_cdr3 <- unique(wm$data[[3]]$CDR3.aa)[1]
p01_163_top_cdr3 <- unique(wm$data[[4]]$CDR3.aa)[1]
p01_190_top_cdr3 <- unique(wm$data[[5]]$CDR3.aa)[1]
p04_003_top_cdr3 <- unique(wm$data[[6]]$CDR3.aa)[1]
p04_006_top_cdr3 <- unique(wm$data[[7]]$CDR3.aa)[1]
wm25_top_cdr3 <- unique(wm$data[[8]]$CDR3.aa)[1]
wm43_top_cdr3 <- unique(wm$data[[9]]$CDR3.aa)[1]
wm46_top_cdr3 <- unique(wm$data[[10]]$CDR3.aa)[1]
wm47_top_cdr3 <- unique(wm$data[[11]]$CDR3.aa)[1]
wm54_top_cdr3 <- unique(wm$data[[12]]$CDR3.aa)[1]
wm65_top_cdr3 <- unique(wm$data[[13]]$CDR3.aa)[1]

# Apply top CDR3s to archR object - need these
topCDR3 <- c(p01_076_top_cdr3, p01_115_top_cdr3, p01_131_top_cdr3, p01_163_top_cdr3, p01_190_top_cdr3, 
             p04_003_top_cdr3, p04_006_top_cdr3,
             wm25_top_cdr3, wm43_top_cdr3, wm46_top_cdr3, wm47_top_cdr3, wm54_top_cdr3, wm65_top_cdr3)
archR$topCDR3[archR$cdr3 %in% topCDR3] <- topCDR3

# Subset out PCs and MBCs
idx01_076_pcmbcs <- BiocGenerics::which(archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells01_076_pcmbcs <- archR$cellNames[idx01_076_pcmbcs]
pcmbcs <- archR[cells01_076_pcmbcs]

table(pcmbcs$samplesClean)
table(pcmbcs$topCDR3)

# Subsetting PC/MBCs per patient
# 01-076
idx01_076_pcmbcs <- BiocGenerics::which(p01_076_archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells01_076_pcmbcs <- p01_076_archR$cellNames[idx01_076_pcmbcs]
p01_076_pcmbcs <- p01_076_archR[cells01_076_pcmbcs]
# 01-115
idx01_115_pcmbcs <- BiocGenerics::which(p01_115_archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells01_115_pcmbcs <- p01_115_archR$cellNames[idx01_115_pcmbcs]
p01_115_pcmbcs <- p01_115_archR[cells01_115_pcmbcs]
# 01-131
idx01_131_pcmbcs <- BiocGenerics::which(p01_131_archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells01_131_pcmbcs <- p01_131_archR$cellNames[idx01_131_pcmbcs]
p01_131_pcmbcs <- p01_131_archR[cells01_131_pcmbcs]
# 01-163
idx01_163_pcmbcs <- BiocGenerics::which(p01_163_archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells01_163_pcmbcs <- p01_163_archR$cellNames[idx01_163_pcmbcs]
p01_163_pcmbcs <- p01_163_archR[cells01_163_pcmbcs]
# 01-190
idx01_190_pcmbcs <- BiocGenerics::which(p01_190_archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells01_190_pcmbcs <- p01_190_archR$cellNames[idx01_190_pcmbcs]
p01_190_pcmbcs <- p01_190_archR[cells01_190_pcmbcs]
# 04-003
idx04_003_pcmbcs <- BiocGenerics::which(p04_003_archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells04_003_pcmbcs <- p04_003_archR$cellNames[idx04_003_pcmbcs]
p04_003_pcmbcs <- p04_003_archR[cells04_003_pcmbcs]
# 04-006
idx04_006_pcmbcs <- BiocGenerics::which(p04_006_archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
cells04_006_pcmbcs <- p04_006_archR$cellNames[idx04_006_pcmbcs]
p04_006_pcmbcs <- p04_006_archR[cells04_006_pcmbcs]

table(p01_076_pcmbcs$topCDR3)/length(p01_076_pcmbcs$cellNames)
table(p01_115_pcmbcs$topCDR3)/length(p01_115_pcmbcs$cellNames)
table(p01_131_pcmbcs$topCDR3)/length(p01_131_pcmbcs$cellNames)
table(p01_163_pcmbcs$topCDR3)/length(p01_163_pcmbcs$cellNames)
table(p01_190_pcmbcs$topCDR3)/length(p01_190_pcmbcs$cellNames)
table(p04_003_pcmbcs$topCDR3)/length(p04_003_pcmbcs$cellNames)
table(p04_006_pcmbcs$topCDR3)/length(p04_006_pcmbcs$cellNames)

length(p01_076_archR$cellNames)
length(p01_115_archR$cellNames)
length(p01_131_archR$cellNames)
length(p01_163_archR$cellNames)
length(p01_190_archR$cellNames)
length(p04_003_archR$cellNames)
length(p04_006_archR$cellNames)

# Subsetting not-PC/MBCs per patient
non_pcmbcs <- c("Age-associated B cells", "Large pre-B cells", "Naive B cells", "Pro-B cells", "Proliferative germinal center B cells", "Small pre-B cells")
# 01-076
idx01_076_non.pcmcbs <- BiocGenerics::which(p01_076_archR$predictedGroup_Co %in% non_pcmbcs)
cells01_076_non.pcmcbs <- p01_076_archR$cellNames[idx01_076_non.pcmcbs]
p01_076_non.pcmcbs <- p01_076_archR[cells01_076_non.pcmcbs]
# 01-115
idx01_115_non.pcmcbs <- BiocGenerics::which(p01_115_archR$predictedGroup_Co %in% non_pcmbcs)
cells01_115_non.pcmcbs <- p01_115_archR$cellNames[idx01_115_non.pcmcbs]
p01_115_non.pcmcbs <- p01_115_archR[cells01_115_non.pcmcbs]
# 01-131
idx01_131_non.pcmcbs <- BiocGenerics::which(p01_131_archR$predictedGroup_Co %in% non_pcmbcs)
cells01_131_non.pcmcbs <- p01_131_archR$cellNames[idx01_131_non.pcmcbs]
p01_131_non.pcmcbs <- p01_131_archR[cells01_131_non.pcmcbs]
# 01-163
idx01_163_non.pcmcbs <- BiocGenerics::which(p01_163_archR$predictedGroup_Co %in% non_pcmbcs)
cells01_163_non.pcmcbs <- p01_163_archR$cellNames[idx01_163_non.pcmcbs]
p01_163_non.pcmcbs <- p01_163_archR[cells01_163_non.pcmcbs]
# 01-190
idx01_190_non.pcmcbs <- BiocGenerics::which(p01_190_archR$predictedGroup_Co %in% non_pcmbcs)
cells01_190_non.pcmcbs <- p01_190_archR$cellNames[idx01_190_non.pcmcbs]
p01_190_non.pcmcbs <- p01_190_archR[cells01_190_non.pcmcbs]
# 04-003
idx04_003_non.pcmcbs <- BiocGenerics::which(p04_003_archR$predictedGroup_Co %in% non_pcmbcs)
cells04_003_non.pcmcbs <- p04_003_archR$cellNames[idx04_003_non.pcmcbs]
p04_003_non.pcmcbs <- p04_003_archR[cells04_003_non.pcmcbs]
# 04-006
idx04_006_non.pcmcbs <- BiocGenerics::which(p04_006_archR$predictedGroup_Co %in% non_pcmbcs)
cells04_006_non.pcmcbs <- p04_006_archR$cellNames[idx04_006_non.pcmcbs]
p04_006_non.pcmcbs <- p04_006_archR[cells04_006_non.pcmcbs]

length(p01_076_non.pcmcbs$cellNames)
length(p01_115_non.pcmcbs$cellNames)
length(p01_131_non.pcmcbs$cellNames)
length(p01_163_non.pcmcbs$cellNames)
length(p01_190_non.pcmcbs$cellNames)
length(p04_003_non.pcmcbs$cellNames)
length(p04_006_non.pcmcbs$cellNames)

table(p01_076_non.pcmcbs$topCDR3)
table(p01_115_non.pcmcbs$topCDR3)
table(p01_131_non.pcmcbs$topCDR3)
table(p01_163_non.pcmcbs$topCDR3)
table(p01_190_non.pcmcbs$topCDR3)
table(p04_003_non.pcmcbs$topCDR3)
table(p04_006_non.pcmcbs$topCDR3)

table(p01_076_non.pcmcbs$topCDR3)/length(p01_076_non.pcmcbs$cellNames)
table(p01_115_non.pcmcbs$topCDR3)/length(p01_115_non.pcmcbs$cellNames)
table(p01_131_non.pcmcbs$topCDR3)/length(p01_131_non.pcmcbs$cellNames)
table(p01_163_non.pcmcbs$topCDR3)/length(p01_163_non.pcmcbs$cellNames)
table(p01_190_non.pcmcbs$topCDR3)/length(p01_190_non.pcmcbs$cellNames)
table(p04_003_non.pcmcbs$topCDR3)/length(p04_003_non.pcmcbs$cellNames)
table(p04_006_non.pcmcbs$topCDR3)/length(p04_006_non.pcmcbs$cellNames)
```
Removing non PC/MBCs
```{r}
idx <- BiocGenerics::which(archR$predictedGroup_Co %in% c("Memory B cells", "Plasma cells"))
removecells <- archR$cellNames[idx]
archR_subbed <- archR[removecells,]
table(archR_subbed$predictedGroup_Co)

# Adding selectCDR3
selectCDR3 <- c("CQQFDSSWTF", "CLQRGNWPLTF", "CQQYYSIPTF", "CQQYNNWPPLTF", "CQQYNNWPITF", "CQQYDSYPYTF", "CQQYGTSSLTF")
archR_subbed$topCDR3[archR_subbed$cdr3 %in% selectCDR3] <- selectCDR3

# Manual clonal vs non clonal annotations
archR_subbed$cell_types <- archR_subbed$predictedGroup_Co

temp <- archR_subbed$cell_types

idx <- BiocGenerics::which(archR_subbed$noPCMBC_Reclustering %in% c("C1", "C7", "C8"))

archR_subbed$cell_types[archR_subbed$noPCMBC_Reclustering %in% c("C1", "C7", "C8")] <- "Non-clonal MBC"
archR_subbed$cell_types[archR_subbed$cell_types == "Memory B cells"] <- "Clonal MBC"
table(archR_subbed$cell_types)

archR_subbed <- addIterativeLSI(
  ArchRProj = archR_subbed,
  useMatrix = "TileMatrix",
  name = "NoControls_noPCMBC_Recluster",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 2000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:20,
  force = T
)

archR_subbed <- addClusters(
  input = archR_subbed,
  reducedDims = "NoControls_noPCMBC_Recluster",
  method = "Seurat",
  name = "noPCMBC_Reclustering",
  resolution = 1,
  force = T
)

archR_subbed <- addUMAP(
    ArchRProj = archR_subbed, 
    reducedDims = "NoControls_noPCMBC_Recluster", 
    name = "noPCMBC_Reclustering_UMAP", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = archR_subbed, colorBy = "cellColData", name = "samplesClean", embedding = "noPCMBC_Reclustering_UMAP")
p1
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/no_PCMBC_PatientReclustered_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = archR_subbed, colorBy = "cellColData", name = "predictedGroup_Co", embedding = "noPCMBC_Reclustering_UMAP")
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/no_PCMBC_PatientReclustered_CellTypist_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = archR_subbed, colorBy = "cellColData", name = "topCDR3", embedding = "noPCMBC_Reclustering_UMAP")
p1
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/no_PCMBC_topCDR3s_UMAP.png", height = 6, width = 10)

p1 <- plotEmbedding(ArchRProj = archR_subbed, colorBy = "cellColData", name = "cell_types", embedding = "noPCMBC_Reclustering_UMAP")
p1
ggsave(p1, filename = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_TRUST4_UpdatedMetadata_noControls/Plots/no_PCMBC_FinalAnnotations_UMAP.png", height = 6, width = 10)

```
SAVE IT OUT!!!
```{r}
saveArchRProject(ArchRProj = archR_subbed, outputDirectory = "WM_TRUST4_PCMBConly_DownstreamReady_Object", load = F)
```

Exploration
```{r}
exp_vol <- repExplore(wm$data, .method = "volume")
p1 <- vis(exp_vol, .by = c("Sample"), .meta = wm$meta, .test = F)
p1
```
More
```{r}
exp_len <- repExplore(wm$data, .method = "len", .col = "aa")
exp_cnt <- repExplore(wm$data, .method = "count")
exp_vol <- repExplore(wm$data, .method = "volume")

p1 <- vis(exp_len)
p2 <- vis(exp_cnt)
p3 <- vis(exp_vol)

p1 / (p2 + p3)
ggsave("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/figures/WM_BasicClonoTypeinfo.png", height = 9, width = 12)

p3
ggsave("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/figures/WM_ClonotypeCounts.png", height = 6, width = 6)
```

```{r}
#repExplore(immune$data, "lens") %>% vis()
repClonality(wm$data, "homeo", .clone.types = c(Small = .0001, Medium = .001, Large = .01, Hyperexpanded = 1)) %>% vis() + 
  scale_x_discrete(labels = c("01-076", "01-115", "01-131", "01-163", "01-190", "04-003", "04-006"))
ggsave("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/figures/WM_CloneRelativeAbundance.png", height = 6, width = 8)

# Rare approach
imm_rare <- repClonality(wm$data, .method = "rare",)
vis(imm_rare)# + vis(imm_rare, .by = "Sample", .meta = wm$meta)
ggsave("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/figures/WM_CloneRelativeAbundance_Rare.png", height = 8, width = 8)

# Top - analyzing most abundant clonotypes
imm_top <- repClonality(wm$data, .method = "top", .head = c(10, 100, 1000, 3000, 10000))
vis(imm_top) + vis(imm_top, .by = "Sample", .meta = wm$meta)
ggsave("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/figures/WM_CloneRelativeAbundance_Top.png", height = 9, width = 12)

```

```{r}
library(dplyr)
library(tibble)
target <- wm$data[[1]] %>%
  dplyr::select(CDR3.aa, V.name) %>%
  head(5)

tc <- trackClonotypes(wm$data, target)
vis(tc)
```
```{r}
target <- c("IGKV3-20*01", "IGKV3-11*01", "IGKV4-1*01", "IGKV3-15*01", "IGKV1-16*01", 
            "IGHV3-33*01", "IGHV3-23*01", "IGHV3-23*01", "IGLV2-23*01", "IGLV3-1*01")
tc <- trackClonotypes(wm$data, target, .col = "v")
vis(tc, .plot = "smooth")
ggsave("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/figures/WM_SelectTopVGenes.png", height = 9, width = 12)
```

# Visualize
```{r}
repClonality(test$data, "homeo") %>% vis()
```
```{r}
tc1 <- trackClonotypes(test$data, list(1, 5), .col = "nt")

```
# Per barcode output
```{r}
# Fix seurat cell ids
seurat <- subset(seurat, subset = source == "NYU")
seurat@meta.data$cell_id_fixed <- gsub("-1_._.", "", colnames(seurat))

# Load barcode files
p01_076_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-076_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_115_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-115_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_131_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-131_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_163_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-163_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p01_190_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_01-190_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p04_003_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_p01_076_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)
p04_006_barcodes <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/barcode_reports/TRUST_04-006_gex_possorted_bam_barcode_airr.tsv", sep = "\t", header = T)

# Fix cell ids
p01_076_barcodes$cell_id_fixed <- gsub("-1", "", p01_076_barcodes$cell_id)
p01_115_barcodes$cell_id_fixed <- gsub("-1", "", p01_115_barcodes$cell_id)
p01_131_barcodes$cell_id_fixed <- gsub("-1", "", p01_131_barcodes$cell_id)
p01_163_barcodes$cell_id_fixed <- gsub("-1", "", p01_163_barcodes$cell_id)
p01_190_barcodes$cell_id_fixed <- gsub("-1", "", p01_190_barcodes$cell_id)
p04_003_barcodes$cell_id_fixed <- gsub("-1", "", p04_003_barcodes$cell_id)
p04_006_barcodes$cell_id_fixed <- gsub("-1", "", p04_006_barcodes$cell_id)

# Merge into single df
all_barcodes <- rbind(p01_076_barcodes, p01_115_barcodes, p01_131_barcodes, p01_163_barcodes, p01_190_barcodes, p04_003_barcodes, p04_006_barcodes)
dim(all_barcodes)
table(all_barcodes$complete_vdj)

# Update metadata
seurat[["v_call"]] <- all_barcodes$v_call[match(seurat@meta.data$cell_id_fixed, all_barcodes$cell_id_fixed)]
seurat[["d_call"]] <- all_barcodes$d_call[match(seurat@meta.data$cell_id_fixed, all_barcodes$cell_id_fixed)]
seurat[["j_call"]] <- all_barcodes$j_call[match(seurat@meta.data$cell_id_fixed, all_barcodes$cell_id_fixed)]
seurat[["cdr1"]] <- all_barcodes$cdr1[match(seurat@meta.data$cell_id_fixed, all_barcodes$cell_id_fixed)]
seurat[["cdr2"]] <- all_barcodes$cdr2[match(seurat@meta.data$cell_id_fixed, all_barcodes$cell_id_fixed)]
seurat[["complete_vdj"]] <- all_barcodes$complete_vdj[match(seurat@meta.data$cell_id_fixed, all_barcodes$cell_id_fixed)]
seurat[["consensus_count"]] <- all_barcodes$consensus_count[match(seurat@meta.data$cell_id_fixed, all_barcodes$cell_id_fixed)]

length(table(seurat@meta.data$v_call))

DimPlot(seurat, group.by = "v_call")
DimPlot(seurat, group.by = "d_call") + NoLegend()
DimPlot(seurat, group.by = "j_call") + NoLegend()
```
Per patient example
```{r}
p04_006 <- subset(seurat, subset = patient == "04-006")

DimPlot(p04_006, group.by = "v_call")
```

```{r}
# Fix input barcode cell ids
p01_115_trust4_cells_fixed <- gsub("-1", "", p01_115_barcodes$cell_id)
p01_115_barcodes$cell_id_fixed <- p01_115_trust4_cells_fixed

# Update seurat metadata by fixed cell id columns
seurat[["v_call"]] <- p01_115_barcodes$v_call[match(seurat@meta.data$cell_id_fixed, p01_115_barcodes$cell_id_fixed)]
seurat[["v_call"]] <- p01_115_barcodes$v_call[match(seurat@meta.data$cell_id_fixed, p01_115_barcodes$cell_id_fixed)]

intersect(seurat@meta.data$cell_id_fixed, p01_115_barcodes$cell_id_fixed)


table(seurat@meta.data$v_call)

patient_sub <- subset(seurat, subset = patient == "01-115")
head(colnames(patient_sub))
trust4_cells <- gsub("-1", "", barcodes$cell_id)
seurat_subbed_cells <- gsub("-1_4_2", "", colnames(patient_sub))
matched_cells <- (intersect(trust4_cells, seurat_subbed_cells))
length(matched_cells)


head(setdiff(trust4_cells, seurat_subbed_cells))
length(setdiff(seurat_subbed_cells, trust4_cells))

head(setdiff(seurat_subbed_cells, trust4_cells))
head(trust4_cells)
```

```


