---
title: "WM_RNATAC_PreliminaryAnalysis"
output: html_document
date: "2023-04-12"
---

# Script for performing basic QC and annotation for both the RNA/ATAC-seq data from the 2 WM samples which were finished in the first week of April, 2023.
# Goals are to 1. Do basic QC and 2. See if we can find the MYD88 Waldenstrom plasma cells

# Using Seurat/Signac WNN vignette - https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis.html#wnn-analysis-of-10x-multiome-rna-atac

# Load libraries
```{r}
library(Seurat)
library(Signac)
library(ggplot2)
library(dplyr)
library(EnsDb.Hsapiens.v86)
library(biovizBase)
library(patchwork)
library(SoupX)

# Needed to change maximum allowable global size in order for this to run
options(future.globals.maxSize=1000000000000000)
```
# Load data via SoupX
```{r}
p01_190 <- load10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/01-190/outs")
p04_003 <- load10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/outs")
p01_076 <- load10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/01-076/outs")
p01_115 <- load10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/01-115/outs")
p01_131 <- load10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/01-131/outs")
p01_163 <- load10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/01-163/outs")
p04_006 <- load10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-006/outs")
```
# SoupX for removal of ambient RNA in droplets
# rho = contamination fraction so rho=1 = 100% of UMIs in a droplet are ambient

```{r}
# 01-190
p01_190 <- autoEstCont(p01_190)
p01_190_rna <- adjustCounts(p01_190)
# 04-003
p04_003 <- autoEstCont(p04_003)
p04_003_rna <- adjustCounts(p04_003)
# 01-076
p01_076 <- autoEstCont(p01_076)
p01_076_rna <- adjustCounts(p01_076)
# 01-115
p01_115 <- autoEstCont(p01_115)
p01_115_rna <- adjustCounts(p01_115)
# 01-131
p01_131 <- autoEstCont(p01_131)
p01_131_rna <- adjustCounts(p01_131)
# 01-163
p01_163 <- autoEstCont(p01_163)
p01_163_rna <- adjustCounts(p01_163)
# 04-006
p04_006 <- autoEstCont(p04_006)
p04_006_rna <- adjustCounts(p04_006)
```
# Trying explicit bcuz 01-163 and 04-006 failed
```{r}
p01_163 <- setContaminationFraction(p01_163, 0.8)

head(p01_163$soupProfile[order(p01_163$soupProfile$est, decreasing = TRUE), ], n = 20)
plotMarkerDistribution(p01_163)
contam_genes <- "PLCG2"
useToEst = estimateNonExpressingCells(p01_163, nonExpressedGeneList = list(IG = contam_genes), 
    clusters = FALSE)

plotMarkerMap(p01_163, geneSet = contam_genes, useToEst = useToEst)


```


# Separate RNA and ATAC counts
```{r}
p1.rna.counts <- p1.data$`Gene Expression`
# p1.atac.counts <- p1.data$Peaks
# 
# p2.rna.counts <- p2.data$`Gene Expression`
# p2.atac.counts <- p2.data$Peaks
# 
# p3.rna.counts <- p3.data$`Gene Expression`
# p3.atac.counts <- p3.data$Peaks
# 
# p4.rna.counts <- p4.data$`Gene Expression`
# p4.atac.counts <- p4.data$Peaks
# 
# p5.rna.counts <- p5.data$`Gene Expression`
# p5.atac.counts <- p5.data$Peaks
# 
# p6.rna.counts <- p6.data$`Gene Expression`
# p6.atac.counts <- p6.data$Peaks
# 
# p7.rna.counts <- p7.data$`Gene Expression`
# p7.atac.counts <- p7.data$Peaks
```
# Create Seurat objects
# Add mito % for filtering
```{r}
#p1 <- CreateSeuratObject(counts = p1.rna.counts)
#p2 <- CreateSeuratObject(counts = p2.rna.counts)
# p3 <- CreateSeuratObject(counts = p3.rna.counts)
# p4 <- CreateSeuratObject(counts = p4.rna.counts)
# p5 <- CreateSeuratObject(counts = p5.rna.counts)
# p6 <- CreateSeuratObject(counts = p6.rna.counts)
# p7 <- CreateSeuratObject(counts = p7.rna.counts)
# 
# #p1[["percent.mt"]] <- PercentageFeatureSet(p1, pattern = "^MT-")
# #p2[["percent.mt"]] <- PercentageFeatureSet(p2, pattern = "^MT-")
# p3[["percent.mt"]] <- PercentageFeatureSet(p3, pattern = "^MT-")
# p4[["percent.mt"]] <- PercentageFeatureSet(p4, pattern = "^MT-")
# p5[["percent.mt"]] <- PercentageFeatureSet(p5, pattern = "^MT-")
# p6[["percent.mt"]] <- PercentageFeatureSet(p6, pattern = "^MT-")
# p7[["percent.mt"]] <- PercentageFeatureSet(p7, pattern = "^MT-")

```
# Add in ATAC data
```{r}
# P1
# p1.grange.counts <- StringToGRanges(rownames(p1.atac.counts), sep = c(":", "-"))
# p1.grange.use <- seqnames(p1.grange.counts) %in% standardChromosomes(p1.grange.counts)
# p1.atac.counts <- p1.atac.counts[as.vector(p1.grange.use),]
# p1.annotations <- GetGRangesFromEnsDb((ensdb = EnsDb.Hsapiens.v86))
# seqlevelsStyle(p1.annotations) <- 'UCSC'
# genome(p1.annotations) <- "hg38"
# 
# p1.chrom.assay <- CreateChromatinAssay(
#    counts = p1.atac.counts,
#    sep = c(":", "-"),
#    genome = 'hg38',
#    #fragments = frag.file,
#    min.cells = 10,
#    annotation = p1.annotations
#  )
# 
# p1[["ATAC"]] <- p1.chrom.assay
# 
# # P2
# p2.grange.counts <- StringToGRanges(rownames(p2.atac.counts), sep = c(":", "-"))
# p2.grange.use <- seqnames(p2.grange.counts) %in% standardChromosomes(p2.grange.counts)
# p2.atac.counts <- p2.atac.counts[as.vector(p2.grange.use),]
# p2.annotations <- GetGRangesFromEnsDb((ensdb = EnsDb.Hsapiens.v86))
# seqlevelsStyle(p2.annotations) <- 'UCSC'
# genome(p2.annotations) <- "hg38"
# 
# p2.chrom.assay <- CreateChromatinAssay(
#    counts = p2.atac.counts,
#    sep = c(":", "-"),
#    genome = 'hg38',
#    #fragments = frag.file,
#    min.cells = 10,
#    annotation = p2.annotations
#  )
# 
# p2[["ATAC"]] <- p2.chrom.assay
```
# Now some QC
```{r}
# VlnPlot(p1, features = c("nCount_ATAC", "nCount_RNA","percent.mt"), ncol = 3,
#   log = TRUE, pt.size = 0) + NoLegend()
# 
# VlnPlot(p2, features = c("nCount_ATAC", "nCount_RNA","percent.mt"), ncol = 3,
#   log = TRUE, pt.size = 0) + NoLegend()
```
# Basic quality filtering based on...idk what? The WNN vignette lol. Adjusted these somewhat after looking at our QC values
# Specifically, I adjusted minimum ATAC count to 5e2 from 5e3 and minimum RNA from 1000 to 500
```{r}
# p1.before <- ncol(p1)
# p2.before <- ncol(p2)
# 
# p1 <- subset(
#   x = p1,
#   subset = nCount_ATAC < 7e4 &
#     nCount_ATAC > 5e2 &
#     nCount_RNA < 25000 &
#     nCount_RNA > 500 &
#     percent.mt < 20
# )
# 
# p2 <- subset(
#   x = p2,
#   subset = nCount_ATAC < 7e4 &
#     nCount_ATAC > 5e2 &
#     nCount_RNA < 25000 &
#     nCount_RNA > 500 &
#     percent.mt < 20
# )
# 
# p1.after <- ncol(p1)
# p2.after <- ncol(p2)
# 
# print(paste0("p1 cell count before = ", p1.before, " and p1 after = ", p1.after))
# print(paste0("p1 cell loss = ", round(1 - p1.after/p1.before, 3)*100, "%"))
# 
# print(paste0("p2 cell count before = ", p2.before, " and p2 after = ", p2.after))
# print(paste0("p2 cell loss = ", round(1 - p2.after/p2.before, 3)*100, "%"))
```
# Standard workflow
```{r}
# DefaultAssay(p1) <- "RNA"
# p1 <- SCTransform(p1, verbose = FALSE) %>% RunPCA() %>% RunUMAP(dims = 1:20, reduction.name = 'umap.rna', reduction.key = 'rnaUMAP_')
# 
# DefaultAssay(p1) <- "ATAC"
# p1 <- RunTFIDF(p1)
# p1 <- FindTopFeatures(p1, min.cutoff = 'q0')
# p1 <- RunSVD(p1)
# p1 <- RunUMAP(p1, reduction = 'lsi', dims = 2:20, reduction.name = "umap.atac", reduction.key = "atacUMAP_")
# 
# DefaultAssay(p2) <- "RNA"
# p2 <- SCTransform(p2, verbose = FALSE) %>% RunPCA() %>% RunUMAP(dims = 1:20, reduction.name = 'umap.rna', reduction.key = 'rnaUMAP_')
# 
# DefaultAssay(p2) <- "ATAC"
# p2 <- RunTFIDF(p2)
# p2 <- FindTopFeatures(p2, min.cutoff = 'q0')
# p2 <- RunSVD(p2)
# p2 <- RunUMAP(p2, reduction = 'lsi', dims = 2:20, reduction.name = "umap.atac", reduction.key = "atacUMAP_")
```
# Calculate WNN graph
```{r}
# p1 <- FindMultiModalNeighbors(p1, reduction.list = list("pca", "lsi"), dims.list = list(1:20, 2:20))
# p1 <- RunUMAP(p1, nn.name = "weighted.nn", reduction.name = "wnn.umap", reduction.key = "wnnUMAP_")
# p1 <- FindClusters(p1, graph.name = "wsnn", algorithm = 3, verbose = FALSE)
# 
# p2 <- FindMultiModalNeighbors(p2, reduction.list = list("pca", "lsi"), dims.list = list(1:20, 2:20))
# p2 <- RunUMAP(p2, nn.name = "weighted.nn", reduction.name = "wnn.umap", reduction.key = "wnnUMAP_")
# p2 <- FindClusters(p2, graph.name = "wsnn", algorithm = 3, verbose = FALSE)
```
# LOOK!
```{r}
# plot1 <- DimPlot(p1, reduction = "umap.rna", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("RNA")
# plot2 <- DimPlot(p1, reduction = "umap.atac", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("ATAC")
# plot3 <- DimPlot(p1, reduction = "wnn.umap", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("WNN")
# plot1 + plot2 + plot3 & NoLegend() & theme(plot.title = element_text(hjust = 0.5))
# ```
# ```{r}
# plot12 <- DimPlot(p2, reduction = "umap.rna", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("RNA")
# plot22 <- DimPlot(p2, reduction = "umap.atac", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("ATAC")
# plot32 <- DimPlot(p2, reduction = "wnn.umap", label = TRUE, label.size = 2.5, repel = TRUE) + ggtitle("WNN")
# plot12 + plot22 + plot32 & NoLegend() & theme(plot.title = element_text(hjust = 0.5))
# ```
# ```{r}
# p1.markers <- FindAllMarkers(p1, only.pos = TRUE, assay = "RNA", min.pct = 0.25)
# p1.markers <- p1.markers[p1.markers$p_val_adj < 0.05,]
# p1.markers <- p1.markers[order(-p1.markers$avg_log2FC),]
# 
# p2.markers <- FindAllMarkers(p2, only.pos = TRUE, assay = "RNA", min.pct = 0.25, logfc.threshold = 0.25)
# p2.markers <- p2.markers[order(-p2.markers$avg_log2FC),]
# 
# # Markers which define clusters 0, 1, 2
# cluster123.markers <- FindMarkers(p1, ident.1 = c(0, 1, 2, 9), ident.2 = c(3, 4, 5, 6, 7, 8))
# cluster123.markers <- cluster123.markers[order(cluster123.markers$avg_log2FC),]
# 
# DimPlot(p1, reduction = "umap.rna")
```
# Work zone
```{r}
# p1.markers.cluster6 <- p1.markers[p1.markers$cluster == "4",]
# p1.markers.cluster6 <- p1.markers.cluster6[order(p1.markers.cluster6$p_val_adj),]
# p1.markers.cluster6
# 
# cluster123.markers
# ```
# ```{r}
# DefaultAssay(p1) <- "RNA"
# FeaturePlot(p1, features = c("CD19", "CD20", "MS4A1", "CD79A", "IGHM"))
# FeaturePlot(p1, features = c("DUSP5", "MZB1", "IGF1"))
# FeaturePlot(p1, features = c("ARID1A", "CD79B", "BCL2", "CXCR4"))
# FeaturePlot(p1, features = c("PDE4D"))
# 
# # ARID1A, CD79B, NFkappa B, BCL2 type genes, CXCR4
# 
# DefaultAssay(p2) <- "RNA"
# FeaturePlot(p2, features = c("CD19", "CD20", "MS4A1", "CD79A", "IGHM"))
# FeaturePlot(p2, features = c("PDE4D"))
```

# RNA only
# p1
```{r}
#p1.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/filtered_feature_bc_matrix")
p1.rna.counts <- p1.data$`Gene Expression`
p1 <- CreateSeuratObject(counts = p1.rna.counts)
p1[["percent.mt"]] <- PercentageFeatureSet(p1, pattern = "^MT-")

plot1 <- VlnPlot(p1, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 500, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot2 <- VlnPlot(p1, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 300, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot3 <- VlnPlot(p1, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = 20, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
p1.qc.patch <- plot1 | plot2 | plot3
#p1.qc.patch
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/p1_BasicQC_Metrics.png", height = 4, width = 8)

p1.before <- ncol(p1)

p1 <- subset(
  x = p1,
  subset = nCount_RNA < 25000 &
    nCount_RNA > 500 &
    nFeature_RNA > 300 &
    percent.mt < 20
)

p1.after <- ncol(p1)

print(paste0("p1 cell count before = ", p1.before, " and p1 after = ", p1.after))
print(paste0("p1 cell loss = ", round(1 - p1.after/p1.before, 3)*100, "%"))
```

# p2
```{r}
#p2.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/filtered_feature_bc_matrix")
p2.rna.counts <- p2.data$`Gene Expression`
p2 <- CreateSeuratObject(counts = p2.rna.counts)
p2[["percent.mt"]] <- PercentageFeatureSet(p2, pattern = "^MT-")

plot1 <- VlnPlot(p2, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 500, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot2 <- VlnPlot(p2, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 300, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot3 <- VlnPlot(p2, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = 20, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
p2.qc.patch <- plot1 | plot2 | plot3
#p2.qc.patch
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/p2_BasicQC_Metrics.png", height = 4, width = 8)

p2.before <- ncol(p2)

p2 <- subset(
  x = p2,
  subset = nCount_RNA < 25000 &
    nCount_RNA > 500 &
    nFeature_RNA > 300 &
    percent.mt < 20
)

p2.after <- ncol(p2)

print(paste0("p2 cell count before = ", p2.before, " and p2 after = ", p2.after))
print(paste0("p2 cell loss = ", round(1 - p2.after/p2.before, 3)*100, "%"))
```
# P3
```{r}
#p3.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/filtered_feature_bc_matrix")
p3.rna.counts <- p3.data$`Gene Expression`
p3 <- CreateSeuratObject(counts = p3.rna.counts)
p3[["percent.mt"]] <- PercentageFeatureSet(p3, pattern = "^MT-")

plot1 <- VlnPlot(p3, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 500, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot2 <- VlnPlot(p3, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 300, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot3 <- VlnPlot(p3, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = 20, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
p3.qc.patch <- plot1 | plot2 | plot3
#p3.qc.patch
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/p3_BasicQC_Metrics.png", height = 4, width = 8)

p3.before <- ncol(p3)

p3 <- subset(
  x = p3,
  subset = nCount_RNA < 25000 &
    nCount_RNA > 500 &
    nFeature_RNA > 300 &
    percent.mt < 20
)

p3.after <- ncol(p3)

print(paste0("p3 cell count before = ", p3.before, " and p3 after = ", p3.after))
print(paste0("p3 cell loss = ", round(1 - p3.after/p3.before, 3)*100, "%"))
```
# P4
```{r}
#p3.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/filtered_feature_bc_matrix")
p4.rna.counts <- p4.data$`Gene Expression`
p4 <- CreateSeuratObject(counts = p4.rna.counts)
p4[["percent.mt"]] <- PercentageFeatureSet(p4, pattern = "^MT-")

plot1 <- VlnPlot(p4, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 500, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot2 <- VlnPlot(p4, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 300, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot3 <- VlnPlot(p4, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = 20, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
p4.qc.patch <- plot1 | plot2 | plot3
#p4.qc.patch
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/p4_BasicQC_Metrics.png", height = 4, width = 8)

p4.before <- ncol(p4)

p4 <- subset(
  x = p4,
  subset = nCount_RNA < 25000 &
    nCount_RNA > 500 &
    nFeature_RNA > 300 &
    percent.mt < 20
)

p4.after <- ncol(p4)

print(paste0("p4 cell count before = ", p4.before, " and p4 after = ", p4.after))
print(paste0("p4 cell loss = ", round(1 - p4.after/p4.before, 3)*100, "%"))
```
# P5
```{r}
#p5.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/filtered_feature_bc_matrix")
p5.rna.counts <- p5.data$`Gene Expression`
p5 <- CreateSeuratObject(counts = p5.rna.counts)
p5[["percent.mt"]] <- PercentageFeatureSet(p5, pattern = "^MT-")

plot1 <- VlnPlot(p5, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 500, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot2 <- VlnPlot(p5, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 300, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot3 <- VlnPlot(p5, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = 20, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
p5.qc.patch <- plot1 | plot2 | plot3
#p5.qc.patch
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/p5_BasicQC_Metrics.png", height = 4, width = 8)

p5.before <- ncol(p5)

p5 <- subset(
  x = p5,
  subset = nCount_RNA < 25000 &
    nCount_RNA > 500 &
    nFeature_RNA > 300 &
    percent.mt < 20
)

p5.after <- ncol(p5)

print(paste0("p5 cell count before = ", p5.before, " and p5 after = ", p5.after))
print(paste0("p5 cell loss = ", round(1 - p5.after/p5.before, 3)*100, "%"))
```
# P6
```{r}
#p6.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/filtered_feature_bc_matrix")
p6.rna.counts <- p6.data$`Gene Expression`
p6 <- CreateSeuratObject(counts = p6.rna.counts)
p6[["percent.mt"]] <- PercentageFeatureSet(p6, pattern = "^MT-")

plot1 <- VlnPlot(p6, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 500, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot2 <- VlnPlot(p6, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 300, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot3 <- VlnPlot(p6, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = 20, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
p6.qc.patch <- plot1 | plot2 | plot3
#p6.qc.patch
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/p6_BasicQC_Metrics.png", height = 4, width = 8)

p6.before <- ncol(p6)

p6 <- subset(
  x = p6,
  subset = nCount_RNA < 25000 &
    nCount_RNA > 500 &
    nFeature_RNA > 300 &
    percent.mt < 20
)

p6.after <- ncol(p6)

print(paste0("p6 cell count before = ", p6.before, " and p6 after = ", p6.after))
print(paste0("p6 cell loss = ", round(1 - p6.after/p6.before, 3)*100, "%"))
```
# P7
```{r}
#p7.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/04-003/filtered_feature_bc_matrix")
p7.rna.counts <- p7.data$`Gene Expression`
p7 <- CreateSeuratObject(counts = p7.rna.counts)
p7[["percent.mt"]] <- PercentageFeatureSet(p7, pattern = "^MT-")

plot1 <- VlnPlot(p7, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 500, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot2 <- VlnPlot(p7, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = 300, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
plot3 <- VlnPlot(p7, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = 20, 
                                                                                    color = "black", 
                                                                                    linewidth = 1, 
                                                                                    linetype = "dashed") + NoLegend()
p7.qc.patch <- plot1 | plot2 | plot3
#p7.qc.patch
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/p7_BasicQC_Metrics.png", height = 4, width = 8)

p7.before <- ncol(p7)

p7 <- subset(
  x = p7,
  subset = nCount_RNA < 25000 &
    nCount_RNA > 500 &
    nFeature_RNA > 300 &
    percent.mt < 20
)

p7.after <- ncol(p7)

print(paste0("p7 cell count before = ", p7.before, " and p7 after = ", p7.after))
print(paste0("p7 cell loss = ", round(1 - p7.after/p7.before, 3)*100, "%"))
```
# Bar plots of cell filtering
```{r}
filtering.df <- data.frame(matrix(ncol = 3, nrow = 7))
colnames(filtering.df) <- c("patient", "cells.before", "cells.after")

patient.ids <- c("01-190", "04-003", "01-131", "01-076", "01-115", "01-163", "04-006")
cells.before <- c(p1.before, p2.before, p3.before, p4.before, p5.before, p6.before, p7.before)
cells.after <- c(p1.after, p2.after,p3.after, p4.after, p5.after, p6.after, p7.after)

filtering.df$patient <- patient.ids
filtering.df$cells.before <- cells.before
filtering.df$cells.after <- cells.after

filtering.df

# Melt df for grouped bar plot
library(reshape2)
melt.df <- melt(filtering.df, id.vars = "patient")

ggplot(melt.df, aes(fill=variable, y=value, x=patient)) + 
    geom_bar(position="dodge", stat="identity") + 
  ylab("Cells") + xlab("Patient") + theme_minimal()
#ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM_scRNA_2ndBatch_FilteringCellLoss_BarPlots.png", height = 4, width = 8)

filtering.df$`loss%` <- (1 - cells.after/cells.before) * 100
filtering.df
```

# Totals
```{r}
sum(cells.before)
sum(cells.after)
```
# Merge
```{r}
p1@meta.data$patient <- "01_190"
p2@meta.data$patient <- "04_003"
p3@meta.data$patient <- "01_131"
p4@meta.data$patient <- "01_076"
p5@meta.data$patient <- "01_115"
p6@meta.data$patient <- "01_163"
p7@meta.data$patient <- "04_006"
merged <- merge(x = p1, y = c(p2, p3, p4, p5, p6, p7))
```

# Now Standard Workflow stuff
```{r}
DefaultAssay(merged) <- "RNA"
merged <- NormalizeData(merged)
merged <- FindVariableFeatures(merged, nfeatures = 2000)
merged <- ScaleData(merged)
merged <- RunPCA(merged, npcs = 25)
```
# Determine point where PCs begin to elbow
# Selects the larger value of:
# 1. the point where the PC only contribute 5% of stdev and PCs cumulatively contribute 90% of the stdev (co1)
# 2. the point where the % change in variation between consecutive PCs is less than 0.1% (co2)
```{r}
ElbowPlot(merged, ndims = 25)
# Determine percent of variation associated with each PC
pct <- merged[["pca"]]@stdev / sum(merged[["pca"]]@stdev) * 100

# Calculate cumulative percents for each PC
cumu <- cumsum(pct)

# Determine which PC exhibits cumulative percent greater than 90% and % variation associated with the PC as less than 5
co1 <- which(cumu > 90 & pct < 5)[1]

# Determine the difference between variation of PC and subsequent PC
co2 <- sort(which((pct[1:length(pct) - 1] - pct[2:length(pct)]) > 0.1), decreasing = T)[1] + 1

# Minimum of the two calculation
pcs <- min(co1, co2)

print(paste0("the optimal PC point is: ", pcs))
```
# 
```{r}
merged <- RunPCA(merged, npcs = 13)
merged <- FindNeighbors(merged, dims = 1:13)
merged <- FindClusters(merged, resolution = 0.5)
merged <- RunUMAP(merged, dims = 1:13)
DimPlot(merged)
#DimPlot(merged, group.by = "patient")
```
# Save out object
```{r}
saveRDS(merged, "/Users/gagled01/morganLab/Waldenstroms/singlecell/WM_Prelim_RNAonly_Merged.rds")
```

# This object was run thru celltypist with low resolution immune reference
```{r}
#annotated.merged <- readRDS("/Users/gagled01/morganLab/Waldenstroms/singlecell/WM_Prelim_RNAonly_CellTypistAnnotated.rds")

# alternative, better way to do it which makes a file with metadata saved. involves exporting the celltypist CSV and then merging it with the existing seurat object.

celltypist.csv <- read.csv(file = "/Users/gagled01/morganLab/Waldenstroms/singlecell/WM_Prelim_CellTypistAnnotationspredicted_labels.csv")
rownames(celltypist.csv) <- celltypist.csv[,1]

merged.annotated <- AddMetaData(object = merged, metadata = celltypist.csv)
head(merged.annotated@meta.data)
saveRDS(merged.annotated, "/Users/gagled01/morganLab/Waldenstroms/singlecell/WM_Prelim_RNAonly_CellTypistAnnotated_AllMetadata.rds")
```
# Make dimplots
```{r}
DimPlot(merged.annotated, group.by = "majority_voting")
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM_Prelim_CellTypistAnnotated_UMAP_Seurat.png", height = 6, width = 8)

DimPlot(merged.annotated, group.by = "patient")
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM_Prelim_CellTypistAnnotated_UMAP_Patient.png", height = 6, width = 8)

table(merged.annotated@meta.data$majority_voting)
```

```{r}
plasma.cells <- subset(merged.annotated, subset = majority_voting == "Plasma cells")
DimPlot(plasma.cells, group.by = "patient")

table(plasma.cells@meta.data$patient)
```
# Dirty way to get celltype relative abundance by patient dataframe for stacked barplots
```{r}
### Split by method
patient_split <- SplitObject(merged.annotated, split.by = "patient")
p1 <- patient_split[[1]]
p2 <- patient_split[[2]]
p3 <- patient_split[[3]]
p4 <- patient_split[[4]]
p5 <- patient_split[[5]]
p6 <- patient_split[[6]]
p7 <- patient_split[[7]]

# Cell counts by patient
table(p1@meta.data$majority_voting)
table(p2@meta.data$majority_voting)
table(p3@meta.data$majority_voting)
table(p4@meta.data$majority_voting)
table(p5@meta.data$majority_voting)
table(p6@meta.data$majority_voting)
table(p7@meta.data$majority_voting)

# Relative abundance by patient
p1_relabun <- data.frame(table(p1@meta.data$majority_voting)/sum(table(p1@meta.data$majority_voting)) * 100)
p2_relabun <- data.frame(table(p2@meta.data$majority_voting)/sum(table(p2@meta.data$majority_voting)) * 100)
p3_relabun <- data.frame(table(p3@meta.data$majority_voting)/sum(table(p3@meta.data$majority_voting)) * 100)
p4_relabun <- data.frame(table(p4@meta.data$majority_voting)/sum(table(p4@meta.data$majority_voting)) * 100)
p5_relabun <- data.frame(table(p5@meta.data$majority_voting)/sum(table(p5@meta.data$majority_voting)) * 100)
p6_relabun <- data.frame(table(p6@meta.data$majority_voting)/sum(table(p6@meta.data$majority_voting)) * 100)
p7_relabun <- data.frame(table(p7@meta.data$majority_voting)/sum(table(p7@meta.data$majority_voting)) * 100)

# Absolute abundance by patient
p1_relabun$abs <- as.numeric(table(p1@meta.data$majority_voting))
p2_relabun$abs <- as.numeric(table(p2@meta.data$majority_voting))
p3_relabun$abs <- as.numeric(table(p3@meta.data$majority_voting))
p4_relabun$abs <- as.numeric(table(p4@meta.data$majority_voting))
p5_relabun$abs <- as.numeric(table(p5@meta.data$majority_voting))
p6_relabun$abs <- as.numeric(table(p6@meta.data$majority_voting))
p7_relabun$abs <- as.numeric(table(p7@meta.data$majority_voting))

p1_relabun$patient <- "01_190"
p2_relabun$patient <- "04_003"
p3_relabun$patient <- "01_131"
p4_relabun$patient <- "01_076"
p5_relabun$patient <- "01_115"
p6_relabun$patient <- "01_163"
p7_relabun$patient <- "04_006"

# bind by rows
allpatient_df <- rbind(p1_relabun, p2_relabun, p3_relabun, p4_relabun, p5_relabun, p6_relabun, p7_relabun)
colnames(allpatient_df) <- c("celltype", "relative", "patient", "absolute")

p2@meta.data$patient[1]
p3@meta.data$patient[1]
p4@meta.data$patient[1]
p5@meta.data$patient[1]
p6@meta.data$patient[1]
p7@meta.data$patient[1]
```
# Stacked bar plot of cell type relative abundances
```{r}
# Stacked Relative Abundance
ggplot(allpatient_df, aes(fill=celltype, y=relative, x=patient)) + 
    geom_bar(position="stack", stat="identity") + 
  xlab(label = "Patient") + ylab("Relative Abundance") +
  theme(axis.text.x = element_text(size = "10")) +
theme_minimal()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM_Prelim_CellTypistAnnotated_StackedBarPlot.png", width = 8, height = 8)

# Stacked Absolute Abundance
ggplot(allpatient_df, aes(fill=celltype, y=absolute, x=patient)) + 
    geom_bar(position="stack", stat="identity") + 
  xlab(label = "Patient") + ylab("Absolute Abundance") +
  theme(axis.text.x = element_text(size = "10")) +
theme_minimal()
ggsave("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/WM_Prelim_CellTypistAnnotated_StackedBarPlot_AbsoluteCounts.png", width = 8, height = 8)

allpatient_df[allpatient_df$celltype == "Plasmablasts",]
```

```{r}
tabula_h5 <- readH5AD("/Users/gagled01/Downloads/TS_immune.h5ad", verbose = TRUE)
tabula_immune <- as.Seurat(tabula_h5, counts = "X", data = NULL)

head(tabula_bm@meta.data)

DimPlot(tabula_bm, group.by = "cell_ontology_class")
DimPlot(tabula_bm, group.by = "free_annotation")
```
#
```{r}
tabula_immune_h5 <- readH5AD(file = "/Users/gagled01/Downloads/TS_immune.h5ad")
tabula_immune <- as.Seurat(tabula_immune_h5, counts = "X", data = NULL)

head(tabula_immune@meta.data)

DimPlot(tabula_immune, group.by = "cell_ontology_class")
DimPlot(tabula_immune, group.by = "free_annotation")
```


#
#
####
# Get markers
```{r}
Idents(merged.annotated) <- "majority_voting"
markers <- FindAllMarkers(merged.annotated, logfc.threshold = 0.25)

plasma.markers <- markers[markers$cluster == "Plasma cells",]
plasma.markers <- plasma.markers[order(plasma.markers$p_val_adj),]
memoryB.markers <- markers[markers$cluster == "Memory B cells",]
memoryB.markers <- memoryB.markers[order(memoryB.markers$p_val_adj),]
ageB.markers <- markers[markers$cluster == "Age-associated B cells",]
ageB.markers <- ageB.markers[order(ageB.markers$p_val_adj),]
naiveB.markers <- markers[markers$cluster == "Naive B cells",]
naiveB.markers <- naiveB.markers[order(naiveB.markers$p_val_adj),]
monocyte.markers <- markers[markers$cluster == "Classical monocytes",]
monocyte.markers <- monocyte.markers[order(monocyte.markers$p_val_adj),]
proB.markers <- markers[markers$cluster == "Pro-B cells",]
proB.markers <- proB.markers[order(proB.markers$p_val_adj),]

head(plasma.markers, n = 10)
head(memoryB.markers, n = 10)
head(ageB.markers, n = 10)
head(naiveB.markers, n = 10)
head(monocyte.markers, n = 10)
head(proB.markers, n = 10)
```
# Make a heatmap
```{r}
library(dplyr)

output_gene_order <- c()

for(i in c("Memory B cells", "Plasma cells", "Age-associated B cells", "Naive B cells", "Classical monocytes", "Pro-B cells")) {
  print(i)

  cluster_genes <- markers %>% dplyr::filter(cluster == i) %>% dplyr::arrange(p_val_adj) %>%
                      dplyr::filter(avg_log2FC > 0) %>%
                      dplyr::select(gene) %>%
                      head(n = 5) %>%
                      purrr::as_vector()
  print(cluster_genes)

  output_gene_order <- c(output_gene_order, cluster_genes)
}

output_gene_order <- unique(output_gene_order)

matrix <- merged.annotated@assays$RNA@scale.data
#matrix.subset <- matrix[rownames(matrix) %in% c(top100_pos_lfc_genes, top100_neg_lfc_genes),]
matrix.roworder.subset <- matrix[rownames(matrix) %in% output_gene_order,]

output_gene_order_sub <- output_gene_order[output_gene_order %in% rownames(matrix.roworder.subset)]

annotation <- HeatmapAnnotation("CellTypist Annotation" = merged.annotated@meta.data$majority_voting,
                                simple_anno_size = unit(3, "mm"),
                                col = list("CellTypist Annotation" = c("Memory B cells" = "#F8766D",
                                                         "Plasma cells" = "#B79F00",
                                                         "Age-associated B cells" = "#00BA38",
                                                         "Naive B cells" = "#00BFC4",
                                                         "Classical monocytes" = "#619CFF",
                                                         "Pro-B cells" = "#F564E3"
                                                         )))

col_fun = colorRamp2(c(-2, 0, 2), c("blue", "white", "red"))

# paste into terminal
Heatmap(matrix.roworder.subset, cluster_rows = F, cluster_columns = TRUE,
        row_order = output_gene_order_sub,
        show_column_dend = F,
        col = col_fun,
        column_names_gp = gpar(fontsize = 0), row_names_gp = gpar(fontsize = 4.5),
        clustering_distance_columns = "euclidean", clustering_method_columns = "average",
        column_dend_height = unit(40,"mm"), row_dend_width = unit(40, "mm"),
        top_annotation = annotation,
        heatmap_legend_param = list(title = "Scaled Expression")
        )
```




