---
title: "WM_ArchR_Analysis"
output: html_document
date: "2023-06-27"
---
Load libraries, set threads, set reference genome, set global size so we don't crash (hopefully)
```{r}
library(ArchR)
library(Seurat)
addArchRThreads(threads = 1)
#addArchRGenome("hg38")
options(future.globals.maxSize=100000000000000000000000) 
```
Set up fragment files
```{r}
setwd("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/ATAC_fragments/ArchR_ReferenceAdd_Fragments")

# hg38 fragment files
fragments <- list.files(getwd())
#fragments <- fragments[-c(8)] # exclude the non-fragment files like the R Notebook, etc.
fragments
```
Load existing ArchR project
```{r}
archR <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/singlecell/data/ATAC_fragments/WM_ReferenceAdd_Unconstrained_Integration_NoGreenleafCOVID_ReclusteredHarmony")
```
Create arrows
```{r}
arrow.files <- createArrowFiles(
  inputFiles = fragments,
  sampleNames = as.character(fragments),
  minTSS = 4,
  minFrags = 1000,
  addTileMat = T,
  addGeneScoreMat = T
)

# In case the arrow files already exist
#arrow.files <- list.files(pattern = "\\.arrow$")
```
# Doublet inference
# Doublets are inevitable in single cell data. The % of doublets is proportional to the number of cells loaded into the experiment. 
# Doublets are especially problematic in the context of developmental/trajectory data because doublets can look like a mixture between 2 cell types which can be confounded with intermediate cell states. THIS WM DATA IS B CELL LINEAGE...SO TRAJECTORY!!! 
# To predict doublets, ArchR mixes the reads from large combos of cells to create synthetic doublets and projects them into a UMAP. Iterating this process 1000x times allows identification of cells in the REAL data which look like synthetic doublets.
# We want to see R values greater than 0.9, which implies that the cells within each synthetic UMAP have significant heterogeneity, which we want.
# In the output plots, use the doublet enrichment plot 
```{r}
dub.scores <- addDoubletScores(
  input = arrow.files,
  k = 10,
  knnMethod = "UMAP",
  LSIMethod = 1
) 
```
########################################################
###### CHAPTER 3 - Create ArchRProject and QC ##########
########################################################
```{r}
archR <- ArchRProject(
  ArrowFiles = arrow.files,
  outputDirectory = "WM_ReferenceData_Update",
  copyArrows = F
)
```
# Clean up sample names and add metadata
```{r}
clean.cellnames <- gsub("atac_fragments.tsv.gz#", "", archR$cellNames)
archR$cellNames_clean <- clean.cellnames

# Iteratively clean up sample names
clean.samplenames <- gsub("_atac_fragments.tsv.gz", "", archR$Sample)
archR$samplesClean <- clean.samplenames
clean.samplenames2 <- gsub("_fragments.tsv.bgz", "", archR$samplesClean)
archR$samplesClean <- clean.samplenames2
clean.samplenames3 <- gsub("_fragments.tsv.gz", "", archR$samplesClean)
archR$samplesClean <- clean.samplenames3
clean.samplenames4 <- gsub("GSM........", "", archR$samplesClean)
archR$samplesClean <- clean.samplenames4
clean.samplenames5 <- gsub("tonsil_1.", "stanford_1", archR$samplesClean)
archR$samplesClean <- clean.samplenames5
clean.samplenames6 <- gsub("tonsil_2.", "stanford_2", archR$samplesClean)
archR$samplesClean <- clean.samplenames6
clean.samplenames7 <- gsub("tonsil_3.", "stanford_3", archR$samplesClean)
archR$samplesClean <- clean.samplenames7
clean.samplenames8 <- gsub(".fragments.tsv.gz", "", archR$samplesClean)
archR$samplesClean <- clean.samplenames8
clean.samplenames9 <- gsub("scATAC_", "", archR$samplesClean)
archR$samplesClean <- clean.samplenames9
clean.samplenames10 <- gsub("_sorted_10k", "", archR$samplesClean)
archR$samplesClean <- clean.samplenames10
clean.samplenames11 <- gsub("10k_PBMC_Multiome_nextgem_Chromium_X", "pbmc_noSort", archR$samplesClean)
archR$samplesClean <- clean.samplenames11
table(archR$samplesClean)

# Assign info about data source
sources <- gsub("stanford_.", "Stanford", archR$samplesClean)
archR$source <- sources
sources2 <- gsub("BCP00.", "QMUL", archR$source)
archR$source <- sources2
sources3 <- gsub("0.-...", "NYU", archR$source)
archR$source <- sources3
case.control4 <- gsub("WM..", "Gustave-Roussy", archR$source)
archR$source <- case.control4
case.control5 <- gsub("BMMC_D...", "Greenleaf_MAPL", archR$source)
archR$source <- case.control5
case.control6 <- gsub("PBMC_D1...", "Greenleaf_MAPL", archR$source)
archR$source <- case.control6
case.control7 <- gsub("HIP0..", "Greenleaf_COVID", archR$source)
archR$source <- case.control7
case.control8 <- gsub("EV08", "Greenleaf_COVID", archR$source)
archR$source <- case.control8
case.control9 <- gsub("pbmc_granulocyte", "10X_pbmc", archR$source)
archR$source <- case.control9
case.control10 <- gsub("pbmc_noSort", "10X_pbmc", archR$source)
archR$source <- case.control10
table(archR$source)

# And case/control data
condition <- gsub("0.-...", "WM", archR$samplesClean)
archR$condition <- condition
condition2 <- gsub("BCP00.", "HD", archR$condition)
archR$condition <- condition2
condition3 <- gsub("stanford_.", "HD", archR$condition)
archR$condition <- condition3
condition4 <- gsub("WM..", "WM", archR$condition)
archR$condition <- condition4
condition5 <- gsub("BMMC_D...", "HD", archR$condition)
archR$condition <- condition5
condition6 <- gsub("PBMC_D1...", "HD", archR$condition)
archR$condition <- condition6
condition7 <- gsub("HIP0..", "HD", archR$condition)
archR$condition <- condition7
condition8 <- gsub("EV08", "HD", archR$condition)
archR$condition <- condition8
condition9 <- gsub("pbmc_granulocyte", "HD", archR$condition)
archR$condition <- condition9
condition10 <- gsub("pbmc_noSort", "HD", archR$condition)
archR$condition <- condition10
table(archR$condition)

# Tissue
tissue <- gsub("0.-...", "BM", archR$samplesClean)
archR$tissue <- tissue
tissue2 <- gsub("BCP00.", "tonsil", archR$tissue)
archR$tissue <- tissue2
tissue3 <- gsub("stanford_.", "tonsil", archR$tissue)
archR$tissue <- tissue3
tissue4 <- gsub("WM..", "BM", archR$tissue)
archR$tissue <- tissue4
tissue5 <- gsub("BMMC_D...", "BM", archR$tissue)
archR$tissue <- tissue5
tissue6 <- gsub("PBMC_D1...", "blood", archR$tissue)
archR$tissue <- tissue6
tissue7 <- gsub("HIP0..", "blood", archR$tissue)
archR$tissue <- tissue7
tissue8 <- gsub("EV08", "blood", archR$tissue)
archR$tissue <- tissue8
tissue9 <- gsub("pbmc_granulocyte", "blood", archR$tissue)
archR$tissue <- tissue9
tissue10 <- gsub("pbmc_noSort", "blood", archR$tissue)
archR$tissue <- tissue10
table(archR$tissue)

# Mutation status
archR$myd88 <- NA
myd88.1 <- gsub("NYU", "+", archR$source)
archR$myd88 <- myd88.1
myd88.2 <- gsub("QMUL", "-", archR$myd88)
archR$myd88 <- myd88.2
myd88.3 <- gsub("Stanford", "-", archR$myd88)
archR$myd88 <- myd88.3
myd88.1 <- gsub("Goustave-Roussy", "+", archR$source)
archR$myd88 <- myd88.1
table(archR$myd88)

# CXCR4
archR$cxcr4 <- archR$samplesClean
cxcr4.1 <- gsub("01-190", "+", archR$cxcr4)
archR$cxcr4 <- cxcr4.1
cxcr4.2 <- gsub("04-006", "+", archR$cxcr4)
archR$cxcr4 <- cxcr4.2
cxcr4.3 <- gsub("01-...", "-", archR$cxcr4)
archR$cxcr4 <- cxcr4.3
cxcr4.4 <- gsub("04-...", "-", archR$cxcr4)
archR$cxcr4 <- cxcr4.4
cxcr4.5 <- gsub("BCP00.", "-", archR$cxcr4)
archR$cxcr4 <- cxcr4.5
cxcr4.6 <- gsub("stanford_.", "-", archR$cxcr4)
archR$cxcr4 <- cxcr4.6
table(archR$cxcr4)  
```
# They have found that the best metrics for QC are the TSS enrichment score and the number of unique frags
# Note that in creating the arrow files we have already set minimum values for both the TSS (n = 4) and n frags (n = 1000)
# But let's still plot it up to see what remains
```{r}
qc.df <- getCellColData(archR, select = c("log10(nFrags)", "TSSEnrichment"))

p <- ggPoint(
  x = qc.df[,1],
  y = qc.df[,2],
  colorDensity = T,
  continuousSet = "sambaNight",
  xlabel = "Log10 Unique Fragments",
  ylabel = "TSS Enrichment",
  xlim = c(log10(500), quantile(qc.df[,1], probs = 0.99)),
  ylim = c(0, quantile(qc.df[,2], probs = 0.99))
) + geom_hline(yintercept = 4, lyt = "dashed") + geom_vline(xintercept = 3, lyt = "dashed")

p

plotPDF(p, name = "TSS-vs-Frags.pdf", ArchRProj = archR, addDOC = F) # save it out
```
# Make ridge and violin plots of TSS enrichment
```{r}
ridge <- plotGroups(
  ArchRProj = archR,
  groupBy = "samplesClean",
  colorBy = "cellColData",
  name = "TSSEnrichment",
  plotAs = "ridges"
)

violin <- plotGroups(
  ArchRProj = archR,
  groupBy = "samplesClean",
  colorBy = "cellColData",
  name = "TSSEnrichment",
  alpha = 0.4,
  plotAs = "violin"
)

ridge
violin
plotPDF(ridge, violin, name = "TSS_Ridge_Violin.pdf", ArchRProj = archR, addDOC = F, width = 5, height = 5)

```
# Make ridge plot of unique frags
```{r}
p3 <- plotGroups(
    ArchRProj = archR, 
    groupBy = "samplesClean", 
    colorBy = "cellColData", 
    name = "log10(nFrags)",
    plotAs = "ridges"
   )

p3
plotPDF(p3, name = "QC-Sample-FragSizes-Ridges.pdf", ArchRProj = archR, addDOC = F, width = 5, height = 5)
```
# Plot fragment size distribution and TSS enrichment by sample
# The TSS enrichment plot should show a clear peak in the center and a small peak to eh right caused by the well-positioned +1 nucleosome
```{r}
p1 <- plotFragmentSizes(ArchRProj = archR)

p2 <- plotTSSEnrichment(ArchRProj = archR)

p1
p2

plotPDF(p1, p2, name = "QC-Sample-FragSizes-TSSProfile_BothSets.pdf", ArchRProj = archR, addDOC = F, width = 5, height = 5)
```
# Subsetting out cells with low TSS in samples 01-190 and 04-003
```{r}
idsRemove <- as.character(archR$cellNames[(archR$samplesClean %in% c("01-190", "04-003") & archR$TSSEnrichment < 8)])
idsKeep <- as.character(archR$cellNames[!(archR$samplesClean %in% c("01-190", "04-003") & archR$TSSEnrichment < 8)])

archR.sub <- subsetArchRProject(ArchRProj = archR,
                   cells = idsKeep,
                   outputDirectory = "ArchR_ReferenceAdd_Subset",
                   dropCells = TRUE,
                   threads = getArchRThreads(),
                   force = TRUE
)

ridge2 <- plotGroups(
  ArchRProj = archR,
  groupBy = "samplesClean",
  colorBy = "cellColData",
  name = "TSSEnrichment",
  plotAs = "ridges"
)

plotPDF(archR, name = "TSS_Ridge_Violin_BelowTSS10_Removed.pdf", ArchRProj = archR, addDOC = F, width = 5, height = 5)

ridge2

frag2 <- plotGroups(
    ArchRProj = archR, 
    groupBy = "samplesClean", 
    colorBy = "cellColData", 
    name = "log10(nFrags)",
    plotAs = "ridges"
   )

frag2
plotPDF(frag2, name = "QC-Sample-FragSizes-Ridges_BelowTSS10_Removed.pdf", ArchRProj = archR, addDOC = F, width = 5, height = 5)
```
# Subsetting out hg19 files as they caused incompatability issues
```{r}
idsKeep <- as.character(archR$cellNames[!(archR$source %in% c("Greenleaf_MAPL"))])
idsKeep2 <- as.character(idsKeep[archR$samplesClean %in% c("01-190", "04-003") & archR$TSSEnrichment < 8])
idsKeep3 <- setdiff(idsKeep, idsKeep2) # get cells that exclude both MAPL and the bumps in 01-190 and 04-003

archR <- subsetArchRProject(ArchRProj = archR,
                   cells = idsKeep3,
                   outputDirectory = "ArchR_ReferenceAdd_Subset",
                   dropCells = TRUE,
                   threads = getArchRThreads(),
                   force = TRUE
)
```
# Default ratio is 1
```{r}
archR <- filterDoublets(archR, filterRatio = 1)
```
# Save project out
# This copies a new set of arrow files and a new ArchRProj to the output dir
# Unless you specify load = F, this existing project in this R studio will still be the ORIGINAL arrow files
```{r}
#saveArchRProject(ArchRProj = archR, outputDirectory = "WM_ReferenceData_Rerun", load = F)
```
# Filtering doublets
# This filters based on a ratio which allows you to apply a consistent filter across different samples with different % of doublets

############################################
### CHAPTER 4 - DIMENSIONALITY REDUCTION ###
############################################

# Dimensionality reduction with scATAC-seq data is tough due to data sparsity. A particular site is either accessible on one allele, both, or neither.
# It ends up that many loci have 0 accessible alleles. As such, you'd be reluctant to say that a cell with 3 insertions in a given site is actually 3x more accessible than another cell with only 1 insertion in the same site. 
# Many analytical strategies then binarizes their ATAC data matrix to 1s and 0s, which is mostly 0s. But this is still difficult because 0 can mean both non-accessible and not-sampled and these are different things biologically.
# If you would perform a standard dimensionality reduction like PCA, the first PC would just be all of the 0 positions
# This is gotten around via a layered dimensionality reduction approach
# First, we use a Latent Semantic Indexing (LSI) which is borrowed from natural language processing
# This calculates the peak frequency per cell, normalized by depth.
# These values are then normalized by the inverse document frequency which weighs features by how often they occur, identifying features that are more specific rather than commonly accessible
# The resultant term frequency-inverse document frequency (TF-IDF) matrix reflects how important a region/peak is to a sample.
# Then, through a technique called singular value decomposition (SVD), the most valuable information across samples is identified and represented in a lower dimensional space. Dam
# Finally, a more conventional dimensionality reduction technique like UMAP can be used to visualize the data

# ArchR offers a few different LSI implementations but by default uses the method introduced in Signac (!), which uses a term frequency that has been depth normalized to 10,000 and again normalized with the log(tf-idf)
# It is suggested to use genome-wide tiles as the inputs to LSI dimensionality reduction, but these can become prohibitively large. ArchR can do this more rapidly using 500bp tiles...but also offers an "estimated LSI" approach which only uses a subset of total cells, thereby speeding up the process and decreasing granularity of the data, which may end up reducing batch effects in the data

#### Iterative Latent Semantic Indexing (LSI) ####
# This approach computes an initial LSI transformation on the most accessible tiles and then another on the top variable features from the first iteration. You can repeat this more if necessary
# LSI is NOT deterministic...so you will not get exactly the same results. So make sure to save the ArchRProject once you have an ideal dimensionality reduction
```{r}
archR <- addIterativeLSI(
  ArchRProj = archR,
  useMatrix = "TileMatrix",
  name = "IterativeLSI",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 5000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:30,
  force = T
)

# Rerunning dim red after removing greenleaf COVID samples
archR <- addIterativeLSI(
  ArchRProj = archR,
  useMatrix = "TileMatrix",
  name = "IterativeLSI_NoGreenleafCOVID",
  iterations = 2,
  clusterParams = list( # based on Seurat's FindClusters()
    resolution = c(0.25),
    sampleCells = 5000,
    n.start = 10
  ),
  varFeatures = 20000,
  dimsToUse = 1:30,
  force = T
)

```
# You can also explicitly correct for batch effect with Harmony
# See documentation for this. Won't do now because only 1 sample lol
####################################
### BATCH CORRECTION VIA HARMONY ###
####################################
```{r}
# adding batch for full dataset integration

archR$batch <- archR$source
archR$batch[archR$samplesClean %in% c("01-190", "04-003")] <- "NYU-1"
archR$batch[archR$samplesClean %in% c("01-076", "01-115", "01-131", "01-163", "04-006")] <- "NYU-2"
archR$batch[archR$samplesClean %in% c("WM25", "WM43", "WM46", "WM47", "WM54", "WM65")] <- "Gustave-Roussy"

#archR <- addHarmony(
#    ArchRProj = archR,
#    reducedDims = "IterativeLSI",
#    name = "Harmony_Source",
#    groupBy = "source",
#    force = T
#)

archR <- addHarmony(
    ArchRProj = archR,
    reducedDims = "IterativeLSI_NoGreenleafCOVID",
    name = "Harmony_Batch_PostSubbing",
    groupBy = "batch",
    force = T
)
```
##############################
### CHAPTER 5 - CLUSTERING ###
##############################
# ArchR basically just uses Seurat's FindClusters()
```{r}
archR <- addClusters(
  input = archR,
  reducedDims = "IterativeLSI",
  method = "Seurat",
  name = "Clusters",
  resolution = 0.8,
  force = TRUE
)

archR <- addClusters(
  input = archR,
  reducedDims = "Harmony_Batch",
  method = "Seurat",
  name = "Harmony_Batch_Clusters",
  resolution = 0.8,
  force = TRUE
)
```
#####################################
### CHAPTER 6 - EMBEDDINGS (UMAP) ###
#####################################

```{r}
archR <- addUMAP(
  ArchRProj = archR,
  reducedDims = "IterativeLSI",
  name = "UMAP",
  nNeighbors = 30,
  minDist = 0.5,
  metric = "cosine",
  force = T
)

archR <- addUMAP(
  ArchRProj = archR,
  reducedDims = "IterativeLSI_NoGreenleafCOVID",
  name = "UMAP",
  nNeighbors = 30,
  minDist = 0.5,
  metric = "cosine",
  force = T
)

archR <- addUMAP(
    ArchRProj = archR, 
    reducedDims = "Harmony_Source", 
    name = "UMAP_Harmony_Source", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

archR <- addUMAP(
    ArchRProj = archR, 
    reducedDims = "Harmony_Batch", 
    name = "UMAP_Harmony_Batch", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

archR <- addUMAP(
    ArchRProj = archR, 
    reducedDims = "Harmony_Batch_PostSubbing", 
    name = "UMAP_Harmony_Batch_PostSubbing", 
    nNeighbors = 30, 
    minDist = 0.5, 
    metric = "cosine",
    force = T
)

p1 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesClean", embedding = "UMAP")
p2 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesClean", embedding = "UMAP_Harmony_Batch")
p3 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesClean", embedding = "UMAP_Harmony_Batch_PostSubbing")

plotPDF(p1, p2, p3, name = "WM_ReferenceAdd_Cleaned_Plot-UMAP-Clusters_BatchCorrectionComparison.pdf", ArchRProj = archR, addDOC = FALSE, width = 5, height = 5)

p1 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "samplesClean", embedding = "UMAP_Harmony_Batch_PostSubbing") #+ 
p2 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "condition", embedding = "UMAP_Harmony_Batch_PostSubbing")
p3 <- plotEmbedding(ArchRProj = archR, colorBy = "cellColData", name = "predictedGroup_Un", embedding = "UMAP_Harmony_Batch_PostSubbing")
plotPDF(p1, p2, p3, name = "WM_ReferenceAdd_Cleaned_Plot-UMAP-Clusters_GeneralMetadata.pdf", ArchRProj = archR, addDOC = FALSE, width = 5, height = 5)

temp <- read.table("/Users/gagled01/morganLab/Waldenstroms/TRUST4_working/outputs/bam_reports/01-076_report.tsv", sep = "\t")
head(temp)
```
Saving it out
```{r}
saveArchRProject(ArchRProj = archR, outputDirectory = "WM_ReferenceAdd_Unintegrated", load = T)
```
################################################
### CHAPTER 7 - GENE SCORES AND MARKER GENES ###
################################################
# Marker gene inputation with MAGIC to improve UMAPs
```{r}
archR <- addImputeWeights(archR)

markerGenes <- c("PAX5", "MS4A1", "MUM1", "CD27", "BCL6", "SDC1", "CD38", "CD24", "CD19")

p <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneScoreMatrix", 
    name = markerGenes, 
    embedding = "UMAP",
    imputeWeights = getImputeWeights(archR)
)

p$PAX5
p$BCL6
p$MUM1
p$MS4A1 # MS4A1 = CD20
p$CD27
p$SDC1 # SDC1 = CD138
p$CD38

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 2),p))

p2
plotPDF(plotList = p2, 
    name = "Plot-UMAP-Marker-Genes-W-Imputation_noBC.pdf", 
    ArchRProj = archR, 
    addDOC = FALSE, width = 5, height = 5)
```
######################## KEY FIGURE ALERT!!!? ###########################
### Plotting Tracks of local chromatin accessibilty for marker genes ###
########################################################################
```{r}
p <- plotBrowserTrack(
    ArchRProj = archR, 
    groupBy = "Clusters", 
    geneSymbol = markerGenes, 
    upstream = 50000,
    downstream = 50000
)

plotPDF(plotList = p, 
    name = "Plot-Tracks-Marker-Genes.pdf", 
    ArchRProj = archR, 
    addDOC = FALSE, width = 5, height = 5)
```
##########################################
### CHAPTER 8 - ALIGNING TO SCRNA DATA ###
##########################################
Subsetting archR project so that there is no mismatch in number of cells between harmony reduction and base object. Necessary for integration to work
```{r}
##############################################################
####################### SOLUTION 1!!!! #######################
##############################################################

# make sure that the length of your reduced dimension (Harmony in particular) is not different from the size of your ArchR object. if so, subset properly
rd <- getReducedDims(archR, reducedDims = 'Harmony_Batch', dimsToUse = 1:30)
length(rownames(rd))
length(archR$cellNames)
subcell = rownames(rd)[rownames(rd) %in% archR$cellNames]

archR.sub = subsetArchRProject(
      ArchRProj = archR,
      cells = rownames(rd),
      outputDirectory = "ArchRSubset_Again",
      dropCells = TRUE,
      logFile = NULL,
      threads = getArchRThreads(),
      force = TRUE
)

##############################################################
####################### SOLUTION 2!!!! #######################
##############################################################

# Downsampling ArchR object - aka solution to everything fuckin CRASHING!!!
# Ended up just removing the Greenleaf COVID samples to get it to run...that dataset was huge (250k cells) and only a handful of them were memory B cells...but still it's sad we don't have them

#greenleaf.cells <- archR$cellNames[archR$source == "Greenleaf_COVID"]
#greenleaf.subsample <- sample(greenleaf.cells, 60000, replace = F)

healthy.cells <- archR$cellNames[archR$source %in% c("QMUL", "Stanford")]
wm.cells <- archR$cellNames[archR$condition == "WM"]
cellsKeep <- c(wm.cells, healthy.cells)
length(cellsKeep)

archR.double.sub <- subsetArchRProject(ArchRProj = archR,
                   cells = cellsKeep,
                   outputDirectory = "ArchR_ReferenceAdd_Downsampled_NoGreenleafCOVID",
                   dropCells = TRUE,
                   threads = getArchRThreads(),
                   force = TRUE
)
```
Aligns ATAC gene score matrix to scRNA GEX matrix. Basically just finds the scRNA cell that looks most similar to each scATAC cell.
ArchR accepts Seurat objects
# Load in Seurat object
```{r}
seurat <- readRDS("/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_SCTransformIntegrated_CTAnnotated_ReferenceAdd_Regress_Unfiltered.rds")
```
# This is an Unconstrained Integration method
```{r}
seurat@meta.data$cellnames <- colnames(seurat)

# Remove Greenleaf COVID samples from Seurat object since we also did it for the ATAC data
healthy.cells <- seurat@meta.data$cellnames[seurat@meta.data$source %in% c("Stanford", "10X", "Grenleaf_MAPL")]
wm.cells <- seurat@meta.data$cellnames[seurat@meta.data$condition == "WM"]
cellsKeep <- c(wm.cells, healthy.cells)
length(cellsKeep)

# downsample seurat object
seurat.down <- subset(seurat, subset = cellnames %in% cellsKeep)
rm(seurat) # remove old object to save space

# Run integration. Make sure FORCE = TRUE. I also set sampleCellsATAC and RNA to 5000 which should just make it run in smaller chunks but shouldn't change anything else about the integration
archR <- addGeneIntegrationMatrix(
  ArchRProj = archR,
  useMatrix = "GeneScoreMatrix",
  matrixName = "GeneIntegrationMatrix",
  reducedDims = "Harmony_Batch",
  seRNA = seurat.down,
  addToArrow = TRUE,
  sampleCellsATAC = 5000,
  sampleCellsRNA = 5000,
  threads = 1,
  groupRNA = "majority_voting",
  nameCell = "predictedCell_Un",
  nameGroup = "predictedGroup_Un",
  nameScore = "predictedScore_Un",
  verbose = TRUE,
  force = TRUE
)

p1 <- plotEmbedding(
    archR, 
    embedding = "UMAP_Harmony_Batch",
    colorBy = "cellColData", 
    name = "predictedGroup_Un"
)

p2 <- plotEmbedding(
    archR, 
    embedding = "UMAP_Harmony_Batch",
    colorBy = "cellColData", 
    name = "samplesClean"
)

p3 <- plotEmbedding(
    archR, 
    embedding = "UMAP_Harmony_Batch",
    colorBy = "cellColData", 
    name = "condition"
)

plotPDF(p1, p2,p3, name = "WM-Integrated_CellTypistAnnotated_UnConstrainedIntegration_UMAPs_ReferenceAdd.pdf", ArchRProj = archR, addDOC = FALSE, width = 5, height = 5)
```
Save out no controls
```{r}
saveArchRProject(ArchRProj = archR, outputDirectory = "WM_ReferenceAdd_Unconstrained_Integration_NoGreenleafCOVID_ReclusteredHarmony", load = F)
```
# Subset out the non B cells which originate from the Greenleaf tonsil study (aka healthy references)
```{r}
table(archR$predictedGroup_Un)

b.lineage.cells <- archR$cellNames[archR$predictedGroup_Un %in% c("Large pre-B cells", "Memory B cells", "Plasma cells", "Small pre-B cells", 
                                                                  "Naive B cells", "Germinal center B cells", "Pro-B cells", "Proliferative germinal center B cells")]
cellsKeep <- c(b.lineage.cells)
length(cellsKeep)

archR.B.cells <- subsetArchRProject(ArchRProj = archR,
                   cells = cellsKeep,
                   outputDirectory = "ArchR_ReferenceAdd_B-LineageOnly",
                   dropCells = TRUE,
                   threads = getArchRThreads(),
                   force = TRUE
)
```

# And this is the constrained integration method, which depends on the output of the unconstrained. In our case, I don't think we need this...but we can pretend we're only interested in Plasma cells and memory B cells. and Naive, for good measure.
```{r}
table(archR$predictedGroup_Un)

cM <- as.matrix(confusionMatrix(archR$Clusters, archR$predictedGroup_Un))
preClust <- colnames(cM)[apply(cM, 1 , which.max)]
cbind(preClust, rownames(cM)) #Assignments

unique(unique(archR$predictedGroup_Un))

# selecting clusters that were assigned memory B or PC
cTNK <- paste0(c("Memory B cells", "Plasma cells"), collapse="|")

# and the rest
#cNonTNK <- paste0(c("Age-associated B cells", "Germinal center B cells", "Large pre-B cells", "Naive B cells",
#                    "Plasmablasts", "Pro-B cells", "Proliferative germinal center B cells", "Small pre-B cells"), collapse="|")
cNonTNK <- paste0(c("Age-associated B cells", "Large pre-B cells", "Non-classical monocytes", "Regulatory T cells", "Tem/Trm cytotoxic T cells",
                    "CD16+ NK cells", "Memory B cells", "pDC", "Small pre-B cells", "Classical monocytes", "Mid erythroid", "Plasma cells", "Tcm/Naive cytotoxic T cells",
                    "Germinal center B cells", "Naive B cells", "Pro-B cells", "Tcm/Naive helper T cells", "HSC/MPP", "NK cells", "Proliferative germinal center B cells",
                    "Tem/Temra cytotoxic T cells"), collapse="|")
cNonTNK

#Assign scATAC to these categories
clustTNK <- rownames(cM)[grep(cTNK, preClust)]
clustTNK

clustNonTNK <- rownames(cM)[grep(cNonTNK, preClust)]
clustNonTNK

# get cells in these categories
rnaTNK <- colnames(seurat)[grep(cTNK, seurat@meta.data$majority_voting)]

rnaNonTNK <- colnames(seurat)[grep(cNonTNK, seurat@meta.data$majority_voting)]

groupList <- SimpleList(
    TNK = SimpleList(
        ATAC = archR$cellNames[archR$Clusters %in% clustTNK],
        RNA = rnaTNK
    ),
    NonTNK = SimpleList(
        ATAC = archR$cellNames[archR$Clusters %in% clustNonTNK],
        RNA = rnaNonTNK
    )    
)

#~5 minutes. lol PSYCH!!!! like 5 hours
archR <- addGeneIntegrationMatrix(
    ArchRProj = archR, 
    useMatrix = "GeneScoreMatrix",
    matrixName = "GeneIntegrationMatrix",
    reducedDims = "Harmony_Batch",
    seRNA = seurat,
    addToArrow = TRUE, 
    sampleCellsATAC = 8000,
    sampleCellsRNA = 8000,
    groupList = groupList,
    groupRNA = "majority_voting",
    nameCell = "predictedCell_Co",
    nameGroup = "predictedGroup_Co",
    nameScore = "predictedScore_Co",
    force = TRUE,
    k.score = 15,
    k.filter = 150 # lowering from 30 to 15 to try to get it to run
)


p1 <- plotEmbedding(
    archR, 
    colorBy = "cellColData", 
    name = "predictedGroup_Co",
    embedding = "UMAP_Harmony_Batch"
)
p1

p2 <- plotEmbedding(
    archR, 
    colorBy = "cellColData", 
    name = "samplesClean",
    embedding = "UMAP_Harmony_Batch"
)

plotPDF(p1,p2, name = "WM_BothSetsNoControls_rPCA-Integrated_CellTypistAnnotated_ConstrainedIntegrationUMAPs.pdf", ArchRProj = archR, addDOC = FALSE, width = 5, height = 5)
```

# Plotting UMAPs overlayed with the  gene expression values from our integrated matrix
```{r}
#archR <- addImputeWeights(archR)

markerGenes <- c("PAX5", "MS4A1", "CD27", "BCL6", "SDC1", "CD38", "CD24", "CD19")
#markerGenes <- c("CD19", "MS4A1", "MME", "CD5", "CD45", "CD23", "CD38", "CD25", "CD11c", "CD103")
markerGenes <- c("CD19", "MS4A1", "MME", "CD5", "PTPRC", "FCER2", "CD38", "IL2RA", "ITGAX", "ITGAE")
# Treon 2014 paper on SPIB and plasma cell differentiation in WM
#markerGenes <- c("SPIB", "POU2F1", "POU2F2", "XBP1", "IRF4", "PRDM1", "ID2", "BCL6", "PAX5", "SP1", "SP3", "EGR1")

p1 <- plotEmbedding(
    ArchRProj = archR, 
    colorBy = "GeneIntegrationMatrix", 
    name = markerGenes, 
    continuousSet = "horizonExtra",
    embedding = "UMAP_Harmony_Batch",
    imputeWeights = getImputeWeights(archR)
)

#Same plots but overlain with gene scores
p2 <- plotEmbedding(
    ArchRProj = archR,
    colorBy = "GeneScoreMatrix",
    continuousSet = "horizonExtra",
    name = markerGenes,
    embedding = "UMAP_Harmony_Batch",
    imputeWeights = getImputeWeights(archR)
)
p2

plotPDF(plotList = p1, p2, 
    name = "Plot-UMAP-Marker-Genes-RNA-W-Imputation_Integrated_GEX_GeneScores.pdf", 
    ArchRProj = archR, 
    addDOC = FALSE, width = 5, height = 5)


```

