---
title: "GSEA_RankedList_Generation"
output: html_document
date: "2023-02-02"
---
################################################################################
Created by Dylan Gagler, MSc - dylan.gagler@nyulangone.org or dylangagler@gmail.com

Code used for generating ranked lists for GSEA preranked analysis.

Ranking is done by first performing differential gene expression analysis using wilcoxon rank-sum test via Seurat's FindMarkers() and then multiplying the average logFC by -log10 of the adjusted p values. 

Sometimes this results in infinitely small p-values, in which case no rank-values can be computed. As such, a quick fix can be implemented by manually enforcing ranking values in the order that the genes appear in the output df.

################################################################################

Load libraries
```{r}
library(Seurat)
library(dplyr)
```
Load data
```{r}
library(ArchR)
library(Seurat)
library(SummarizedExperiment)
library(ggpp)
library(ggrepel)
addArchRThreads(threads = 1)
addArchRGenome("hg38")
options(future.globals.maxSize=1000000000000000000000000000) 
archR <- loadArchRProject(path = "/Users/gagled01/morganLab/Waldenstroms/Waldenstroms_Git/WM_DeviationsAdded_BatchCorrected")

# Fix annotations
archR$final_annotation[archR$final_annotation == "Non-clonal MBC"] <- "Polyclonal MBC"
archR$final_annotation[archR$final_annotation == "Non-clonal PC"] <- "Polyclonal PC"
archR$final_annotation[archR$final_annotation == "Intermediate MBC"] <- "WM43"
table(archR$final_annotation)

archR$hasTopCDR3[archR$final_annotation == "MBC-like MBC" & archR$samplesClean == "WM54"] <- "Yes"

# Make new col for patient UMAP annotations
archR$samplesRefCombined <- archR$samplesClean
archR$samplesRefCombined[archR$condition == "HD"] <- "Reference"
```
Do the differential gene expression
```{r}
DE.markers <- getMarkerFeatures(
  ArchRProj = archR, 
  useMatrix = "GeneIntegrationMatrix",
  groupBy = "final_annotation",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = c("MBC-like MBC"),
  bgdGroups = c("PC-like MBC")
)

DE.markers_list <- getMarkers(DE.markers) # DO NOT FILTER GENES!!!! FOR GSERA!!!!! FEKEKEEEE

# Clonal MBC
df.gex <- DE.markers_list$`MBC-like MBC`
df.gex <- as.data.frame(df.gex)
df.gex <- df.gex[order(-df.gex$MeanDiff),]
```
# Generate ranked list
```{r}
# Generate rank score by mulitplying avg log FC by negative log10 of p value.
rank.score <- sign(df.gex$Log2FC) * -log10(df.gex$FDR)
#rank.score <- normaldf.gex$MeanDiff

# Generate gene list
genes <- df.gex$name

# Put into df
rank.df <- data.frame(genes, rank.score)
rownames(rank.df) <- rank.df$genes
rank.df <- rank.df[order(-rank.df$rank.score),]
rank.df
```
Write out ranked list
```{r}
write.table(rank.df, "/Users/gagled01/morganLab/Waldenstroms/GSEA/ArchR_MBC-likeMBC_vs_PC-liekMBC_forGSEA_UpMBC-like.rnk",
            sep = "\t",
            quote = F,
            row.names = F,
            col.names = F)
```

