---
title: "WM_scRNA"
output: html_document
date: "2023-07-24"
---
Code used for the analysis of single-cell RNA data using a Seurat v4 pipeline. Includes preprocessing and QC for both WM and reference samples and subsequent integration using a Seurat SCTransform framework.

# Load libraries
```{r}
library(Seurat)
library(ggplot2)
library(dplyr)
library(patchwork)
library(SoupX)
library(celda)
library(DropletUtils)
library(glmGamPoi)
options(future.globals.maxSize=10000000000000000)
```
# Load cellranger outputs, filter, etc. without SoupX
```{r}
p01_076.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-076_outs/filtered_feature_bc_matrix")
p01_115.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-115_outs/filtered_feature_bc_matrix")
p01_131.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-131_outs/filtered_feature_bc_matrix")
p01_163.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-163_outs/filtered_feature_bc_matrix")
p01_190.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-190_outs/filtered_feature_bc_matrix")
p04_003.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/04-003_outs/filtered_feature_bc_matrix")
p04_006.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/04-006_outs/filtered_feature_bc_matrix")
wm25.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM25_outs/filtered_feature_bc_matrix")
wm43.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM43_outs/filtered_feature_bc_matrix")
wm46.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM46_outs/filtered_feature_bc_matrix")
wm47.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM47_outs/filtered_feature_bc_matrix")
wm54.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM54_outs/filtered_feature_bc_matrix")
wm65.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM65_outs/filtered_feature_bc_matrix")

# Create seurat objects, standard filter thresholds
p01_076_srat <- CreateSeuratObject(p01_076.data$`Gene Expression`, min.cells = 3, min.features = 200)
p01_115_srat <- CreateSeuratObject(p01_115.data$`Gene Expression`, min.cells = 3, min.features = 200)
p01_131_srat <- CreateSeuratObject(p01_131.data$`Gene Expression`, min.cells = 3, min.features = 200)
p01_163_srat <- CreateSeuratObject(p01_163.data$`Gene Expression`, min.cells = 3, min.features = 200)
p01_190_srat <- CreateSeuratObject(p01_190.data$`Gene Expression`, min.cells = 3, min.features = 200)
p04_003_srat <- CreateSeuratObject(p04_003.data$`Gene Expression`, min.cells = 3, min.features = 200)
p04_006_srat <- CreateSeuratObject(p04_003.data$`Gene Expression`, min.cells = 3, min.features = 200)
p04_006_srat <- CreateSeuratObject(p04_003.data$`Gene Expression`, min.cells = 3, min.features = 200)
wm25_srat <- CreateSeuratObject(wm25.data$`Gene Expression`, min.cells = 3, min.features = 200)
wm43_srat <- CreateSeuratObject(wm43.data$`Gene Expression`, min.cells = 3, min.features = 200)
wm46_srat <- CreateSeuratObject(wm46.data$`Gene Expression`, min.cells = 3, min.features = 200)
wm47_srat <- CreateSeuratObject(wm47.data$`Gene Expression`, min.cells = 3, min.features = 200)
wm54_srat <- CreateSeuratObject(wm54.data$`Gene Expression`, min.cells = 3, min.features = 200)
wm65_srat <- CreateSeuratObject(wm65.data$`Gene Expression`, min.cells = 3, min.features = 200)

# Fix cellnames 
p01_076_cells.fixed <- paste("01-076", colnames(p01_076_srat), sep = "-")
p01_115_cells.fixed <- paste("01-115", colnames(p01_115_srat), sep = "-")
p01_131_cells.fixed <- paste("01-131", colnames(p01_131_srat), sep = "-")
p01_163_cells.fixed <- paste("01-163", colnames(p01_163_srat), sep = "-")
p01_190_cells.fixed <- paste("01-190", colnames(p01_190_srat), sep = "-")
p04_003_cells.fixed <- paste("04-003", colnames(p04_003_srat), sep = "-")
p04_006_cells.fixed <- paste("04-006", colnames(p04_006_srat), sep = "-")
wm25_cells.fixed <- paste("WM25", colnames(wm25_srat), sep = "-")
wm43_cells.fixed <- paste("WM43", colnames(wm43_srat), sep = "-")
wm46_cells.fixed <- paste("WM46", colnames(wm46_srat), sep = "-")
wm47_cells.fixed <- paste("WM47", colnames(wm47_srat), sep = "-")
wm54_cells.fixed <- paste("WM54", colnames(wm54_srat), sep = "-")
wm65_cells.fixed <- paste("WM65", colnames(wm65_srat), sep = "-")

# Rename cells with fixed names
p01_076_srat <- RenameCells(p01_076_srat, new.names = p01_076_cells.fixed)
p01_115_srat <- RenameCells(p01_115_srat, new.names = p01_115_cells.fixed)
p01_131_srat <- RenameCells(p01_131_srat, new.names = p01_131_cells.fixed)
p01_163_srat <- RenameCells(p01_163_srat, new.names = p01_163_cells.fixed)
p01_190_srat <- RenameCells(p01_190_srat, new.names = p01_190_cells.fixed)
p04_003_srat <- RenameCells(p04_003_srat, new.names = p04_003_cells.fixed)
p04_006_srat <- RenameCells(p04_006_srat, new.names = p04_006_cells.fixed)
wm25_srat <- RenameCells(wm25_srat, new.names = wm25_cells.fixed)
wm43_srat <- RenameCells(wm43_srat, new.names = wm43_cells.fixed)
wm46_srat <- RenameCells(wm46_srat, new.names = wm46_cells.fixed)
wm47_srat <- RenameCells(wm47_srat, new.names = wm47_cells.fixed)
wm54_srat <- RenameCells(wm54_srat, new.names = wm54_cells.fixed)
wm65_srat <- RenameCells(wm65_srat, new.names = wm65_cells.fixed)
```
Load data with SoupX
SoupX is a package to estimate ambient mRNAs (the "soup") in single-cell RNA droplets by:
1. calculating the profile of the soup
2. estimating the fraction of UMIs are derived from the soup in each droplet
3. produce a corrected count table

The rho value = the contamination %...so rho of 0 = no contam and rho of 1 = 100% contamination
For the output plots, the solid black curve shows something like the frequency of different estimates of rho and the red line indicates the peak, which reflects the rho estimate. 
```{r}
p01_076.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-076_outs/filtered_feature_bc_matrix")
p01_115.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-115_outs/filtered_feature_bc_matrix")
p01_131.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-131_outs/filtered_feature_bc_matrix")
p01_163.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-163_outs/filtered_feature_bc_matrix")
p01_190.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/01-190_outs/filtered_feature_bc_matrix")
p04_003.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/04-003_outs/filtered_feature_bc_matrix")
p04_006.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/04-006_outs/filtered_feature_bc_matrix")
wm25.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM25_outs/filtered_feature_bc_matrix")
wm43.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM43_outs/filtered_feature_bc_matrix")
wm46.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM46_outs/filtered_feature_bc_matrix")
wm47.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM47_outs/filtered_feature_bc_matrix")
wm54.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM54_outs/filtered_feature_bc_matrix")
wm65.data <- Read10X("/Users/gagled01/morganLab/Waldenstroms/singlecell/data/RNA/WM65_outs/filtered_feature_bc_matrix")

# Estimate the contamination fraction
p01_190.data <- autoEstCont(p01_190.data)
p04_003.data <- autoEstCont(p04_003.data)
p01_076.data <- autoEstCont(p01_076.data, tfidfMin = 0.75)
p01_115.data <- autoEstCont(p01_115.data)
p01_131.data <- autoEstCont(p01_131.data)
p01_163.data <- autoEstCont(p01_163.data, tfidfMin = 0.75)
p04_006.data <- autoEstCont(p04_006.data, tfidfMin = 0.7)
wm25.data <- autoEstCont(wm25.data)
wm43.data <- autoEstCont(wm43.data)
wm46.data <- autoEstCont(wm46.data, tfidfMin = 0.8, soupQuantile = 0.95, forceAccept = TRUE)
wm47.data <- autoEstCont(wm47.data, tfidfMin = 0.7)
wm54.data <- autoEstCont(wm54.data)
wm65.data <- autoEstCont(wm65.data, tfidfMin = 0.7)

# Remove ambient RNA
p01_190_rna <- adjustCounts(p01_190.data) #
p04_003_rna <- adjustCounts(p04_003.data)
p01_076_rna <- adjustCounts(p01_076.data)
p01_115_rna <- adjustCounts(p01_115.data)
p01_131_rna <- adjustCounts(p01_131.data)
p01_163_rna <- adjustCounts(p01_163.data)
p04_006_rna <- adjustCounts(p04_006.data)
wm25_rna <- adjustCounts(wm25.data)
wm43_rna <- adjustCounts(wm43.data)
wm46_rna <- adjustCounts(wm46.data)
wm47_rna <- adjustCounts(wm47.data)
wm54_rna <- adjustCounts(wm54.data)
wm65_rna <- adjustCounts(wm65.data)

# Convert to Seurat
p01_190_srat <- CreateSeuratObject(p01_190_rna, min.cells = 3, min.features = 200)
p04_003_srat <- CreateSeuratObject(p04_003_rna, min.cells = 3, min.features = 200)
p01_076_srat <- CreateSeuratObject(p01_076_rna, min.cells = 3, min.features = 200)
p01_115_srat <- CreateSeuratObject(p01_115_rna, min.cells = 3, min.features = 200)
p01_131_srat <- CreateSeuratObject(p01_131_rna, min.cells = 3, min.features = 200)
p01_163_srat <- CreateSeuratObject(p01_163_rna, min.cells = 3, min.features = 200)
p04_006_srat <- CreateSeuratObject(p04_006_rna, min.cells = 3, min.features = 200)
wm25_srat <- CreateSeuratObject(wm25_rna, min.cells = 3, min.features = 200)
wm43_srat <- CreateSeuratObject(wm43_rna, min.cells = 3, min.features = 200)
wm46_srat <- CreateSeuratObject(wm46_rna, min.cells = 3, min.features = 200)
wm47_srat <- CreateSeuratObject(wm47_rna, min.cells = 3, min.features = 200)
wm54_srat <- CreateSeuratObject(wm54_rna, min.cells = 3, min.features = 200)
wm65_srat <- CreateSeuratObject(wm65_rna, min.cells = 3, min.features = 200)

rm(p01_190.data, p01_076.data, p04_003.data, p01_115.data, p01_131.data, p01_163.data, p04_006.data, wm25.data, wm43.data, wm46.data, wm47.data, wm54.data, wm65.data,
   p01_190_rna, p01_076_rna, p04_003_rna, p01_115_rna, p01_131_rna, p01_163_rna, p04_006_rna, wm25_rna, wm43_rna, wm46_rna, wm47_rna, wm54_rna, wm65_rna)
```
# Basic metadata annotation
```{r}
p01_190_srat[["patient"]] <- "01-190"
p04_003_srat[["patient"]] <- "04-003"
p01_076_srat[["patient"]] <- "01-076"
p01_115_srat[["patient"]] <- "01-115"
p01_131_srat[["patient"]] <- "01-131"
p01_163_srat[["patient"]] <- "01-163"
p04_006_srat[["patient"]] <- "04-006"
wm25_srat[["patient"]] <- "WM25"
wm43_srat[["patient"]] <- "WM43"
wm46_srat[["patient"]] <- "WM46"
wm47_srat[["patient"]] <- "WM47"
wm54_srat[["patient"]] <- "WM54"
wm65_srat[["patient"]] <- "WM65"

p01_190_srat[["batch"]] <- "1"
p04_003_srat[["batch"]] <- "1"
p01_076_srat[["batch"]] <- "2"
p01_115_srat[["batch"]] <- "2"
p01_131_srat[["batch"]] <- "2"
p01_163_srat[["batch"]] <- "2"
p04_006_srat[["batch"]] <- "2"
wm25_srat[["batch"]] <- "3"
wm43_srat[["batch"]] <- "3"
wm46_srat[["batch"]] <- "3"
wm47_srat[["batch"]] <- "3"
wm54_srat[["batch"]] <- "3"
wm65_srat[["batch"]] <- "3"

p01_190_srat[["source"]] <- "NYU"
p04_003_srat[["source"]] <- "NYU"
p01_076_srat[["source"]] <- "NYU"
p01_115_srat[["source"]] <- "NYU"
p01_131_srat[["source"]] <- "NYU"
p01_163_srat[["source"]] <- "NYU"
p04_006_srat[["source"]] <- "NYU"
wm25_srat[["source"]] <- "NYU"
wm43_srat[["source"]] <- "NYU"
wm46_srat[["source"]] <- "NYU"
wm47_srat[["source"]] <- "NYU"
wm54_srat[["source"]] <- "NYU"
wm65_srat[["source"]] <- "NYU"

p01_190_srat[["condition"]] <- "WM"
p04_003_srat[["condition"]] <- "WM"
p01_076_srat[["condition"]] <- "WM"
p01_115_srat[["condition"]] <- "WM"
p01_131_srat[["condition"]] <- "WM"
p01_163_srat[["condition"]] <- "WM"
p04_006_srat[["condition"]] <- "WM"
wm25_srat[["condition"]] <- "WM"
wm43_srat[["condition"]] <- "WM"
wm46_srat[["condition"]] <- "WM"
wm47_srat[["condition"]] <- "WM"
wm54_srat[["condition"]] <- "WM"
wm65_srat[["condition"]] <- "WM"
```
# Getting % mitochondrial transcripts
```{r}
p01_190_srat[["percent.mt"]] <- PercentageFeatureSet(p01_190_srat, pattern = "^MT-")
p04_003_srat[["percent.mt"]] <- PercentageFeatureSet(p04_003_srat, pattern = "^MT-")
p01_076_srat[["percent.mt"]] <- PercentageFeatureSet(p01_076_srat, pattern = "^MT-")
p01_115_srat[["percent.mt"]] <- PercentageFeatureSet(p01_115_srat, pattern = "^MT-")
p01_131_srat[["percent.mt"]] <- PercentageFeatureSet(p01_131_srat, pattern = "^MT-")
p01_163_srat[["percent.mt"]] <- PercentageFeatureSet(p01_163_srat, pattern = "^MT-")
p04_006_srat[["percent.mt"]] <- PercentageFeatureSet(p04_006_srat, pattern = "^MT-")
wm25_srat[["percent.mt"]] <- PercentageFeatureSet(wm25_srat, pattern = "^MT-")
wm43_srat[["percent.mt"]] <- PercentageFeatureSet(wm43_srat, pattern = "^MT-")
wm46_srat[["percent.mt"]] <- PercentageFeatureSet(wm46_srat, pattern = "^MT-")
wm47_srat[["percent.mt"]] <- PercentageFeatureSet(wm47_srat, pattern = "^MT-")
wm54_srat[["percent.mt"]] <- PercentageFeatureSet(wm54_srat, pattern = "^MT-")
wm65_srat[["percent.mt"]] <- PercentageFeatureSet(wm65_srat, pattern = "^MT-")
```
# Plot function
```{r}
qc_plots <- function(seurat.object, min_umi, min_gene, perc_mito) {
  patient <- seurat.object@meta.data$patient[1] # Get patient label for plot
  p1 <- VlnPlot(seurat.object, features = "nCount_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = min_umi, # UMI plot
                                                                                              color = "black", 
                                                                                              linewidth = 1,
                                                                                              linetype = "dashed") + NoLegend()
  p2 <- VlnPlot(seurat.object, features = "nFeature_RNA", log = TRUE, pt.size = 0) + geom_hline(yintercept = min_gene, # Gene count
                                                                                              color = "black",
                                                                                              linewidth = 1,
                                                                                              linetype = "dashed") + NoLegend()
  p3 <- VlnPlot(seurat.object, features = "percent.mt", log = TRUE, pt.size = 0) + geom_hline(yintercept = perc_mito, # % mito 
                                                                                              color = "black",
                                                                                              linewidth = 1,
                                                                                              linetype = "dashed") + NoLegend()
  qc.patch <- p1 | p2 | p3
  ggsave(paste0("/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/", patient, "_BasicQC_Metrics.png"), height = 4, width = 8)
}
```
# QC filtering function
```{r}
qc_filter <- function(seurat.object, umi_upper, umi_lower, gene_upper, gene_lower, mito_filter) {
  patient <- seurat.object@meta.data$patient[1] # Get patient label for plot
  cells.before <- ncol(seurat.object)
  
  seurat.filtered <- subset(x = seurat.object, 
                            subset = nCount_RNA > umi_lower &
                              nCount_RNA < umi_upper &
                              nFeature_RNA > gene_lower &
                              nFeature_RNA < gene_upper &
                              percent.mt < mito_filter
                            )
  cells.after <- ncol(seurat.filtered)
  
  print(paste0(patient, " cell count before = ", cells.before, " and after = ", cells.after))
  print(paste0(patient, " cell loss = ", round(1 - cells.after/cells.before, 3)*100, "%"))
  
  return(seurat.filtered)
}
```
# Generate plots
```{r}
qc_plots(p01_190_srat, 300, 200, 20)
qc_plots(p04_003_srat, 300, 200, 20)
qc_plots(p01_076_srat, 300, 200, 20)
qc_plots(p01_115_srat, 300, 200, 20)
qc_plots(p01_131_srat, 300, 200, 20)
qc_plots(p01_163_srat, 300, 200, 20)
qc_plots(p04_006_srat, 300, 200, 20)
qc_plots(wm25_srat, 300, 200, 20)
qc_plots(wm43_srat, 300, 200, 20)
qc_plots(wm46_srat, 300, 200, 20)
qc_plots(wm47_srat, 300, 200, 20)
qc_plots(wm54_srat, 300, 200, 20)
qc_plots(wm65_srat, 300, 200, 20)
```
# Filtering
```{r}
p01_190_filtered <- qc_filter(p01_190_srat, 10000, 300, 3000, 200, 20)
p04_003_filtered <- qc_filter(p04_003_srat, 10000, 300, 3000, 200, 20)
p01_076_filtered <- qc_filter(p01_076_srat, 10000, 300, 3000, 200, 20)
p01_115_filtered <- qc_filter(p01_115_srat, 10000, 300, 3000, 200, 20)
p01_131_filtered <- qc_filter(p01_131_srat, 10000, 300, 3000, 200, 20)
p01_163_filtered <- qc_filter(p01_163_srat, 10000, 300, 3000, 200, 20)
p04_006_filtered <- qc_filter(p04_006_srat, 10000, 300, 3000, 200, 20)
wm25_filtered <- qc_filter(wm25_srat, 10000, 300, 3000, 200, 20)
wm43_filtered <- qc_filter(wm43_srat, 10000, 300, 3000, 200, 20)
wm46_filtered <- qc_filter(wm46_srat, 10000, 300, 3000, 200, 20)
wm47_filtered <- qc_filter(wm47_srat, 10000, 300, 3000, 200, 30)
wm54_filtered <- qc_filter(wm54_srat, 10000, 300, 3000, 200, 20)
wm65_filtered <- qc_filter(wm65_srat, 10000, 300, 3000, 200, 30)

#rm(p01_190_srat, p04_003_srat, p01_076_srat, p01_115_srat, p01_131_srat, p01_163_srat, p04_006_srat)
# prefilter median cell count
median( length(colnames(p01_076_srat)), length(colnames(p01_115_srat)), length(colnames(p01_131_srat)), length(colnames(p01_163_srat)), length(colnames(p01_190_srat)), 
        length(colnames(p04_003_srat)), length(colnames(p04_006_srat)), length(colnames(wm25_srat)), length(colnames(wm43_srat)), length(colnames(wm46_srat)), length(colnames(wm47_srat)),
        length(colnames(wm54_srat)), length(colnames(wm65_srat))
)

# post filter median cell count
median( length(colnames(p01_076_filtered)), length(colnames(p01_115_filtered)), length(colnames(p01_131_filtered)), length(colnames(p01_163_filtered)), length(colnames(p01_190_filtered)), 
        length(colnames(p04_003_filtered)), length(colnames(p04_006_filtered)), length(colnames(wm25_filtered)), length(colnames(wm43_filtered)), length(colnames(wm46_filtered)), length(colnames(wm47_filtered)),
        length(colnames(wm54_filtered)), length(colnames(wm65_filtered))
)
```
Merge data and save
```{r}
wm.rna <- merge(p01_190_srat, 
                c(p04_003_srat, 
                p01_076_srat, 
                p01_115_srat, 
                p01_131_srat, 
                p01_163_srat, 
                p04_006_srat,
                wm25_srat,
                wm43_srat,
                wm46_srat,
                wm47_srat,
                wm54_srat,
                wm65_srat)) 

pre.merged <- c(colnames(p01_190_srat),  
  colnames(p04_003_srat), 
  colnames(p04_006_srat), 
  colnames(p01_076_srat), 
  colnames(p01_115_srat), 
  colnames(p01_131_srat), 
  colnames(p01_163_srat), 
  colnames(wm25_srat), 
  colnames(wm43_srat), 
  colnames(wm46_srat), 
  colnames(wm47_srat), 
  colnames(wm54_srat), 
  colnames(wm65_srat))

intersect(pre.merged, gsub("_1", "", colnames(wm.rna)))
length(intersect(pre.merged, gsub("_1", "", colnames(wm.rna))))
wm.rna

wm.rna.filtered <- merge(p01_190_filtered, 
                c(p04_003_filtered, 
                p01_076_filtered, 
                p01_115_filtered, 
                p01_131_filtered, 
                p01_163_filtered, 
                p04_006_filtered,
                wm25_filtered,
                wm43_filtered,
                wm46_filtered,
                wm47_filtered,
                wm54_filtered,
                wm65_filtered)) 
```
Save em out
```{r}
saveRDS(wm.rna, "/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_WM_Merged_Unfiltered.rds")
saveRDS(wm.rna.filtered, "/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_WM_Merged_QCFiltered.rds")
```
# Getting cell counts before and after
```{r}
library(Seurat)
pre.filt <- readRDS("/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_WM_Merged_Unfiltered.rds")
post.filt <- readRDS("/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_WM_Merged_QCfiltered.rds")
```
########################################################################################################################################################################################################
### NOW SAME PROTOCOL WITH THE GREENLEAF LAB REFERENCE DATA ###

Not doing ambient RNA removal because:
1. they don't supply cellranger outputs (so no clustering to use for SoupX)
2. they don't provide raw + filtered feature matrices, just raw matrices
########################################################################################################################################################################################################
# Load data
```{r}
tonsil.1a.data <- Read10X('/Users/gagled01/morganLab/Waldenstroms/singlecell/data/reference_RNA/Tonsil1a_count/outs/raw_feature_bc_matrix')
tonsil.1b.data <- Read10X('/Users/gagled01/morganLab/Waldenstroms/singlecell/data/reference_RNA/Tonsil1b_count/outs/raw_feature_bc_matrix')
tonsil.2a.data <- Read10X('/Users/gagled01/morganLab/Waldenstroms/singlecell/data/reference_RNA/Tonsil2a_count/outs/raw_feature_bc_matrix')
tonsil.2b.data <- Read10X('/Users/gagled01/morganLab/Waldenstroms/singlecell/data/reference_RNA/Tonsil2b_count/outs/raw_feature_bc_matrix')
tonsil.3a.data <- Read10X('/Users/gagled01/morganLab/Waldenstroms/singlecell/data/reference_RNA/Tonsil3a_count/outs/raw_feature_bc_matrix')
tonsil.3b.data <- Read10X('/Users/gagled01/morganLab/Waldenstroms/singlecell/data/reference_RNA/Tonsil3b_count/outs/raw_feature_bc_matrix')
```
# Create Seurat objects
```{r}
tonsil.1a <- CreateSeuratObject(counts = tonsil.1a.data$`Gene Expression`, min.cells=3, min.features=200)
tonsil.1b <- CreateSeuratObject(counts = tonsil.1b.data$`Gene Expression`, min.cells=3, min.features=200)
tonsil.2a <- CreateSeuratObject(counts = tonsil.2a.data$`Gene Expression`, min.cells=3, min.features=200)
tonsil.2b <- CreateSeuratObject(counts = tonsil.2b.data$`Gene Expression`, min.cells=3, min.features=200)
tonsil.3a <- CreateSeuratObject(counts = tonsil.3a.data$`Gene Expression`, min.cells=3, min.features=200)
tonsil.3b <- CreateSeuratObject(counts = tonsil.3b.data$`Gene Expression`, min.cells=3, min.features=200)
```
# Basic metadata annotation
```{r}
tonsil.1a[["patient"]] <- "tonsil1"
tonsil.1b[["patient"]] <- "tonsil1"
tonsil.2a[["patient"]] <- "tonsil2"
tonsil.2b[["patient"]] <- "tonsil2"
tonsil.3a[["patient"]] <- "tonsil3"
tonsil.3b[["patient"]] <- "tonsil3"

tonsil.1a[["batch"]] <- "tonsil1"
tonsil.1b[["batch"]] <- "tonsil1"
tonsil.2a[["batch"]] <- "tonsil2"
tonsil.2b[["batch"]] <- "tonsil2"
tonsil.3a[["batch"]] <- "tonsil3"
tonsil.3b[["batch"]] <- "tonsil3"

tonsil.1a[["source"]] <- "Stanford"
tonsil.1b[["source"]] <- "Stanford"
tonsil.2a[["source"]] <- "Stanford"
tonsil.2b[["source"]] <- "Stanford"
tonsil.3a[["source"]] <- "Stanford"
tonsil.3b[["source"]] <- "Stanford"

tonsil.1a[["condition"]] <- "Healthy"
tonsil.1b[["condition"]] <- "Healthy"
tonsil.2a[["condition"]] <- "Healthy"
tonsil.2b[["condition"]] <- "Healthy"
tonsil.3a[["condition"]] <- "Healthy"
tonsil.3b[["condition"]] <- "Healthy"
```
# Getting % mitochondrial transcripts
```{r}
tonsil.1a[["percent.mt"]] <- PercentageFeatureSet(tonsil.1a, pattern = "^MT-")
tonsil.1b[["percent.mt"]] <- PercentageFeatureSet(tonsil.1b, pattern = "^MT-")
tonsil.2a[["percent.mt"]] <- PercentageFeatureSet(tonsil.2a, pattern = "^MT-")
tonsil.2b[["percent.mt"]] <- PercentageFeatureSet(tonsil.2b, pattern = "^MT-")
tonsil.3a[["percent.mt"]] <- PercentageFeatureSet(tonsil.3a, pattern = "^MT-")
tonsil.3b[["percent.mt"]] <- PercentageFeatureSet(tonsil.3b, pattern = "^MT-")
```
# QC plots
```{r}
qc_plots(tonsil.1a, 300, 200, 20)
qc_plots(tonsil.1b, 300, 200, 20)
qc_plots(tonsil.2a, 300, 200, 20)
qc_plots(tonsil.2b, 300, 200, 20)
qc_plots(tonsil.3a, 300, 200, 20)
qc_plots(tonsil.3b, 300, 200, 20)
```
# They filtered this data by:
- number of genes between 200 and 7500
- % mito between 0 - 20
- ADTs less than 4000
# QC filtering function for Stanford (ADTs included)
```{r}
qc_filter_adts <- function(seurat.object, gene_upper, gene_lower, mito_filter) {
  patient <- seurat.object@meta.data$patient[1] # Get patient label for plot
  cells.before <- ncol(seurat.object)
  
  seurat.filtered <- subset(x = seurat.object, 
                            subset = nFeature_RNA > gene_lower &
                              nFeature_RNA < gene_upper &
                              percent.mt < mito_filter #&
                              #nFeature_ADT < 4000
                            )
  cells.after <- ncol(seurat.filtered)
  
  print(paste0(patient, " cell count before = ", cells.before, " and after = ", cells.after))
  print(paste0(patient, " cell loss = ", round(1 - cells.after/cells.before, 3)*100, "%"))
  
  return(seurat.filtered)
}
```
# Filter Stanford
```{r}
tonsil.1a.filtered <- qc_filter_adts(tonsil.1a, 7500, 200, 20)
tonsil.1b.filtered <- qc_filter_adts(tonsil.1b, 7500, 200, 20)
tonsil.2a.filtered <- qc_filter_adts(tonsil.2a, 7500, 200, 20)
tonsil.2b.filtered <- qc_filter_adts(tonsil.2b, 7500, 200, 20)
tonsil.3a.filtered <- qc_filter_adts(tonsil.3a, 7500, 200, 20)
tonsil.3b.filtered <- qc_filter_adts(tonsil.3b, 7500, 200, 20)

#rm(tonsil.1a, tonsil.1b, tonsil.2a, tonsil.2b, tonsil.3a, tonsil.3b)
prefilt <- c(ncol(tonsil.1a), ncol(tonsil.1b), ncol(tonsil.2a), ncol(tonsil.2b), ncol(tonsil.3a), ncol(tonsil.3b))
prefilt_total <- sum(prefilt)
prefilt_median <- median(prefilt)

postfilt <- c(ncol(tonsil.1a.filtered), ncol(tonsil.1b.filtered), ncol(tonsil.2a.filtered), ncol(tonsil.2b.filtered), ncol(tonsil.3a.filtered), ncol(tonsil.3b.filtered))
postfilt_total <- sum(postfilt)
postfilt_median <- median(postfilt)
```
# Merging data into disease and control for SCTransform integration. Save out
```{r}
wm.rna <- merge(p01_190_filtered, 
                c(p04_003_filtered, 
                p01_076_filtered, 
                p01_115_filtered, 
                p01_131_filtered, 
                p01_163_filtered, 
                p04_006_filtered)) 

ctrl.rna <- merge(tonsil.1a.filtered,
                  c(tonsil.1b.filtered,
                  tonsil.2a.filtered,
                  tonsil.2b.filtered,
                  tonsil.3a.filtered,
                  tonsil.3b.filtered))

saveRDS(wm.rna, "/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_WM_MergedQCFiltered.rds")
saveRDS(ctrl.rna, "/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_Ctrl_MergedQCFiltered.rds")

rm(p01_190_filtered, p04_003_filtered, p01_076_filtered, p01_115_filtered, p01_131_filtered, p01_163_filtered, p04_006_filtered)
rm(tonsil.1a.filtered, tonsil.1b.filtered, tonsil.2a.filtered, tonsil.2b.filtered, tonsil.3a.filtered, tonsil.3b.filtered)
```
########################################################################################################################################################################################################
Integration with Seurat SCTransform. As per - https://satijalab.org/seurat/articles/integration_introduction.html
Involves running SCTransform on filtered ctrl data, then on WM data. Then selecting integration anchors, finding integration anchors, and integrating.
Had some serious difficulty getting FindIntegrationAnchors to run. Ultimately had to run "rpca" reduction AND specify the dims to use to match those used in SCTransform
########################################################################################################################################################################################################

Load libraries
```{r}
library(Seurat)
library(ggplot2)
library(dplyr)
options(future.globals.maxSize=10000000000000000)
```
Load data
```{r}
wm.rna <- readRDS("/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_WM_MergedQCFiltered.rds")
ctrl.rna <- readRDS("/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_Ctrl_MergedQCFiltered.rds")
```
SCTransform on both WM and ctrl
```{r}
ctrl.rna <- SCTransform(ctrl.rna, assay = "RNA", new.assay.name = "SCT", vst.flavor = "v2", ncells = 5000, vars.to.regress = "batch", method = "glmGamPoi") %>%
  RunPCA(npcs = 20) %>%
  RunUMAP(reduction = "pca", dims = 1:20) %>%
  FindNeighbors(reduction = "pca", dims = 1:20) %>%
  FindClusters(resolution = 0.7)

wm.rna <- SCTransform(wm.rna, assay = "RNA", new.assay.name = "SCT", vst.flavor = "v2", ncells = 5000, vars.to.regress = "batch", method = "glmGamPoi", verbose = F) %>%
  RunPCA(npcs = 20)
```
Integration set up
```{r}
ifnb.list <- list(ctrl = ctrl.rna, stim = wm.rna)
features <- SelectIntegrationFeatures(object.list = ifnb.list)
ifnb.list <- PrepSCTIntegration(object.list = ifnb.list, anchor.features = features)
rm(wm.rna, ctrl.rna)
```
Perform integration
```{r}
anchors <- FindIntegrationAnchors(object.list = ifnb.list, normalization.method = "SCT", reduction = "rpca",
    anchor.features = features, dims = 1:20)
integrated <- IntegrateData(anchorset = anchors, normalization.method = "SCT", dims = 1:20)
```
Integrated Analysis
```{r}
integrated <- RunPCA(integrated, dims = 1:20)
integrated <- RunUMAP(integrated, reduction = "pca", dims = 1:20)
integrated <- FindNeighbors(integrated, reduction = "pca", dims = 1:20)
integrated <- FindClusters(integrated, resolution = 0.7)
DimPlot(integrated, reduction = "umap")
```
Bonus adding forgetten metadata
```{r}
integrated@meta.data$case.control <- integrated@meta.data$source
integrated@meta.data$case.control[integrated@meta.data$source == "NYU"] <- "WM"
integrated@meta.data$case.control[integrated@meta.data$source == "Stanford"] <- "HD"
table(integrated@meta.data$case.control)
```
Save out
```{r}
saveRDS(integrated, "/Users/gagled01/morganLab/Waldenstroms/singlecell/objects/scRNA_SCTransformed_Integrated.rds")
```
