---
title: "WM_SNV_indels"
output: html_document
date: "2024-05-14"
---

```{r}
library(maftools)
library(devgru)
library(vcfR)
```

```{r}
maf <- read.maf("/Users/gagled01/morganLab/Waldenstroms/WGS/mafs/wm.hq.snv.indel.maf.gz")
```
# Create new clinical subtypes
```{r}
# Overall subtypes
maf@clinical.data$subtype <- "NA"
maf@clinical.data$subtype[maf@clinical.data$Tumor_Sample_Barcode %in% c("01-076_B_S10_001", "01-190_B_S6_001", "WM46_B_S4_001", "WM47_B_S2_001", "WM65_B_S5_001",
                                                                        "WM33_CD19Pos", "WM52_CD19Pos_KappaPos", "WM70_CD19Pos_KappaPos", "WM9_CD19pos", "WM31_CD19Pos")] <- "MBC-like"

maf@clinical.data$subtype[maf@clinical.data$Tumor_Sample_Barcode %in% c("01-115_B_S11_001", "01-131_B_S9_001", "01-163_B_S7_001", "04-006_B_S8_001", "WM54_B_S3_001",
                                                                        "WM37_CD19Pos", "WM32_CD19Pos", "WM42_CD19Pos", "WM69_CD19Pos_KappaPos", "WM27_CD19Pos", "WM30_CD19Pos",
                                                                        "WM66_CD19Pos_GammaPos", "WM29_CD19Pos", "WM8_CD19Pos", "WM22_CD20Pos", "WM72_CD19Pos_LambdaNeg", "WM26_CD19Pos",
                                                                        "WM43_B_S1_001", "WM48_CD19Pos_KappaNeg", "WM68_CD19Pos_KappaPos", "WM73_CD19Pos_KappaNeg", "WM49_CD19Pos_KappaNeg")] <- "PC-like"
table(maf@clinical.data$subtype)

v1 <- maf@clinical.data$Tumor_Sample_Barcode
v2 <- maf@clinical.data$subtype

df <- data.frame(v1, v2)
df
```
# Annotate MAF with CNV and mutations
```{r}
# annotate del 6q
maf@clinical.data$del6q <- "-"
maf@clinical.data$del6q[maf@clinical.data$Tumor_Sample_Barcode %in% c("01-190_B_S6_001", "WM37_CD19Pos", "WM42_CD19Pos", "WM48_CD19Pos_KappaNeg", "WM66_CD19Pos_GammaPos",
                                                           "WM68_CD19Pos_KappaPos", "WM69_CD19Pos_KappaPos", "WM70_CD19Pos_KappaPos", "WM73_CD19Pos_KappaNeg")] <- "+"
maf@clinical.data$del6q <- as.factor(maf@clinical.data$del6q)
table(maf@clinical.data$del6q)

# trisomy 4
maf@clinical.data$tri4 <- "-"
maf@clinical.data$tri4[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM26_CD19Pos", "WM30_CD19Pos", "WM33_CD19Pos", "WM37_CD19Pos", "WM70_CD19Pos_KappaPos",
                                                           "WM72_CD19Pos_LambdaNeg", "WM8_CD19Pos")] <- "+"
table(maf@clinical.data$tri4)

# del 13p
maf@clinical.data$del13p <- "-"
maf@clinical.data$del13p[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM43_B_S1_001", "WM46_B_S4_001", "WM47_B_S2_001", "WM54_B_S3_001", "WM33_CD19Pos",
                                                           "WM49_CD19Pos_KappaNeg", "WM68_CD19Pos_KappaPos")] <- "+"
table(maf@clinical.data$del13p)

# del 22p
maf@clinical.data$del22p <- "-"
maf@clinical.data$del22p[maf@clinical.data$Tumor_Sample_Barcode %in% c("04-006_B_S8_001", "WM43_B_S1_001", "WM47_B_S2_001", "WM65_B_S5_001", "WM29_CD19Pos",
                                                           "WM37_CD19Pos", "WM48_CD19Pos_KappaNeg", "WM68_CD19Pos_KappaPos", "WM70_CD19Pos_KappaPos")] <- "+"
table(maf@clinical.data$del22p)

# del 22p
maf@clinical.data$tri18 <- "-"
maf@clinical.data$tri18[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM26_CD19Pos", "WM30_CD19Pos", "WM72_CD19Pos_LambdaNeg", "WM8_CD19Pos")] <- "+"
maf@clinical.data$tri18 <- as.factor(maf@clinical.data$tri18)
table(maf@clinical.data$tri18)

# del 17p
maf@clinical.data$del17p <- "-"
maf@clinical.data$del17p[maf@clinical.data$Tumor_Sample_Barcode %in% c("01-190_B_S6_001", "WM43_B_S1_001", "WM47_B_S2_001")] <- "+"
maf@clinical.data$del17p <- as.factor(maf@clinical.data$del17p)
table(maf@clinical.data$del17p)

# cxcr4
maf@clinical.data$myd88 <- "+"
maf@clinical.data$myd88[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM49_CD19Pos_KappaNeg")] <- "+"
maf@clinical.data$myd88 <- as.factor(maf@clinical.data$myd88)
table(maf@clinical.data$myd88)

maf@clinical.data$cxcr4 <- "-"
maf@clinical.data$cxcr4[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM70_CD19Pos_KappaPos", "01-190_B_S6_001", "WM33_CD19Pos", "WM26_CD19Pos", "WM31_CD19Pos",
                                                                      "WM72_CD19Pos_LambdaNeg", "WM65_B_S5_001", "04-006_B_S8_001")] <- "+"
maf@clinical.data$cxcr4 <- as.factor(maf@clinical.data$cxcr4)
table(maf@clinical.data$cxcr4)

# h1-4
maf@clinical.data$hist1h1e <- "-"
maf@clinical.data$hist1h1e[maf@clinical.data$Tumor_Sample_Barcode %in% c("01-076_B_S10_001", "WM65_B_S5_001", "WM46_B_S4_001", "WM46_B_S4_001", "WM42_CD19Pos")] <- "+"
maf@clinical.data$hist1h1e <- as.factor(maf@clinical.data$hist1h1e)
table(maf@clinical.data$hist1h1e)

# arid1a
maf@clinical.data$arid1a <- "-"
maf@clinical.data$arid1a[maf@clinical.data$Tumor_Sample_Barcode %in% c("01-190_B_S6_001", "WM70_CD19Pos_KappaPos", "WM32_CD19Pos", "WM29_CD19Pos", "WM47_B_S2_001")] <- "+"
maf@clinical.data$arid1a <- as.factor(maf@clinical.data$arid1a)
table(maf@clinical.data$arid1a)

# nik
maf@clinical.data$nik <- "-"
maf@clinical.data$nik[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM46_B_S4_001", "WM47_B_S2_001", "WM9_CD19pos")] <- "+"
maf@clinical.data$nik <- as.factor(maf@clinical.data$nik)
table(maf@clinical.data$nik)

# sp140
maf@clinical.data$sp140 <- "-"
maf@clinical.data$sp140[maf@clinical.data$Tumor_Sample_Barcode %in% c("01-115_B_S11_001", "01-163_B_S7_001", "01-163_B_S7_001")] <- "+"
maf@clinical.data$sp140 <- as.factor(maf@clinical.data$sp140)
table(maf@clinical.data$sp140)

# tp53
maf@clinical.data$tp53 <- "-"
maf@clinical.data$tp53[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM70_CD19Pos_KappaPos", "WM33_CD19Pos", "WM68_CD19Pos_KappaPos")] <- "+"
maf@clinical.data$tp53 <- as.factor(maf@clinical.data$tp53)
table(maf@clinical.data$tp53)

# kmt2d
maf@clinical.data$kmt2d <- "-"
maf@clinical.data$kmt2d[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM32_CD19Pos", "WM47_B_S2_001", "WM46_B_S4_001", "WM9_CD19pos")] <- "+"
maf@clinical.data$kmt2d <- as.factor(maf@clinical.data$kmt2d)
table(maf@clinical.data$kmt2d)

# igll5
maf@clinical.data$igll5 <- "-"
maf@clinical.data$igll5[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM70_CD19Pos_KappaPos", "01-190_B_S6_001", "WM33_CD19Pos", "WM26_CD19Pos",
                                                                      "WM31_CD19Pos", "WM72_CD19Pos_LambdaNeg", "01-076_B_S10_001", "WM32_CD19Pos",
                                                                      "WM29_CD19Pos", "01-115_B_S11_001", "01-131_B_S9_001", "WM22_CD20Pos",
                                                                      "WM30_CD19Pos", "WM43_B_S1_001", "WM54_B_S3_001", "WM73_CD19Pos_KappaNeg",
                                                                      "WM8_CD19Pos")] <- "+"
maf@clinical.data$igll5 <- as.factor(maf@clinical.data$igll5)
table(maf@clinical.data$igll5)

# sex
maf@clinical.data$sex <- "M"
maf@clinical.data$sex[maf@clinical.data$Tumor_Sample_Barcode %in% c("WM26_CD19Pos", "WM29_CD19Pos", "WM30_CD19Pos", "WM43_B_S1_001", "WM47_B_S2_001", "WM52_CD19Pos_KappaPos", "WM54_B_S3_001",
                                                                    "WM67_CD19Pos_KappaPos", "WM68_CD19Pos_KappaPos", "01-076_B_S10_001", "01-131_B_S9_001", "01-190_B_S6_001")] <- "F"
maf@clinical.data$sex <- as.factor(maf@clinical.data$sex)
table(maf@clinical.data$sex)

# Write out updated maf
write.mafSummary(maf, "/Users/gagled01/morganLab/Waldenstroms/WGS/mafs/wm.hq.snv.indel.ClinDataUpdated.maf.gz")
```
# Subset mafs by clin data
```{r}
# Get sample ids by subtype
mbc.samps <- maf@clinical.data$Tumor_Sample_Barcode[maf@clinical.data$subtype == "MBC-like"]
pc.samps <- maf@clinical.data$Tumor_Sample_Barcode[maf@clinical.data$subtype == "PC-like"]

#Once you have your sample list, use subsetMaf
mbc.maf = subsetMaf(maf = maf, tsb = mbc.samps, mafObj = TRUE)
length(mbc.maf@clinical.data$Tumor_Sample_Barcode)
pc.maf = subsetMaf(maf = maf, tsb = pc.samps, mafObj = TRUE)
length(pc.maf@clinical.data$Tumor_Sample_Barcode)

# WM49 is being dropped from the int objects due to having no non-synonymous variants!!!

# get a maf where the WGS without RNA data are excluded
maf.sub.samps <- maf@clinical.data$Tumor_Sample_Barcode[maf@clinical.data$subtype %in% c ("PC-like", "MBC-like")]

#Once you have your sample list, use subsetMaf
sub.maf = subsetMaf(maf = maf, tsb = maf.sub.samps, mafObj = TRUE)
length(sub.maf@clinical.data$Tumor_Sample_Barcode)

maf@gene.summary[maf@gene.summary$Hugo_Symbol == "SPI1",]
```
# pval for sex and age
```{r}
sub.maf@clinical.data %>% 
  dplyr::select(sex, subtype) %>%
  drop_na(sex, subtype) %>%
  summarise(fisher.test(subtype, sex)$p.value)
round(as.numeric(lymph.pval), 3)
```

# Get patients with CXCR4 and del6q
```{r}
maf.cxcr4 <- maf@clinical.data$Tumor_Sample_Barcode[maf@clinical.data$cxcr4 == "+"]
maf.del6q <- maf@clinical.data$Tumor_Sample_Barcode[maf@clinical.data$del6q == "+"]

```

# Check MYD88 patient ID
```{r}
pdf(file = "/Users/gagled01/morganLab/MYD88_IDCheck.pdf", width = 15, height = 10)
oncoplot(sub.maf, removeNonMutated = F, showTumorSampleBarcodes = T)
dev.off()
```
# get del6q patient
```{r}
del.6q.samps <- maf@clinical.data$Tumor_Sample_Barcode[maf@clinical.data$del6q == "+"]
del6q.maf <- subsetMaf(maf = maf, tsb = del.6q.samps, mafObj = TRUE)
del6q.maf@clinical.data$Tumor_Sample_Barcode

pdf(file = "FullMutationalPanel.pdf", width = 15, height = 10)
oncoplot(maf = sub.maf, genes = c("MYD88", "IGLL5", "CXCR4", "H1-4", "MAP3K14", "ARID1A", "SP140", "CD79B", "TP53", "EZH2", "CARD11", "KMT2D"), showTumorSampleBarcodes = T,  removeNonMutated = FALSE, legend_height = 0.1, legendFontSize = 0.1)
dev.off()
```
Make subtype comparison summary plot
```{r}
pdf(file = "SubtypeOncoplotAnnotatedDel6p2.pdf", width = 15, height = 10)
coOncoplot(mbc.maf, pc.maf, m1Name = 'MBC-like', m2Name = 'PC-like', genes = c("MYD88", "IGLL5", "CXCR4", "H1-4", "MAP3K14", "ARID1A", "SP140", "CD79B", "TP53", "EZH2", "CARD11", "KMT2D"),  removeNonMutated = FALSE,
           showSampleNames = TRUE)
dev.off()

x <- sub.maf@clinical.data$del6q
y <- sub.maf@clinical.data$subtype_combo

fisher.test(x, y)

pdf(file = "SubtypeOncoplotAnnotatedDel6p.pdf", width = 18, height = 10)
coOncoplot(mbc.maf, pc.maf, m1Name = 'MBC-like', m2Name = 'PC-like', genes = c("MYD88", "IGLL5", "CXCR4", "H1-4", "MAP3K14", "ARID1A", "SP140", "CD79B", "TP53", "EZH2", "CARD11", "KMT2D"),  removeNonMutated = FALSE,
           geneNamefont = 1, gene_mar = 1.4,
           outer_mar = 4,)
dev.off()
```
Testing stats of mutations
```{r}
# CXCR4
cxcr4.test <- maf@clinical.data %>% 
  dplyr::select(cxcr4, subtype) %>%
  drop_na(cxcr4, subtype) %>%
  summarise(fisher.test(subtype, cxcr4)$p.value)
cxcr4.pval <- round(as.numeric(cxcr4.test), 3)

# HIST1H1E
hist1h1e.test <- maf@clinical.data %>% 
  dplyr::select(hist1h1e, subtype) %>%
  drop_na(hist1h1e, subtype) %>%
  summarise(fisher.test(subtype, hist1h1e)$p.value)
hist1h1e.pval <- round(as.numeric(hist1h1e.test), 3)

# ARID1A
arid1a.test <- maf@clinical.data %>% 
  dplyr::select(arid1a, subtype) %>%
  drop_na(arid1a, subtype) %>%
  summarise(fisher.test(subtype, arid1a)$p.value)
arid1a.pval <- round(as.numeric(arid1a.test), 3)

# SP140
sp140.test <- maf@clinical.data %>% 
  dplyr::select(sp140, subtype) %>%
  drop_na(sp140, subtype) %>%
  summarise(fisher.test(subtype, sp140)$p.value)
sp140.pval <- round(as.numeric(sp140.test), 3)

# NIK
nik.test <- maf@clinical.data %>% 
  dplyr::select(nik, subtype) %>%
  drop_na(nik, subtype) %>%
  summarise(fisher.test(subtype, nik)$p.value)
nik.pval <- round(as.numeric(nik.test), 3)

# TP53
tp53.test <- maf@clinical.data %>% 
  dplyr::select(tp53, subtype) %>%
  drop_na(tp53, subtype) %>%
  summarise(fisher.test(subtype, tp53)$p.value)
tp53.pval <- round(as.numeric(tp53.test), 3)

# igll5
igll5.test <- maf@clinical.data %>% 
  dplyr::select(igll5, subtype) %>%
  drop_na(igll5, subtype) %>%
  summarise(fisher.test(subtype, igll5)$p.value)
igll5.pval <- round(as.numeric(igll5.test), 3)

# kmt2d
kmt2d.test <- maf@clinical.data %>% 
  dplyr::select(kmt2d, subtype) %>%
  drop_na(kmt2d, subtype) %>%
  summarise(fisher.test(subtype, kmt2d)$p.value)
kmt2d.pval <- round(as.numeric(kmt2d.test), 3)

# del6q
del6q.test <- maf@clinical.data %>% 
  dplyr::select(del6q, subtype) %>%
  drop_na(del6q, subtype) %>%
  summarise(fisher.test(subtype, del6q)$p.value)
del6q.pval <- round(as.numeric(del6q.test), 3)

mutation.df <- as.data.frame(matrix(nrow = 9, ncol = 4))
colnames(mutation.df) <- c("Aberration", "MBC-like", "PC-like", "p-value")
mutation.df$Aberration <- c("CXCR4", "IGLL5", "HIST1H1E", "ARID1A", "KMT2D", "SP140", "NIK", "TP53", "del6q")
mutation.df$`p-value` <- c(cxcr4.pval, igll5.pval, hist1h1e.pval, arid1a.pval, kmt2d.pval, sp140.pval, nik.pval, tp53.pval, del6q.pval)
mutation.df$`MBC-like` <- c(as.numeric(table(mbc.maf@clinical.data$cxcr4)[2]),
                            as.numeric(table(mbc.maf@clinical.data$igll5)[2]),
                            as.numeric(table(mbc.maf@clinical.data$hist1h1e)[2]),
                            as.numeric(table(mbc.maf@clinical.data$arid1a)[2]),
                            as.numeric(table(mbc.maf@clinical.data$kmt2d)[2]),
                            as.numeric(table(mbc.maf@clinical.data$sp140)[2]),
                            as.numeric(table(mbc.maf@clinical.data$nik)[2]),
                            as.numeric(table(mbc.maf@clinical.data$tp53)[2]),
                            as.numeric(table(mbc.maf@clinical.data$del6q)[2]))
mutation.df$`PC-like` <- c(as.numeric(table(pc.maf@clinical.data$cxcr4)[2]),
                            as.numeric(table(pc.maf@clinical.data$igll5)[2]),
                            as.numeric(table(pc.maf@clinical.data$hist1h1e)[2]),
                            as.numeric(table(pc.maf@clinical.data$arid1a)[2]),
                            as.numeric(table(pc.maf@clinical.data$kmt2d)[2]),
                            as.numeric(table(pc.maf@clinical.data$sp140)[2]),
                            as.numeric(table(pc.maf@clinical.data$nik)[2]),
                            as.numeric(table(pc.maf@clinical.data$tp53)[2]),
                            as.numeric(table(pc.maf@clinical.data$del6q)[2]))
mutation.df
```

```{r}
gistic <- read.table("/Users/gagled01/morganLab/Waldenstroms/WGS/cnvs/scores.gistic", sep = "\t", header = T)
gistic.highfreq <- gistic[gistic$frequency > 0.05,]

gistic.6q <- gistic[gistic$Chromosome == "6",]

jabba <- read.table("/Users/gagled01/morganLab/Waldenstroms/WGS/cnvs/wm.hq.jabba.cnv.bed", sep = "\t", header = T)
jabba

chr6 <- jabba[jabba$chrom == "chr6",]
del6q <- chr6[chr6$total < 2 & chr6$width > 3000000,] # for normal dels
loh6 <- chr6[chr6$total == 2 & chr6$minor != 1 & chr6$width > 3000000,] # for cases where we have copy neutral LOH

chr4 <- jabba[jabba$chrom == "chr4",]
trisomy.4 <- chr4[chr4$total > 2,]

chr22 <- jabba[jabba$chrom == "chr22",]
del22 <- chr22[chr22$total < 2 & chr22$width > 3000000,]
loh22 <- chr22[chr22$total == 2 & chr22$minor != 1 & chr22$width > 3000000,]

chr8 <- jabba[jabba$chrom == "chr8",]
amp8 <- chr8[chr8$total > 2 & chr8$width > 3000000,]
loh8 <- chr8[chr8$total == 2 & chr8$minor != 1 & chr8$width > 3000000,]

chr12 <- jabba[jabba$chrom == "chr12",]
gain12 <- chr12[chr12$total > 2 & chr12$width > 3000000,]

chr19 <- jabba[jabba$chrom == "chr19",]
del19 <- chr19[chr19$total < 2 & chr19$width > 3000000,]
loh19 <- chr19[chr19$total == 2 & chr19$minor != 1 & chr19$width > 3000000,]

chr17 <- jabba[jabba$chrom == "chr17",]
del17 <- chr17[chr17$total < 2 & chr17$width > 3000000,]
loh17 <- chr17[chr17$total == 2 & chr17$minor != 1 & chr17$width > 3000000,]

chr13 <- jabba[jabba$chrom == "chr13",]
del13 <- chr13[chr13$total < 2 & chr13$width > 3000000,]
loh13 <- chr13[chr13$total == 2 & chr13$minor != 1 & chr13$width > 3000000,]

chr18 <- jabba[jabba$chrom == "chr18",]
gain18 <- chr18[chr18$total > 2 & chr18$width > 3000000,]

chr3 <- jabba[jabba$chrom == "chr3",]
gain3 <- chr3[chr3$total > 2 & chr3$width > 3000000,]
loh3 <- chr3[chr3$total == 2 & chr3$minor != 1 & chr3$width > 3000000,]


chr5 <- jabba[jabba$chrom == "chr5",]
any5 <- chr5[chr5$total != 2 & chr5$width > 3000000,]
loh5 <- chr5[chr5$total == 2 & chr5$minor != 1 & chr5$width > 3000000,]

chr10 <- jabba[jabba$chrom == "chr10",]
any10 <- chr10[chr10$total != 2 & chr10$width > 3000000,]
loh10 <- chr10[chr10$total == 2 & chr10$minor != 1 & chr10$width > 3000000,]

wm27 <- jabba[jabba$id == "WM27_CD19Pos" & jabba$width > 3000000 & jabba$minor < 1,]
table(wm27$chrom)

wm42 <- jabba[jabba$id == "WM42_CD19Pos" & jabba$width > 3000000 & jabba$minor < 1,]
table(wm42$chrom)

wm31 <- jabba[jabba$id == "WM31_CD19Pos" & jabba$width > 3000000 & jabba$minor < 1,]
table(wm31$chrom)
```

```{r}
oncoplot(mbc.maf, showTumorSampleBarcodes = T, legend_height = 0.1, genes = c("MYD88", "IGLL5", "CXCR4", "H1-4", "MAP3K14", "ARID1A", "SP140", "CD79B", "TP53", "EZH2", "CARD11", "KMT2D", "SPI1"))
oncoplot(pc.maf, showTumorSampleBarcodes = T, genes = c("MYD88", "IGLL5", "CXCR4", "H1-4", "MAP3K14", "ARID1A", "SP140", "CD79B", "TP53", "EZH2", "CARD11", "KMT2D"))

oncoplot(maf, showTumorSampleBarcodes = T, genes = c("MYD88", "SPI1"), removeNonMutated = F)

maf@data[, t_vaf := t_alt_count /( t_alt_count+t_ref_count) * 100] # main data slot with non-syn variants
mbc.maf@data[, t_vaf := t_alt_count /( t_alt_count+t_ref_count) * 100] # main data slot with non-syn variants
pc.maf@data[, t_vaf := t_alt_count /( t_alt_count+t_ref_count) * 100] # main data slot with non-syn variants

mbc.cxcr4 <- mbc.maf@data[mbc.maf@data$Hugo_Symbol == "CXCR4",]
pc.cxcr4 <- pc.maf@data[pc.maf@data$Hugo_Symbol == "CXCR4",]

mbc.cxcr4.vafs <- data.frame(sample = mbc.cxcr4$Tumor_Sample_Barcode, vaf = mbc.cxcr4$t_vaf)
ggplot(mbc.cxcr4.vafs, aes(x = sample, y = vaf, fill = sample)) + geom_boxplot()

pc.cxcr4.vafs <- data.frame(sample = pc.cxcr4$Tumor_Sample_Barcode, vaf = pc.cxcr4$t_vaf)
ggplot(pc.cxcr4.vafs, aes(x = sample, y = vaf, fill = sample)) + geom_boxplot()

plotVaf(mbc.maf, genes = c("CXCR4"))
plotVaf(pc.maf, genes = c("CXCR4"))

maf@clinical.data$Tumor_Sample_Barcode

# Check WM54
wm54.maf = subsetMaf(maf = maf, tsb = c("01-115_B_S11_001", "WM54_B_S3_001"), mafObj = TRUE)
```

# Tumor mutational burden
```{r}
# comparing to 33 TCGA cohorts
pdf(file = "/Users/gagled01/morganLab/Waldenstroms/singlecell/figures/FinalFigures/SuppFigure18_TumorMutationalBurden_Compare.pdf", height = 8, width = 8)
laml.mutload = tcgaCompare(maf = c(mbc.maf, pc.maf), cohortName = c("MBC-like", "PC-like"), logscale = TRUE, capture_size = 50)
dev.off()

tcgaCompare(maf = c(mbc.maf, pc.maf), cohortName = c("MBC-like", "PC-like"), logscale = TRUE, capture_size = 50)

mbc.tmb <- tmb(mbc.maf)
mbc.tmb$subtype <- "MBC-like"

pc.tmb <- tmb(pc.maf)
pc.tmb$subtype <- "PC-like"

mbc.tmb
pc.tmb

median(mbc.tmb$total_perMB)
median(pc.tmb$total_perMB)

tmb_perMB <- c(mbc.tmb$total_perMB, pc.tmb$total_perMB)
samples <- c(mbc.tmb$Tumor_Sample_Barcode, pc.tmb$Tumor_Sample_Barcode)
subtype <- c(mbc.tmb$subtype, pc.tmb$subtype)

tmbs
samples
compare.df <- data.frame(samples, tmb_perMB, subtype)
compare.df

t.test(as.numeric(compare.df$tmb_perMB) ~ as.character(compare.df$subtype), alternative = "two.sided")

tmb(mbc.maf)
tmb(pc.maf)
```
```{r}
mbc.patients <- mbc.maf@clinical.data$Tumor_Sample_Barcode
mbc.subtype <- c(replicate(10, "MBC-like"))
mbc.cxcr4 <- c(replicate(5, "+"), replicate(5, "-"))
mbc.h1.4 <- c(replicate(4, "+"), replicate(6, "-"))
mbc.nik <- c(replicate(3, "+"), replicate(7, "-"))
mbc.arid1a <- c(replicate(3, "+"), replicate(7, "-"))
mbc.sp140 <- c(replicate(0, "+"), replicate(10, "-"))
mbc.df <- data.frame(mbc.patients, mbc.subtype, mbc.cxcr4, mbc.h1.4, mbc.nik, mbc.arid1a, mbc.sp140)

pc.patients <- pc.maf@clinical.data$Tumor_Sample_Barcode
pc.subtype <- c(replicate(22, "PC-like"))
pc.cxcr4 <- c(replicate(3, "+"), replicate(19, "-"))
pc.h1.4 <- c(replicate(1, "+"), replicate(21, "-"))
pc.nik <- c(replicate(0, "+"), replicate(22, "-"))
pc.arid1a <- c(replicate(2, "+"), replicate(20, "-"))
pc.sp140 <- c(replicate(3, "+"), replicate(19, "-"))


replicate(3, "balls")
```
# Get vafs
```{r}
pdf("/Users/gagled01/morganLab/Waldenstroms/Revisions_Round1/VAF_Barplot_RelevantGenes.pdf", height = 6, width = 10)
plotVaf(maf, keepGeneOrder = T, genes = c("MYD88", "IGLL5", "CXCR4", "H1-4", "MAP3K14", "ARID1A", "SP140", "CD79B", "TP53", "KMT2D"))
dev.off()
```

